// bluerain-app-device-explorer/lib/components/Event/EventDetailPage/EventDetailPage.widget.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactFa = require('react-fa');

var _widget = require('bluerain-ui/lib/Widget/widget');

var _widget2 = _interopRequireDefault(_widget);

var _widgetHeader = require('bluerain-ui/lib/Widget/widgetHeader');

var _widgetHeader2 = _interopRequireDefault(_widgetHeader);

var _widgetTitle = require('bluerain-ui/lib/Widget/widgetTitle');

var _widgetTitle2 = _interopRequireDefault(_widgetTitle);

var _widgetContent = require('bluerain-ui/lib/Widget/widgetContent');

var _widgetContent2 = _interopRequireDefault(_widgetContent);

var _EventDetail = require('./EventDetailContainer/EventDetail.container');

var _EventDetail2 = _interopRequireDefault(_EventDetail);

var _EventParameter = require('./EventParameterContainer/EventParameter.container');

var _EventParameter2 = _interopRequireDefault(_EventParameter);

var _Device = require('./DeviceContainer/Device.container');

var _Device2 = _interopRequireDefault(_Device);

var _PostCard = require('./PostCardComponent/PostCard.component');

var _PostCard2 = _interopRequireDefault(_PostCard);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Created by anser on 12/30/16.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


var propTypes = {
  data: _react.PropTypes.object
};

var defaultProps = {};

var EventDetailWidget = function (_React$Component) {
  _inherits(EventDetailWidget, _React$Component);

  function EventDetailWidget() {
    _classCallCheck(this, EventDetailWidget);

    return _possibleConstructorReturn(this, (EventDetailWidget.__proto__ || Object.getPrototypeOf(EventDetailWidget)).apply(this, arguments));
  }

  _createClass(EventDetailWidget, [{
    key: 'render',
    value: function render() {
      if (this.props.data.node) {
        var event = this.props.data.node.events.edges[0];
        if (event === null || event === undefined) {
          // exception handling
          return _react2.default.createElement(
            'div',
            { className: 'app-content' },
            _react2.default.createElement(
              'div',
              { className: 'container' },
              _react2.default.createElement(
                'div',
                { className: 'v-align' },
                _react2.default.createElement(
                  'div',
                  { className: 'v-align-middle empty-block' },
                  _react2.default.createElement(_reactFa.Icon, { name: 'home fa-5x' }),
                  _react2.default.createElement(
                    'h4',
                    null,
                    'No events found for this post'
                  )
                )
              )
            )
          );
        }
        return _react2.default.createElement(
          'div',
          null,
          EventDetailWidget.renderPostCardWidget(this.props.data),
          EventDetailWidget.renderDetailWidget(event),
          EventDetailWidget.renderParameterWidget(event),
          EventDetailWidget.renderDeviceWidget(event)
        );
      }
      return _react2.default.createElement('div', null);
    }
  }], [{
    key: 'renderDeviceWidget',
    value: function renderDeviceWidget(data) {
      if (data.node.sourceDevice === null) {
        return _react2.default.createElement('div', null);
      }
      return _react2.default.createElement(
        _widget2.default,
        null,
        _react2.default.createElement(
          _widgetHeader2.default,
          null,
          _react2.default.createElement(
            _widgetTitle2.default,
            null,
            'Device'
          )
        ),
        _react2.default.createElement(
          _widgetContent2.default,
          null,
          _react2.default.createElement(_Device2.default, { data: data })
        )
      );
    }
  }, {
    key: 'renderParameterWidget',
    value: function renderParameterWidget(data) {
      return _react2.default.createElement(
        _widget2.default,
        null,
        _react2.default.createElement(
          _widgetHeader2.default,
          null,
          _react2.default.createElement(
            _widgetTitle2.default,
            null,
            'Event Parameters'
          )
        ),
        _react2.default.createElement(
          _widgetContent2.default,
          null,
          _react2.default.createElement(_EventParameter2.default, { data: data })
        )
      );
    }
  }, {
    key: 'renderDetailWidget',
    value: function renderDetailWidget(data) {
      return _react2.default.createElement(
        _widget2.default,
        null,
        _react2.default.createElement(
          _widgetHeader2.default,
          null,
          _react2.default.createElement(
            _widgetTitle2.default,
            null,
            'Event Details'
          )
        ),
        _react2.default.createElement(
          _widgetContent2.default,
          null,
          _react2.default.createElement(_EventDetail2.default, { data: data })
        )
      );
    }
  }, {
    key: 'renderPostCardWidget',
    value: function renderPostCardWidget(data) {
      if (data.node.events.edges[0].node.asset === null) {
        return _react2.default.createElement('div', null);
      }
      return _react2.default.createElement(
        _widget2.default,
        null,
        _react2.default.createElement(
          _widgetHeader2.default,
          null,
          _react2.default.createElement(
            _widgetTitle2.default,
            null,
            data.node.events.edges[0].node.asset.name
          )
        ),
        _react2.default.createElement(
          _widgetContent2.default,
          null,
          _react2.default.createElement(_PostCard2.default, { data: data })
        )
      );
    }
  }]);

  return EventDetailWidget;
}(_react2.default.Component);

exports.default = EventDetailWidget;


EventDetailWidget.propTypes = propTypes;
EventDetailWidget.defaultProps = defaultProps;
