// bluerain-app-device-explorer/src/components/Device/DeviceSettings/DeviceGeneralSettings/AssetUpdateForm.js
/**
 * Created by anser on 1/17/17.
 */
import React, { PropTypes } from 'react';
import { graphql } from 'react-apollo';
// import { withNotifications } from 'bluerain-client-services';
import { fromGlobalId } from 'graphql-relay';
import ListViewItem from 'bluerain-ui/lib/ListView/ListViewItem';
import { withNotifications } from '../../../Notifications/withNotifications';
import AssetUpdateMutation from './AssetUpdateMutation.graphql';


const propTypes = {
  name: PropTypes.string.isRequired,
  data: PropTypes.object,
  onNameChange: PropTypes.func,
  onSubmit: PropTypes.func,
  notification: PropTypes.object,
  mutate: PropTypes.func
};

const defaultProps = {
  name: '',
  onSubmit: () => {
  },
};

let limitLength;
let extract;
let extraValidations;
class AssetUpdateForm extends React.Component {

  constructor(props) {
    super(props);
    // ********** VALIDATIONS *****************
    limitLength = (str, length) => str.substring(0, length);
    extract = (str, pattern) => (str.match(pattern) || []).pop() || '';
    extraValidations = str => extract(str, '[0-9a-zA-Z: ]+');
    // ********** VALIDATIONS *****************

    this.state = {
      value: '',
      onNameChangeCallback: props.onNameChange,
      onSubmitCallback: props.onSubmit
    };

    this.handleChange = this.handleChange.bind(this);
    this.onNameChange = this.onNameChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentWillReceiveProps(props) {
    const state = props.data.node.name;

    this.setState({
      value: state
    });
  }

  onNameChange(event) {
    const state = this.state;
    const name = event.target.value;

    state.data.name = name;

    this.state.onNameChangeCallback(name);
    this.setState(state);
  }


  // Form on submit
  onSubmit(event) {
    const currentName = this.props.data.node.name;
    const newName = this.state.value;
    const { addNotification } = this.props.notification;
    if (!newName) {
      addNotification({
        title: 'Missing',
        message: 'Please provide Asset name',
        status: 'error',
        dismissible: true,
        dismissAfter: 3000,
        position: 'br',
        level: 'error'
      });
      event.preventDefault();
      return false;
    }
    if (newName === currentName) {
      addNotification({
        title: 'Warning',
        message: 'Please provide a new asset name',
        status: 'error',
        dismissible: true,
        dismissAfter: 3000,
        position: 'br',
        level: 'error'
      });
      event.preventDefault();
      return false;
    }
    const variables = {
      where: {
        id: fromGlobalId(this.props.data.node.id).id
      },
      data: {
        name: newName
      }
    };
    // mutation to update asset
    this.props.mutate({
      variables
    });
    addNotification({
      title: 'Success',
      message: 'Asset name updated',
      status: 'success',
      dismissible: true,
      dismissAfter: 3000,
      position: 'br',
      level: 'error'
    });
    // browserHistory.push('/app/device-explorer');
    event.preventDefault();
  }

  // Get value of asset name
  handleChange(event) {
    this.setState({
      value: limitLength(extraValidations(event.target.value), 25)
    });
  }


  render() {
    if (!this.props.data.node || !this.props.data.node.name) {
      return (<div />);
    }
    const {
      id,
    } = this.props.data.node;
    return (
      <form onSubmit={this.onSubmit}>
        <ListViewItem>
          <label htmlFor="name">Asset Name</label>
          <input
            type="text" value={this.state.value ? this.state.value : this.props.data.node.name}
            onChange={this.handleChange} className="form-control"
            aria-describedby="name" placeholder="Asset Name"
          />
          <input
            type="hidden" value={id} className="form-control"
            aria-describedby="name" placeholder="Asset Name"
          />
          <small id="imageSize" className="form-text text-muted">Select a new name for your asset.</small>
        </ListViewItem>
        <ListViewItem>
          <button type="submit" className="btn btn-primary">Update</button>
        </ListViewItem>
      </form>
    );
  }
}


// export default AssetUpdateForm;

export default
(
  graphql(AssetUpdateMutation)(
    withNotifications(AssetUpdateForm)
  )
);

AssetUpdateForm.propTypes = propTypes;
AssetUpdateForm.defaultProps = defaultProps;
