// bluerain-app-device-explorer/src/pages/Assets/Location/AssetLocationsPage.js
/**
 * Created by anser on 2/28/17.
 */
import React, { PropTypes } from 'react';
import { Row, Col, Button } from 'reactstrap';
import { withSystemNav } from 'bluerain-client-services';
import PrimaryAppNav from 'bluerain-ui/lib/Nav/PrimaryAppNav';
import SecondaryAppNav from 'bluerain-ui/lib/Nav/SecondaryAppNav';
import { Link, withRouter } from 'react-router';
import AssetLocationWidget from '../../../components/Asset/AssetLocation';
import { sitesGridViewUrl, sitesMapViewUrl } from '../../../routes/urlBuilder';

const propTypes = {
  title: PropTypes.string,
  systemNav: PropTypes.func,
  location: PropTypes.func
};

const defaultProps = {
  title: 'Assets Locations'
};

class AssetLocationPage extends React.Component {

  renderPrimaryNav() {
    const title = this.props.title;
    const { toggle } = this.props.systemNav.actions;
    // const previous = {
    //   title: 'Sites',
    //   url: '/app/device-explorer'
    // };
    return (
      <PrimaryAppNav title={title} toggleMenu={toggle}>
        <Link to="/app/device-explorer/site/new"><Button color="primary"><i className="fa fa-plus hidden-sm-up" /><span
          className="hidden-xs-down"
        >Add Site</span></Button></Link>&nbsp;
      </PrimaryAppNav>
    );
  }

  renderSecondaryNav() {
    const { pathname } = this.props.location;
    return (<SecondaryAppNav
      items={[
        {
          label: 'Grid View',
          url: '/app/device-explorer',
          class: (pathname === sitesGridViewUrl()) ? 'active' : ''
        }, {
          label: 'Map View',
          url: '/app/device-explorer/asset_locations',
          class: (pathname === sitesMapViewUrl()) ? 'active' : ''
        }
      ]}
    />);
  }

  static renderContent() {
    return (
      <Row>
        <Col lg="12">
          <AssetLocationWidget />
        </Col>
      </Row>
    );
  }

  render() {
    return (
      <div>
        {this.renderPrimaryNav()}
        {this.renderSecondaryNav()}
        {AssetLocationPage.renderContent()}
      </div>
    );
  }
}

AssetLocationPage.propTypes = propTypes;
AssetLocationPage.defaultProps = defaultProps;

export default withSystemNav(withRouter(AssetLocationPage));

// <Link to="/app/device-explorer/transfer_requests"><Button color="primary"><i className="fa fa-plus hidden-sm-up"  /><span className="hidden-xs-down">Transfer Requests</span></Button></Link>&nbsp;
