// bluerain-app-device-explorer/src/components/Asset/SonosSensorInfo/SonosSensorInfo.component.js
import React, { PropTypes } from 'react';
import Icon from 'react-fa';
import 'bluerain-bootstrap-theme/dist/css/components/data-list.css';
import 'bluerain-bootstrap-theme/dist/css/components/card-view.css';
import CardBlock from 'reactstrap/lib/CardBlock';
import CardViewImg from './CardViewImg';
import CardView from './CardView';

const propTypes = {
  data: PropTypes.object,
};

const defaultProps = {};

export default class SonosSensorInfo extends React.Component {

  static renderConnectionStatus(connected) {

    let status = null;

    if (typeof (connected) !== 'boolean') {
      return (<div />);
    } else if (connected) {
      status = (<span>Online <Icon name="circle text-success" /></span>);
    } else {
      status = (<span>Offline <Icon name="circle text-danger" /></span>);
    }
    return (
      <li className="list-group-item">
        <div className="data-label">Status</div>
        <div className="data-value">{status}</div>
      </li>
    );
  }

  static renderSongName(song, connected) {
    if (!song) {
      return (
        <li className="list-group-item">
          <div className="data-label">Song Name</div>
          <div className="data-value">Unknown</div>
        </li>
      );
    }
    if (connected) {
      return (
        <li className="list-group-item">
          <div className="data-label">Song Name</div>
          <div className="data-value">{song}</div>
        </li>
      );
    }
    return (<div />);
  }

  static renderAlbumName(album, connected) {
    if (!album) {
      return (
        <li className="list-group-item">
          <div className="data-label">Album Name</div>
          <div className="data-value">Unknown</div>
        </li>
      );
    }
    if (connected) {
      return (
        <li className="list-group-item">
          <div className="data-label">Album Name</div>
          <div className="data-value">{album}</div>
        </li>
      );
    }
    return (<div />);
  }

  static renderArtistName(artist, connected) {
    if (!artist) {
      return (
        <li className="list-group-item">
          <div className="data-label">Artist Name</div>
          <div className="data-value">Unknown</div>
        </li>
      );
    }
    if (connected) {
      return (
        <li className="list-group-item">
          <div className="data-label">Artist Name</div>
          <div className="data-value">{artist}</div>
        </li>
      );
    }
    return (<div />);
  }

  static renderImage(image, connected) {
    if (!image) {
      return (
        <CardViewImg image="https://s3-us-west-2.amazonaws.com/bluerainimages/music.png" />
      );
    }
    if (connected) {
      return (
        <CardViewImg image={image} />
      );
    }
    return (<div />);
  }

  render() {
    if (!this.props.data.node || !this.props.data.node.state) {
      return (<div />);
    }

    const {
      connected,
      song,
      album,
      artist,
      image
    } = this.props.data.node.state;

    return (

      <CardView>
        { SonosSensorInfo.renderImage(image, connected) }
        <CardBlock>
          <ul className="list-group data-list">
            { SonosSensorInfo.renderConnectionStatus(connected) }
            { SonosSensorInfo.renderSongName(song, connected) }
            { SonosSensorInfo.renderAlbumName(album, connected) }
            { SonosSensorInfo.renderArtistName(artist, connected) }
          </ul>
        </CardBlock>
      </CardView>
    );
  }
}

SonosSensorInfo.propTypes = propTypes;
SonosSensorInfo.defaultProps = defaultProps;
