// bluerain-app-device-explorer/lib/components/Asset/Charts/TPLinkTimeChart/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _reactApollo = require('react-apollo');

var _graphqlRelay = require('graphql-relay');

var _TPLinkTimeChart = require('./TPLinkTimeChart.container');

var _TPLinkTimeChart2 = _interopRequireDefault(_TPLinkTimeChart);

var _query = require('./query.graphql');

var _query2 = _interopRequireDefault(_query);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// import TimeChartStateful from './TPLinkTimeChart.stateful';
/**
 * Created by atif on 12/18/16.
 */
exports.default = (0, _reactApollo.graphql)(_query2.default, {
  options: function options(args) {
    var startDateFormatted = args.startDate;
    var endDateFormatted = args.endDate;
    var queryString = {
      where: {
        type: 'command',
        assetId: (0, _graphqlRelay.fromGlobalId)(args.assetId).id,
        time: { between: [startDateFormatted, endDateFormatted] },
        'data.params.standby': { exists: true }
      },
      order: 'time DESC'
    };

    // `[{\"condition\":\"exists\",\"key\":\"data.params.standby\"},{\"condition\":\"descending\",\"key\":\"time\"},{\"condition\":\"lessThanOrEqualTo\",\"key\":\"time\",\"value\":{\"__type\":\"Date\",
    // \"iso\":\"${endDateFormatted}\"}}, {\"condition\":\"greaterThanOrEqualTo\",\"key\":\"time\",\"value\":{\"__type\":\"Date\",\"iso\":\"${startDateFormatted}\"}}]`;

    return { variables: { assetId: args.assetId, queryString: queryString } };
  }
})(_TPLinkTimeChart2.default);
