// bluerain-app-device-explorer/lib/pages/SitePage/SitePage.component.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactstrap = require('reactstrap');

var _reactRouter = require('react-router');

var _bluerainClientServices = require('bluerain-client-services');

var _AppContent = require('bluerain-ui/lib/Layout/AppContent');

var _AppContent2 = _interopRequireDefault(_AppContent);

var _PrimaryAppNav = require('bluerain-ui/lib/Nav/PrimaryAppNav');

var _PrimaryAppNav2 = _interopRequireDefault(_PrimaryAppNav);

var _WidgetsGroup = require('../../components/Asset/WidgetsGroup/WidgetsGroup.stateful');

var _WidgetsGroup2 = _interopRequireDefault(_WidgetsGroup);

var _WidgetsGroup3 = require('../../components/Asset/WidgetsGroup/WidgetsGroup.graphql');

var _WidgetsGroup4 = _interopRequireDefault(_WidgetsGroup3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  title: _react.PropTypes.string,
  siteId: _react.PropTypes.string,
  data: _react.PropTypes.object,
  hideLoading: _react.PropTypes.func
};

var defaultProps = {
  title: 'Site',
  siteId: '',
  hideLoading: function hideLoading() {}
};

var Page = function (_React$Component) {
  _inherits(Page, _React$Component);

  function Page(props) {
    _classCallCheck(this, Page);

    var _this = _possibleConstructorReturn(this, (Page.__proto__ || Object.getPrototypeOf(Page)).call(this, props));

    _this.toggle = _this.toggle.bind(_this);
    _this.state = {
      dropdownOpen: false
    };
    return _this;
  }

  _createClass(Page, [{
    key: 'toggle',
    value: function toggle() {
      this.setState({
        dropdownOpen: !this.state.dropdownOpen
      });
    }
  }, {
    key: 'componentDidMount',
    value: function componentDidMount() {
      var loading = this.props.data.loading;

      if (!loading) {
        this.props.hideLoading();
      }
    }
  }, {
    key: 'renderPrimaryNav',
    value: function renderPrimaryNav() {

      var title = this.props.title;
      var icon = void 0;
      var btnStyle = {
        margin: '0 7px '
      };
      var previous = {
        title: 'Sites',
        url: '/app/device-explorer'
      };

      try {
        title = this.props.data.node.name;
      } catch (e) {
        console.log('ERROR =>', e);
      }

      try {
        icon = this.props.data.node.image;
      } catch (e) {
        console.log('ERROR =>', e);
      }

      if (icon === undefined || icon === null) {
        try {
          icon = this.props.data.node.siteType.icon;
        } catch (e) {
          console.log('Error =>', e);
        }
      }

      var addDeviceUrl = '/app/device-explorer/sites/' + this.props.siteId + '/add-device?siteName=' + this.props.data.node.name;
      var siteEditUrl = '/app/device-explorer/sites/' + this.props.siteId + '/edit';

      return _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(
          _PrimaryAppNav2.default,
          { title: title, icon: icon, previous: previous },
          _react2.default.createElement(
            _reactstrap.ButtonDropdown,
            { isOpen: this.state.dropdownOpen, toggle: this.toggle },
            _react2.default.createElement(
              _reactstrap.Button,
              { id: 'caret', color: 'primary' },
              _react2.default.createElement(
                _reactRouter.Link,
                { to: addDeviceUrl, style: { "color": "white" } },
                _react2.default.createElement('i', { className: 'fa fa-plus hidden-xm-down', style: { "color": "white" } }),
                _react2.default.createElement(
                  'span',
                  {
                    className: 'hidden-xs-down', style: btnStyle
                  },
                  'Add Device'
                )
              )
            ),
            _react2.default.createElement(_reactstrap.DropdownToggle, { caret: true, color: 'primary' }),
            _react2.default.createElement(
              _reactstrap.DropdownMenu,
              null,
              _react2.default.createElement(
                _reactstrap.DropdownItem,
                null,
                _react2.default.createElement(
                  _reactRouter.Link,
                  { to: siteEditUrl },
                  _react2.default.createElement('i', { className: 'fa fa-gear hidden-xm-down' }),
                  _react2.default.createElement(
                    'span',
                    { className: 'hidden-xs-down', style: btnStyle },
                    'Settings'
                  )
                ),
                ' \xA0'
              )
            )
          )
        )
      );
    }
  }, {
    key: 'renderContent',
    value: function renderContent() {
      var siteId = this.props.siteId;

      var WidgetGroup = (0, _WidgetsGroup4.default)(_WidgetsGroup2.default, siteId);

      return _react2.default.createElement(
        _AppContent2.default,
        null,
        _react2.default.createElement(
          _reactstrap.Container,
          null,
          _react2.default.createElement(
            _reactstrap.Row,
            null,
            _react2.default.createElement(
              _reactstrap.Col,
              { lg: '12', className: 'pt-2' },
              _react2.default.createElement(WidgetGroup, { siteId: siteId })
            )
          )
        )
      );
    }
  }, {
    key: 'render',
    value: function render() {
      return _react2.default.createElement(
        'div',
        null,
        this.renderPrimaryNav(),
        Page.renderSecondaryNav(),
        this.renderContent()
      );
    }
  }], [{
    key: 'renderSecondaryNav',
    value: function renderSecondaryNav() {}
  }]);

  return Page;
}(_react2.default.Component);

Page.propTypes = propTypes;
Page.defaultProps = defaultProps;

exports.default = (0, _bluerainClientServices.withLoadingBar)(Page);
