// bluerain-app-device-explorer/src/components/Asset/Charts/TPLinkTimeChart/index.js
/**
 * Created by atif on 12/18/16.
 */
import { graphql } from 'react-apollo';
import { fromGlobalId } from 'graphql-relay';
// import TimeChartStateful from './TPLinkTimeChart.stateful';
import TimeChartContainer from './TPLinkTimeChart.container';
import ViewerQuery from './query.graphql';

export default graphql(ViewerQuery, {
  options: (args) => {
    const startDateFormatted = args.startDate;
    const endDateFormatted = args.endDate;
    const queryString = {
      where: {
        type: 'command',
        assetId: fromGlobalId(args.assetId).id,
        time: { between: [startDateFormatted, endDateFormatted] },
        'data.params.standby': { exists: true }
      },
      order: 'time DESC'
    };

    // `[{\"condition\":\"exists\",\"key\":\"data.params.standby\"},{\"condition\":\"descending\",\"key\":\"time\"},{\"condition\":\"lessThanOrEqualTo\",\"key\":\"time\",\"value\":{\"__type\":\"Date\",
    // \"iso\":\"${endDateFormatted}\"}}, {\"condition\":\"greaterThanOrEqualTo\",\"key\":\"time\",\"value\":{\"__type\":\"Date\",\"iso\":\"${startDateFormatted}\"}}]`;

    return { variables: { assetId: args.assetId, queryString } };
  }
})(TimeChartContainer);
