// bluerain-app-device-explorer/lib/components/Device/AddDevice/SecondScreen/AddDeviceSecondScreen.widget.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _widget = require('bluerain-ui/lib/Widget/widget');

var _widget2 = _interopRequireDefault(_widget);

var _widgetHeader = require('bluerain-ui/lib/Widget/widgetHeader');

var _widgetHeader2 = _interopRequireDefault(_widgetHeader);

var _widgetTitle = require('bluerain-ui/lib/Widget/widgetTitle');

var _widgetTitle2 = _interopRequireDefault(_widgetTitle);

var _widgetContent = require('bluerain-ui/lib/Widget/widgetContent');

var _widgetContent2 = _interopRequireDefault(_widgetContent);

var _AddDeviceSecondScreen = require('./AddDeviceSecondScreen.component');

var _AddDeviceSecondScreen2 = _interopRequireDefault(_AddDeviceSecondScreen);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  networks: _react.PropTypes.array,
  clickHandler: _react.PropTypes.func
};

var SiteFormWidget = function (_React$Component) {
  _inherits(SiteFormWidget, _React$Component);

  function SiteFormWidget() {
    _classCallCheck(this, SiteFormWidget);

    var _this = _possibleConstructorReturn(this, (SiteFormWidget.__proto__ || Object.getPrototypeOf(SiteFormWidget)).call(this));

    _this.networkList = [];
    _this.clickHandler = function (name) {
      // console.log(`${name} clicked`);
    };
    return _this;
  }

  _createClass(SiteFormWidget, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      this.networkList = this.props.networks || this.networkList;
      this.clickHandler = this.props.clickHandler || this.clickHandler;

      this.networkList = this.props.networks;

      var i = 0;
      return _react2.default.createElement(
        _widget2.default,
        null,
        _react2.default.createElement(
          _widgetHeader2.default,
          null,
          _react2.default.createElement(
            _widgetTitle2.default,
            null,
            'Now connect this AC with one of the following internet connection'
          )
        ),
        _react2.default.createElement(
          _widgetContent2.default,
          null,
          this.networkList.map(function (item) {
            return _react2.default.createElement(_AddDeviceSecondScreen2.default, {
              key: i++, wifiName: item.ssid, secure: item.secure,
              onClick: _this2.clickHandler
            });
          })
        )
      );
    }
  }]);

  return SiteFormWidget;
}(_react2.default.Component);

SiteFormWidget.propTypes = propTypes;

exports.default = SiteFormWidget;
