// bluerain-ui/lib/ComponentState/ComponentState.component.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Card = require('reactstrap/lib/Card');

var _Card2 = _interopRequireDefault(_Card);

var _CardImg = require('reactstrap/lib/CardImg');

var _CardImg2 = _interopRequireDefault(_CardImg);

require('bluerain-bootstrap-theme/dist/css/components/component-state.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  image: _react.PropTypes.string,
  imageSize: _react.PropTypes.string,
  card: _react.PropTypes.bool,
  title: _react.PropTypes.string,
  description: _react.PropTypes.string,
  linkText: _react.PropTypes.string,
  linkTag: _react.PropTypes.oneOfType([_react.PropTypes.func, _react.PropTypes.string]),
  linkOnClick: _react.PropTypes.func
};

var defaultProps = {
  imageSize: '100px',
  card: true,
  linkTag: 'a'
};

var ComponentState = function (_React$Component) {
  _inherits(ComponentState, _React$Component);

  function ComponentState() {
    _classCallCheck(this, ComponentState);

    return _possibleConstructorReturn(this, (ComponentState.__proto__ || Object.getPrototypeOf(ComponentState)).apply(this, arguments));
  }

  _createClass(ComponentState, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          image = _props.image,
          imageSize = _props.imageSize,
          card = _props.card,
          title = _props.title,
          description = _props.description,
          linkText = _props.linkText,
          linkTag = _props.linkTag,
          linkOnClick = _props.linkOnClick,
          props = _objectWithoutProperties(_props, ['image', 'imageSize', 'card', 'title', 'description', 'linkText', 'linkTag', 'linkOnClick']);

      var imageWidth = '';

      if (imageSize === 'sm') {
        imageWidth = '50px';
      } else if (imageSize === 'lg') {
        imageWidth = '150px';
      } else if (imageSize === 'lgg') {
        imageWidth = '310px';
      } else {
        imageWidth = '100px';
      }

      var MainTag = 'div';

      if (card === true) {
        MainTag = _Card2.default;
      }

      return _react2.default.createElement(
        MainTag,
        _extends({ className: 'component-state' }, props),
        ComponentState.renderImage(image, imageWidth),
        ComponentState.renderTitle(title),
        ComponentState.renderDescription(description),
        ComponentState.renderLink(linkTag, linkText, linkOnClick)
      );
    }
  }], [{
    key: 'renderImage',
    value: function renderImage(url, width) {
      var imagSRC = 'https://s3-us-west-2.amazonaws.com/bluerainimages/' + url;
      if (url) {
        return _react2.default.createElement(_CardImg2.default, { width: width, src: imagSRC });
      }
    }
  }, {
    key: 'renderTitle',
    value: function renderTitle(title) {
      if (title) {
        return _react2.default.createElement(
          'h4',
          { className: 'card-title' },
          title
        );
      }
    }
  }, {
    key: 'renderDescription',
    value: function renderDescription(description) {
      if (description) {
        return _react2.default.createElement(
          'p',
          { className: 'card-text text-muted' },
          description
        );
      }
    }
  }, {
    key: 'renderLink',
    value: function renderLink(Tag, text, onClick) {
      if (text) {
        return _react2.default.createElement(
          Tag,
          { onClick: onClick },
          text
        );
      }
    }
  }]);

  return ComponentState;
}(_react2.default.Component);

exports.default = ComponentState;


ComponentState.propTypes = propTypes;
ComponentState.defaultProps = defaultProps;
