// bluerain-app-device-explorer/src/components/Asset/Charts/RefrigeratorAmbientTemperatureChart/RefAmbientTemperatureChart.container.js
/**
 * Created by atif on 12/18/16.
 */
import React, { PropTypes } from 'react';
import TemperatureChartComponent from './RefAmbientTemperatureChart.component';

const propTypes = {
  data: PropTypes.object,
  tooltip: PropTypes.bool,
  startDate: PropTypes.string,
  endDate: PropTypes.string,
  label: PropTypes.string,
  showGraph: PropTypes.bool,
  linkTag: PropTypes.oneOfType([
    React.PropTypes.func,
    React.PropTypes.string
  ]),
  to: PropTypes.string,
  color: PropTypes.string
};

const defaultPropTypes = {
  startDate: '01/01/1970',
  endDate: '01/01/2099',
  tooltip: true,
  showGraph: true,
  label: 'Ambient Temperature',
  linkTag: 'a',
  to: '#'
};

export default class TemperatureChartContainer extends React.Component {

  render() {
    const {
      color,
      tooltip,
      data: dbData,
      label,
      linkTag,
      to,
      showGraph,
      ...rest
    } = this.props;

    const data = [];
    let lastReading = 'No data found';

    if (this.props.data === undefined || this.props.data.loading) {
      lastReading = 'Loading data';
    }

    try {

      let dataFromQuery = dbData.Event.EventFind.edges;
      dataFromQuery = dataFromQuery.map(item => item.node);

      dataFromQuery.forEach((item) => {
        if (item.data && item.data.temperature.t4 !== undefined) {
          data.push({
            time: new Date(item.time),
            temperature: ((item.data.temperature.t4 / 1024) - 0.5) * 100
          });
        }
      });

      if (data.length > 0) {
        data.sort((a, b) => (a.time > b.time) ? 1 : ((b.time > a.time) ? -1 : 0));
        const sum = data.reduce((a, b) => a + b.temperature, 0);
        const average = sum / data.length;
        lastReading = `${average.toFixed(0)} ℃`;
      }
    } catch (error) {
      console.info('Temperature Chart no data found');
    }

    if (showGraph) {
      return (
        <TemperatureChartComponent
          data={data}
          x="time"
          y={datum => (datum.temperature)}
          color={this.props.color}
          label={label}
          value={lastReading}
          tooltip={this.props.tooltip}
          linkTag={linkTag}
          to={to}
          {...rest}
        />

      );
    }
    return (
      <TemperatureChartComponent
        color={color}
        label={label}
        value={lastReading}
        tooltip={tooltip}
        linkTag={linkTag}
        to={to}
        {...rest}
      />
    );
  }
}

TemperatureChartContainer.propTypes = propTypes;
TemperatureChartContainer.defaultProps = defaultPropTypes;
