// bluerain-app-device-explorer/dev/index.js
import React from 'react';
import ReactDOM from 'react-dom';
import { ApolloProvider } from 'react-apollo';
import { browserHistory, Router } from 'react-router';
import { createStore, combineReducers, applyMiddleware, compose } from 'redux';
import { reducers, loadingBarReducer } from 'bluerain-client-services';
import thunk from 'redux-thunk';

import 'bluerain-bootstrap-theme/dist/css/bluerain-bootstrap-theme.css';

import Layout from './Layout';

import app from '../src/';
import client from './client';
import rootRoute from '../src/routes';

// const injectTapEventPlugin = require('react-tap-event-plugin');
// injectTapEventPlugin();

const rootNode = document.createElement('div'); // eslint-disable-line no-undef
document.body.appendChild(rootNode); // eslint-disable-line no-undef

const routes = {
  path: '/',
  indexRoute: {
    onEnter: (nextState, replace) => replace({ pathname: `/app/${app.slug}` })
  },
  childRoutes: [{
    path: 'app',
    component: Layout,
    indexRoute: {
      onEnter: (nextState, replace) => replace({ pathname: `/app/${app.slug}` })
    },
    childRoutes: [rootRoute]
  }]
};


export const store = createStore(
  combineReducers({
    bluerain: reducers,
    loadingBar: loadingBarReducer,
    apollo: client.reducer(),
  }),
  {}, // initial state
  compose(
      applyMiddleware(thunk),
      applyMiddleware(client.middleware()),
      // If you are using the devToolsExtension, you can add it here also
      window.devToolsExtension ? window.devToolsExtension() : f => f // eslint-disable-line no-undef
  )
);

ReactDOM.render(
  <ApolloProvider store={store} client={client}>
    <Router
      history={browserHistory}
      routes={routes}
    />
  </ApolloProvider>,
  rootNode
);
