// bluerain-app-device-explorer/src/components/Asset/Charts/EnergyChart/EnergyChart.container.js
/**
 * Created by atif on 12/18/16.
 */
import React, { PropTypes } from 'react';
import EnergyChartComponent from './EnergyChart.component';

const propTypes = {
  data: PropTypes.object,
  tooltip: PropTypes.bool,
  startDate: PropTypes.string,
  endDate: PropTypes.string,
  color: PropTypes.string,
  label: PropTypes.string,
  showGraph: PropTypes.bool,
  linkTag: PropTypes.oneOfType([
    React.PropTypes.func,
    React.PropTypes.string
  ]),
  to: PropTypes.string
};

const defaultPropTypes = {
  startDate: '01/01/1970',
  endDate: '01/01/2099',
  tooltip: true,
  showGraph: true,
  label: 'Energy Units', linkTag: 'a',
  to: '#'
};

export default class EnergyChartContainer extends React.Component {

  render() {

    const {
      showGraph,
      color,
      tooltip,
      data: dbData,
      label,
      linkTag,
      to,
      ...rest
    } = this.props;

    const data = [];

    let lastReading = 'No data found';

    if (this.props.data === undefined || this.props.data.loading) {
      lastReading = 'Loading data';
    }

    try {
      let dataFromQuery = dbData.Event.EventFind.edges;
      dataFromQuery = dataFromQuery.map(item => item.node);
      if (dataFromQuery && dataFromQuery.length > 0) {
        const baseEnergy = dataFromQuery[dataFromQuery.length - 1].data.energy;
        dataFromQuery.forEach((item) => {
          if (item.data && item.data.energy !== undefined) {
            data.push({
              time: new Date(item.time),
              energy: item.data.energy - baseEnergy
            });
          }
        });
      }

      if (data.length > 0) {
        data.sort((a, b) => (a.time > b.time ) ? 1 : ((b.time > a.time) ? -1 : 0));

        lastReading = (data[data.length - 1].energy - data[0].energy).toFixed(2);
        lastReading = `${lastReading} KWh`;
      }
    } catch (error) {
      // console.info('Energy Chart no data found');
    }

    if (showGraph) {
      return (
        <EnergyChartComponent
          data={data}
          x="time"
          y={datum => (datum.energy)}
          color={color}
          label={label}
          value={lastReading}
          tooltip={tooltip}
          linkTag={linkTag}
          to={to}
          {...rest}
        />
      );
    }
    return (
      <EnergyChartComponent
        color={color}
        label={label}
        value={lastReading}
        tooltip={tooltip}
        linkTag={linkTag}
        to={to}
        {...rest}
      />
    );
  }
}

EnergyChartContainer.propTypes = propTypes;
EnergyChartContainer.defaultProps = defaultPropTypes;
