// bluerain-app-device-explorer/src/components/Report/PowerReport/PowerReport.component.js
/**
 * Created by atif on 12/18/16.
 */
import React, { PropTypes } from 'react';
import AreaChart from 'bluerain-ui/lib/Card/ChartCard/AreaChartAxis.component';
// import moment from 'moment';
// import { Table } from 'reactstrap';

const propTypes = {
  data: PropTypes.array,
  x: PropTypes.string,
  y: PropTypes.func,
  color: PropTypes.string,
  tooltip: PropTypes.bool
};

const defaultProps = {
  data: []
};

export default class PowerReportComponent extends React.Component {

  render() {

    const records = this.props.data;
    const fillColor = 'rgba(255,0,0,0.3)';
    return (
      <div>
        <div style={{ width: '100%', margin: '0 auto' }}>
          <AreaChart
            data={records}
            x={this.props.x}
            y={this.props.y}
            tooltip={this.props.tooltip}
            height={300}
            fillColor={fillColor}
            strokeColor={this.props.color}
            dateFormat="DD, MMM YYYY, hh:mm:ss a"
            interpolation="basis"
          />
        </div>
        {/*<div className="panel-data border-top">*/}
          {/*<div className="table-responsive">*/}
            {/*<Table>*/}
              {/*<thead className="thead-default">*/}
                {/*<tr>*/}
                  {/*<th>Date/Time</th>*/}
                  {/*<th>Value (W)</th>*/}
                {/*</tr>*/}
              {/*</thead>*/}
              {/*<tbody>*/}
                {/*{*/}
                {/*this.props.data.map((item, index) => (*/}
                  {/*<tr key={index}>*/}
                    {/*<td>{moment(item.time).format('DD, MMM YYYY, hh:mm:ss a')}</td>*/}
                    {/*<td>{item.power.toFixed(2)}</td>*/}
                  {/*</tr>*/}
                {/*))*/}
              {/*}*/}
              {/*</tbody>*/}
            {/*</Table>*/}
          {/*</div>*/}
        {/*</div>*/}
      </div>
    );
  }
}

PowerReportComponent.propTypes = propTypes;
PowerReportComponent.defaultProps = defaultProps;
