// bluerain-app-device-explorer/lib/pages/Assets/Single/Overview/OverviewPage.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactstrap = require('reactstrap');

var _reactApollo = require('react-apollo');

var _query = require('./query.graphql');

var _query2 = _interopRequireDefault(_query);

var _LightControl = require('../../../../components/Asset/AssetControl/LightControl');

var _LightControl2 = _interopRequireDefault(_LightControl);

var _SocketControl = require('../../../../components/Asset/AssetControl/SocketControl');

var _SocketControl2 = _interopRequireDefault(_SocketControl);

var _TPLinkSensorInfo = require('../../../../components/Asset/AssetSensorInfo/TPLinkSensorInfo');

var _TPLinkSensorInfo2 = _interopRequireDefault(_TPLinkSensorInfo);

var _Timeline = require('../../../../components/Post/Timeline');

var _Timeline2 = _interopRequireDefault(_Timeline);

var _ChartSlider = require('../../../../components/Asset/Charts/ChartSlider/ChartSlider.widget');

var _ChartSlider2 = _interopRequireDefault(_ChartSlider);

var _RefrigeratorChartSlider = require('../../../../components/Asset/Charts/RefrigeratorChartSlider/RefrigeratorChartSlider.widget');

var _RefrigeratorChartSlider2 = _interopRequireDefault(_RefrigeratorChartSlider);

var _ChartSlider3 = require('../../../../components/Asset/Charts/TPLinkChartSlider/ChartSlider.widget');

var _ChartSlider4 = _interopRequireDefault(_ChartSlider3);

var _ChartSlider5 = require('../../../../components/Asset/Charts/LifxChartSlider/ChartSlider.widget');

var _ChartSlider6 = _interopRequireDefault(_ChartSlider5);

var _ComponentRegistry = require('../../../../components/ComponentRegistry');

var _ComponentRegistry2 = _interopRequireDefault(_ComponentRegistry);

var _LightControlSensorInfo = require('../../../../components/Asset/AssetSensorInfo/LightControlSensorInfo');

var _LightControlSensorInfo2 = _interopRequireDefault(_LightControlSensorInfo);

var _FitbitSensorInfo = require('../../../../components/Asset/FitbitSensorInfo');

var _FitbitSensorInfo2 = _interopRequireDefault(_FitbitSensorInfo);

var _reactWaypoint = require('react-waypoint');

var _reactWaypoint2 = _interopRequireDefault(_reactWaypoint);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var AssetControl = _ComponentRegistry2.default.getComponent('AssetControl', 'AC').content;
var SensorInfo = _ComponentRegistry2.default.getComponent('SensorInfo', 'AC').content;
var RefrigeratorSensorInfo = _ComponentRegistry2.default.getComponent('RefrigeratorSensorInfo', 'Fridge').content;
var RefrigeratorLocationMapView = _ComponentRegistry2.default.getComponent('RefrigeratorLocationMapView', 'Fridge').content;
var SpeakersControl = _ComponentRegistry2.default.getComponent('AssetControl', 'Speakers').content;


var propTypes = {
  assetId: _react.PropTypes.string,
  data: _react.PropTypes.object
};

var defaultProps = {};

var OverviewPage = function (_React$Component) {
  _inherits(OverviewPage, _React$Component);

  function OverviewPage(props) {
    _classCallCheck(this, OverviewPage);

    var _this = _possibleConstructorReturn(this, (OverviewPage.__proto__ || Object.getPrototypeOf(OverviewPage)).call(this, props));

    _initialiseProps.call(_this);

    _this.state = {
      fix: false
    };
    _this._handleLeave = _this._handleLeave.bind(_this);
    _this._handleEnter = _this._handleEnter.bind(_this);
    _this.flag = false;
    return _this;
  }

  _createClass(OverviewPage, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      var fix = this.state.fix;

      try {
        var assetId = this.props.assetId;


        if (!this.props.data.node || !this.props.data.node.assetType) {
          return _react2.default.createElement('div', null);
        }

        var name = this.props.data.node.assetType.name;

        if (name === 'Speakers') {
          return _react2.default.createElement(
            'div',
            null,
            _react2.default.createElement(
              'section',
              null,
              _react2.default.createElement(
                _reactstrap.Container,
                null,
                _react2.default.createElement(
                  _reactstrap.Row,
                  null,
                  _react2.default.createElement(
                    _reactstrap.Col,
                    { lg: '4', md: '5' },
                    _react2.default.createElement(SpeakersControl, { assetId: assetId })
                  ),
                  _react2.default.createElement(
                    _reactstrap.Col,
                    { lg: '8', md: '7', className: 'col-faded' },
                    _react2.default.createElement(_Timeline2.default, { assetId: assetId })
                  )
                )
              )
            )
          );
        } else if (name === 'Light') {
          return _react2.default.createElement(
            'div',
            null,
            _react2.default.createElement(
              'section',
              { className: 'well' },
              _react2.default.createElement(_ChartSlider6.default, { assetId: assetId })
            ),
            _react2.default.createElement(
              'section',
              null,
              _react2.default.createElement(
                _reactstrap.Container,
                null,
                _react2.default.createElement(
                  _reactstrap.Row,
                  null,
                  _react2.default.createElement(
                    _reactstrap.Col,
                    { lg: '4', md: '5' },
                    _react2.default.createElement(_LightControl2.default, { assetId: assetId }),
                    _react2.default.createElement(_LightControlSensorInfo2.default, { assetId: assetId })
                  ),
                  _react2.default.createElement(
                    _reactstrap.Col,
                    { lg: '8', md: '7', className: 'col-faded' },
                    _react2.default.createElement(_Timeline2.default, { assetId: assetId })
                  )
                )
              )
            )
          );
        } else if (name === 'Socket') {
          return _react2.default.createElement(
            'div',
            null,
            _react2.default.createElement(
              'section',
              { className: 'well' },
              _react2.default.createElement(_ChartSlider4.default, { assetId: assetId })
            ),
            _react2.default.createElement(
              'section',
              null,
              _react2.default.createElement(
                _reactstrap.Container,
                null,
                _react2.default.createElement(
                  _reactstrap.Row,
                  null,
                  _react2.default.createElement(
                    _reactstrap.Col,
                    { lg: '4', md: '5' },
                    _react2.default.createElement(_SocketControl2.default, { assetId: assetId }),
                    _react2.default.createElement(_TPLinkSensorInfo2.default, { assetId: assetId })
                  ),
                  _react2.default.createElement(
                    _reactstrap.Col,
                    { lg: '8', md: '7', className: 'col-faded' },
                    _react2.default.createElement(_Timeline2.default, { assetId: assetId })
                  )
                )
              )
            )
          );
        } else if (name === 'Air Conditioner') {
          return _react2.default.createElement(
            'div',
            null,
            _react2.default.createElement(
              'section',
              { className: 'well' },
              _react2.default.createElement(_ChartSlider2.default, { assetId: assetId })
            ),
            _react2.default.createElement(
              'section',
              null,
              _react2.default.createElement(
                _reactstrap.Container,
                null,
                _react2.default.createElement(
                  _reactstrap.Row,
                  null,
                  _react2.default.createElement(
                    _reactstrap.Col,
                    { style: fix ? { position: 'fixed', top: '0px', width: '313px' } : null, lg: '4', md: '5', id: 'abc' },
                    _react2.default.createElement(AssetControl, { assetId: assetId }),
                    _react2.default.createElement(SensorInfo, { assetId: assetId })
                  ),
                  _react2.default.createElement(
                    _reactstrap.Col,
                    { style: fix ? { marginLeft: '314px' } : null, lg: '8', md: '7', className: 'col-faded' },
                    _react2.default.createElement(_reactWaypoint2.default, { onEnter: function onEnter(props) {
                        return _this2._handleEnter(props);
                      },
                      onLeave: function onLeave(props) {
                        return _this2._handleLeave(props);
                      } }),
                    _react2.default.createElement(_Timeline2.default, { assetId: assetId })
                  )
                ),
                _react2.default.createElement('br', null)
              )
            )
          );
        } else if (name === 'Person') {
          return _react2.default.createElement(
            'div',
            null,
            _react2.default.createElement('section', { className: 'well' }),
            _react2.default.createElement(
              'section',
              null,
              _react2.default.createElement(
                _reactstrap.Container,
                null,
                _react2.default.createElement(
                  _reactstrap.Row,
                  null,
                  _react2.default.createElement(
                    _reactstrap.Col,
                    { lg: '4', md: '5' },
                    _react2.default.createElement(_FitbitSensorInfo2.default, { assetId: assetId })
                  ),
                  _react2.default.createElement(_reactstrap.Col, { lg: '8', md: '7', className: 'col-faded' })
                )
              )
            )
          );
        } else if (name === 'Refrigerator') {
          return _react2.default.createElement(
            'div',
            null,
            _react2.default.createElement(
              'section',
              { className: 'well' },
              _react2.default.createElement(_RefrigeratorChartSlider2.default, { assetId: assetId })
            ),
            _react2.default.createElement(
              'section',
              null,
              _react2.default.createElement(
                _reactstrap.Container,
                null,
                _react2.default.createElement(
                  _reactstrap.Row,
                  null,
                  _react2.default.createElement(
                    _reactstrap.Col,
                    { lg: '4', md: '5' },
                    _react2.default.createElement(RefrigeratorLocationMapView, { assetId: assetId }),
                    _react2.default.createElement(RefrigeratorSensorInfo, { assetId: assetId })
                  ),
                  _react2.default.createElement(
                    _reactstrap.Col,
                    { lg: '8', md: '7', className: 'col-faded' },
                    _react2.default.createElement(_Timeline2.default, { assetId: assetId })
                  )
                )
              )
            )
          );
        }

        return _react2.default.createElement(
          'h1',
          null,
          ' Unknown Device Found '
        );
      } catch (error) {
        console.log(error);
      }
    }
  }]);

  return OverviewPage;
}(_react2.default.Component);

var _initialiseProps = function _initialiseProps() {
  var _this3 = this;

  this._handleEnter = function (props) {
    var windowWidth = document.body.clientWidth;
    if (props.waypointTop >= 0 && _this3.flag && windowWidth > 1000) {
      _this3.setState({ fix: false });
    }
  };

  this._handleLeave = function (props) {
    var windowWidth = document.body.clientWidth;
    if (props.waypointTop <= 0 && windowWidth > 1000) {
      _this3.setState({ fix: true });
      _this3.flag = true;
    }
  };
};

OverviewPage.propTypes = propTypes;
OverviewPage.defaultProps = defaultProps;

// export default OverviewPage;

exports.default = (0, _reactApollo.graphql)(_query2.default, {
  options: function options(_ref) {
    var assetId = _ref.assetId;
    return { variables: { assetId: assetId } };
  }
})(OverviewPage);
