// bluerain-app-device-explorer/src/pages/Assets/AddAsset/Recheck.js
import React, { PropTypes } from 'react';
import 'bluerain-bootstrap-theme/dist/css/components/data-list.css';
import ProgressLabel from 'react-progress-label';


const propTypes = {
  jumpToStep: PropTypes.func

};


export default class Recheck extends React.Component {

  constructor(props) {
    super();

    this.state = {
      text: 'Establishing Connection',
      progress: 1
    };
  }


  componentDidMount() {
    this.interval = setInterval(() => {
      let progress = this.state.progress;

      if (progress === 100) {
        clearInterval(this.interval);
        this.props.jumpToStep(4);
        this.setState({ dymmyvalue: true });
      }

      progress++;
      this.setState({ progress });

    }, 200);
  }

  componentWillUnmount() {

    if (this.interval) {
      clearInterval(this.interval);
    }

  }

  isValidated() {
    if (this.state.dymmyvalue === true) {

      return true;
    }
    return false;
  }


  renderToolbar() {
    return (
      <Ons.Toolbar>
        <div className="center">this.state.Progress</div>
      </Ons.Toolbar>
    );
  }

  render() {
    const style1 = {
      padding: 10,
      margin: 10,
      backgroundColor: '#ececec',
      color: '#333',
      fontfamily: 'monospace',
      fontSize: 32,
      textAlign: 'center',
      textColor: '#23a3ba'
    };
    return (

      <div style={style1}>

        <p >
          {this.state.text}
        </p>

        <ProgressLabel
          progress={this.state.progress}
          startDegree={0}
          progressWidth={50}
          trackWidth={50}
          cornersWidth={12}
          size={200}
          fillColor="#f3f3f3"
          trackColor="#23a3ba"
          progressColor="#bff3fc"
        >
          <text>{this.state.progress}</text>
        </ProgressLabel>

      </div>
    );
  }
}

Recheck.propTypes = propTypes;
