// bluerain-app-device-explorer/src/pages/Assets/AddAsset/Step4.js
import React from 'react';
import 'bluerain-bootstrap-theme/dist/css/components/data-list.css';
import LocalSdk from 'localsdk';
import AddDeviceFourthScreenWidget from '../../../components/Device/AddDevice/FourthScreen/AddDeviceFourthScreen.widget';
import Loading from './Loading';

const localSDK = new LocalSdk();

// in this step we will show if device is connected
// to internet or show the device could not be connected
export default class Step4 extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      stateFlag: false,
      reason: ''

    };


    const data = { data: 'connected' };

    localSDK.getStatus(data, (response) => {
      if (response.connected === 1) {
        this.setState({
          stateFlag: true,
          connectedTo: `${response.SSID}`
        });

      } else {

        if (response.reason === undefined) {
          this.setState({ reason: 'Device could not be connected ' });
        } else {

          this.setState({ reason: ` ${response.reason}!` });
        }
        this.setState({ stateFlag: false });
      }

    });


  }

  isValidated() {
    if (window.location.search && window.location.search.search('state') && window.location.search.search('state') !== -1) {
      history.back();
    }
    if (this.state.stateFlag !== false) {
      return true;
    }
    return false;
  }


  render() {


    return (
      <div>
        {this.state.stateFlag ? (
          <AddDeviceFourthScreenWidget />
          ) : (

            <Loading reason={this.state.reason} />
          )}
      </div>

    );
  }

}
