// bluerain-app-device-explorer/src/pages/Assets/AddAsset/Step1.js
import React, { PropTypes } from 'react';
import 'bluerain-bootstrap-theme/dist/css/components/data-list.css';
import AddDeviceFirstScreenWidget from '../../../components/Device/AddDevice/FirstScreen/AddDeviceFirstScreen.widget';
import Loading from './Loading';


const propTypes = {
  getStore: PropTypes.func,
  updateStoreData: PropTypes.func,
  updateStoreMacId: PropTypes.func,
  jumpToStep: PropTypes.func

};

/* in this step we will show user the basic first screen
 after he clicks the next button we will show the loading component*/


export default class Step1 extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      stateFlag: true
    };


  }


  onsubmit() {

  }

  handleState(value) {
    this.setState({ stateFlag: value });
  }


  isValidated() {

    // if(this.state.stateFlag===false ){
    this.props.getStore().localSDK.getList((data) => {

      if (data.error) {

        this.setState({
          stateFlag: false
        });


      } else {
        this.setState({
          stateFlag: true

        });

        this.props.updateStoreData(data);
        this.props.updateStoreMacId(data.macid);
        this.props.jumpToStep(1);

      }
    });
    // }


    // event.preventDefault();
    //  this.setState({ step: 'loading' });
    if (this.props.getStore().data.networks) {

      return true;


    }
    this.setState({
      stateFlag: false
    });
    return false;

  }


  render() {

    return (
      <div className="step Step1">
        {this.state.stateFlag ? (
            <AddDeviceFirstScreenWidget onSubmit={this.onsubmit} />
          ) : (

            <Loading handleState={this.handleState.bind(this)} statevalue={1} />
          )}
      </div>


    );
  }
}

Step1.propTypes = propTypes;
