// bluerain-app-device-explorer/src/components/Asset/Charts/TPLinkChartSlider/ChartSlider.widget.js
/**
 * Created by atif on 12/18/16.
 */
import React, { PropTypes } from 'react';
import Widget from 'bluerain-ui/lib/Widget/widget';
import WidgetTitle from 'bluerain-ui/lib/Widget/widgetTitle';
import WidgetHeader from 'bluerain-ui/lib/Widget/widgetHeader';
import WidgetContent from 'bluerain-ui/lib/Widget/widgetContent';
import { Container } from 'reactstrap';
import { Link } from 'react-router';
import Slider from 'react-slick';
import OverlaySlider from 'bluerain-ui/lib/Slider/OverlaySlider';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';
import moment from 'moment';
import '../../../../css/custom_datepicker.css';
import TPLinkTimeChart from '../TPLinkTimeChart';

const propTypes = {
  assetId: PropTypes.string
};

const defaultProps = {
  assetId: ''
};

export default class ChartSliderWidget extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      focused: false,
      date: moment(),
    };

    this.onDateChange = this.onDateChange.bind(this);
    this.onFocusChange = this.onFocusChange.bind(this);
  }

  onDateChange(date) {
    this.setState({ date });
  }

  onFocusChange({ focused }) {
    this.setState({ focused });
  }

  render() {
    const { focused, date } = this.state;

    const assetId = this.props.assetId;
    const startDateFormatted = moment(date).startOf('day').toISOString();
    const endDateFormatted = moment(date).endOf('day').toISOString();

    const settings = {
      dots: true,
      infinite: false,
      adaptiveHeight: false,
      speed: 500,
      slidesToShow: 3,
      arrows: false,
      slidesToScroll: 1,
      responsive: [
        {
          breakpoint: 1200,
          settings: {
            slidesToShow: 3,
            slidesToScroll: 3,
            infinite: true
          }
        },
        {
          breakpoint: 992,
          settings: {
            slidesToShow: 3,
            slidesToScroll: 3
          }
        },
        {
          breakpoint: 768,
          settings: {
            slidesToShow: 2,
            slidesToScroll: 2
          }
        },
        {
          breakpoint: 576,
          settings: {
            slidesToShow: 1,
            slidesToScroll: 1
          }
        }
      ]
    };

    const reportBaseUrl = `/app/device-explorer/assets/${assetId}`;

    return (
      <Widget>
        <Container>
          <WidgetHeader>
            <WidgetTitle>Reports</WidgetTitle>
            <SingleDatePicker
              {...this.props}
              id="date_input"
              date={date}
              numberOfMonths={1}
              focused={focused}
              onDateChange={this.onDateChange}
              onFocusChange={this.onFocusChange}
              isOutsideRange={() => false}
            />
          </WidgetHeader>
        </Container>
        <WidgetContent>
          <OverlaySlider>
            <Slider {...settings}>
              <div>
                <TPLinkTimeChart
                  linkTag={Link}
                  to={`${reportBaseUrl}/reports/time`}
                  assetId={assetId}
                  startDate={startDateFormatted}
                  endDate={endDateFormatted}
                  queryString=""
                />
              </div>
            </Slider>
          </OverlaySlider>
        </WidgetContent>
      </Widget>

    );
  }
}

ChartSliderWidget.propTypes = propTypes;
ChartSliderWidget.defaultProps = defaultProps;
