// bluerain-app-device-explorer/src/components/Report/EnergySavingReport/EnergySavingReport.container.js
import React, { PropTypes } from 'react';
import EnergySavingReportComponent from './EnergySavingReport.component';

const propTypes = {
  data: PropTypes.object,
  tooltip: PropTypes.bool,
  startDate: PropTypes.string,
  endDate: PropTypes.string,
  color: PropTypes.string
};

const defaultPropTypes = {
  startDate: '01/01/1970',
  endDate: '01/01/2099',
  tooltip: true,
  color: 'yellow'
};
export default class EnergySavingReportContainer extends React.Component {

  render() {

    const {
      color,
      tooltip,
      data: dbData,
      ...rest
    } = this.props;


    const data = [];

    try {
      let dataFromQuery = dbData.Event.EventFind.edges;
      const maxMetaValue = dbData.node.meta.max;
      dataFromQuery = dataFromQuery.map(item => item.node);
      dataFromQuery.forEach((item) => {
        if (item.data && item.data.current !== undefined) {
          const currentLatest = item.data.current;
          let effPercent = ((maxMetaValue - currentLatest) / maxMetaValue) * 100;
          if (effPercent < 0) {
            effPercent = 0;
          } else if (effPercent > 100) {
            effPercent = 100;
          }
          data.push({
            time: new Date(item.time),
            current: effPercent
          });
        }
      });

      if (data.length > 0) {
        data.sort((a, b) => (a.time > b.time) ? 1 : ((b.time > a.time) ? -1 : 0));
      }

    } catch (error) {
      console.info('Energy Saving Report: no data found');
    }
    return (
      <EnergySavingReportComponent
        data={data}
        x="time"
        y={datum => (datum.current)}
        tooltip={tooltip}
        color={color}
        {...rest}
      />
    );
  }
}

EnergySavingReportContainer.propTypes = propTypes;
EnergySavingReportContainer.defaultProps = defaultPropTypes;
