// bluerain-app-device-explorer/src/components/Asset/AssetControl/SocketControl/SocketControl.component.js
import React, { PropTypes } from 'react';
import SwitchControl from '../ControlItems/SwitchControl';

const propTypes = {
  // Standby
  standby: PropTypes.bool,
  disableStandby: PropTypes.bool,
  onStandbyToggle: PropTypes.func,
};

const defaultProps = {
  // Standby
  standby: true,
  disableStandby: false,
  onStandbyToggle: () => {
  },

};

export default class SocketControlComponent extends React.Component {

  constructor(props) {
    super(props);

    this.onStandbyToggle = this.onStandbyToggle.bind(this);

    this.state = {
      standby: props.standby,
      disableStandby: props.disableStandby,
    };
  }

  componentWillReceiveProps(props) {
    this.setState(props);
  }

  onStandbyToggle(standby) {
    this.setState({ standby });
    this.props.onStandbyToggle(standby);
  }

  getStandbyControl() {
    const { standby, disableStandby } = this.state;

    return (
      <div>
        <SwitchControl
          label="Standby"
          disabled={disableStandby}
          toggled={!standby}
          onToggle={value => this.onStandbyToggle(!value)}
          className="list-group-item"
          switchState={standby}
        />
      </div>
    );
  }


  render() {
    return (
      <div className="card list-group device-control-list">
        {this.getStandbyControl()}
      </div>
    );
  }
}

SocketControlComponent.propTypes = propTypes;
SocketControlComponent.defaultProps = defaultProps;
