// bluerain-app-device-explorer/src/components/Asset/Charts/FridgeOpenTimeChart/FridgeOpenTimeChart.component.js
/**
 * Created by anser on 3/7/16.
 */
import React, { PropTypes } from 'react';
import AreaChartCard from 'bluerain-ui/lib/Card/ChartCard/AreaChartCard.component';

const propTypes = {
  label: PropTypes.string,
  value: PropTypes.string,
  data: PropTypes.array,
  tooltip: PropTypes.bool,
  tooltipIsBoolean: PropTypes.bool,
  tooltipBooleanTrueLabel: PropTypes.string,
  tooltipBooleanFalseLabel: PropTypes.string,
  color: PropTypes.string,
  linkTag: PropTypes.oneOfType([
    React.PropTypes.func,
    React.PropTypes.string
  ]),
  onClick: PropTypes.func.isRequired,
  x: PropTypes.string,
  y: PropTypes.func,
  to: PropTypes.string
};

const defaultProps = {
  label: 'Door Open Time',
  value: '',
  data: [],
  color: 'blue-grey',
  linkTag: 'a',
  to: '#',
  tooltipIsBoolean: true,
  tooltipBooleanTrueLabel: 'Off',
  tooltipBooleanFalseLabel: 'On',
  onClick: () => {
  }
};

export default class FridgeOpenTimeChartComponent extends React.Component {

  render() {
    const {
      data,
      x,
      y,
      color,
      label,
      value,
      tooltip,
      to,
      linkTag,
      tooltipIsBoolean,
      tooltipBooleanTrueLabel,
      tooltipBooleanFalseLabel,
      ...props
    } = this.props;

    return (
      <AreaChartCard
        data={data}
        x={x}
        y={y}
        color={color}
        label={label}
        value={value}
        tooltip={tooltip}
        linkTag={linkTag}
        to={to}
        tooltipIsBoolean={tooltipIsBoolean}
        tooltipBooleanTrueLabel={tooltipBooleanFalseLabel}
        tooltipBooleanFalseLabel={tooltipBooleanTrueLabel}
        {...props}
      />
    );
  }
}

FridgeOpenTimeChartComponent.propTypes = propTypes;
FridgeOpenTimeChartComponent.defaultProps = defaultProps;
