// bluerain-app-device-explorer/src/components/Asset/Charts/FridgeOpenTimeChart/FridgeOpenTimeChart.container.js
/**
 * Created by anser on 3/7/16.
 */
import React, { PropTypes } from 'react';
import moment from 'moment';
import FridgeOpenTimeChartComponent from './FridgeOpenTimeChart.component';

const propTypes = {
  data: PropTypes.object,
  tooltip: PropTypes.bool,
  startDate: PropTypes.string,
  endDate: PropTypes.string,
  label: PropTypes.string,
  showGraph: PropTypes.bool,
  linkTag: PropTypes.oneOfType([
    React.PropTypes.func,
    React.PropTypes.string
  ]),
  to: PropTypes.string,
  color: PropTypes.string
};

const defaultPropTypes = {
  startDate: '01/01/1970',
  endDate: '01/01/2099',
  tooltip: true,
  showGraph: true,
  linkTag: 'a',
  to: '#'
};

let data = [];
let i;
let diffArr = [];
let prevStatus = 1;
let roundedValue;
let sum;
let sumInMints;
let sumInHours;
let diff;
let doorOpenCount = 0;
let label;
export default class FridgeOpenTimeChartContainer extends React.Component {

  renderTimeDiff(data, i) {
    diffArr = [];
    data.reverse(); // to get data in descending order
    for (i = 1; i < data.length; i++) {
      prevStatus = data[i - 1].data.door; // get previous value of door status
      if (prevStatus === 1) { // only get values for door false as previous value
        const time1 = moment(new Date(data[i].time));
        const time2 = moment(new Date(data[i - 1].time));
        diff = time1.diff(time2, 'seconds'); // difference of two consecutive values in seconds
        if (diffArr.indexOf(diff) < 0) diffArr.push(diff); // push in array while avoiding duplicate entries
      }
    }
    doorOpenCount = 0;
    for (i = 0; i < data.length; i++) {
      prevStatus = data[i].data.door; // get previous value of door status
      if (prevStatus === 1) { // only get values for door true as previous value
        doorOpenCount++;
      }
    }

    sum = diffArr.reduce((a, b) => a + b); // get sum of values in array
    sumInHours = sum / 3600; // convert seconds to hours
    sum %= 3600;
    sumInMints = sum / 60; // convert seconds to minutes
    if (sumInMints > 0 && sumInMints < 1 && sumInHours < 1) {
      sumInMints = 1;
      // roundedValue = 'Less Than 1 m';
      roundedValue = ` ${Math.round(sum)} s`;
    } else {
      roundedValue = `${Math.floor(sumInHours)} h` + ` ${Math.round(sumInMints)} m`;
    }
  }

  render() {
    const {
      color,
      tooltip,
      data: dbData,
      linkTag,
      to,
      showGraph,
      ...rest
    } = this.props;
    let lastReading = 'No data found';

    if (this.props.data === undefined || this.props.data.loading) {
      lastReading = 'Loading data';
    }

    try {

      let dataFromQuery = dbData.Event.EventFind.edges;
      if (dataFromQuery.length === 0) {
        data = [];
        label = 'Door Open Time';
      } else {
        dataFromQuery = dataFromQuery.map(item => item.node);
        data = [];
        label = 'Door Open Time';
        {
          this.renderTimeDiff(dataFromQuery, i);
        }
        dataFromQuery.forEach((item, i) => {
          if (item.data && item.data.door !== undefined) {
            data.push({
              time: new Date(item.time),
              door: item.data.door ? 0 : 1
            });
          }
        });
        if (data.length > 0) {
          data.sort((a, b) => (a.time > b.time) ? 1 : ((b.time > a.time) ? -1 : 0));
          // lastReading = `${data[data.length - 1].standby ? 'On' : 'Off'}`;
          lastReading = roundedValue;
          label = `Door Opened ${doorOpenCount} Times`;
        }

      }

    } catch (error) {
      // console.info('Time Chart no data found');
    }
    // lastReading = roundedValue;
    if (showGraph) {
      return (
        <FridgeOpenTimeChartComponent
          data={data}
          x="time"
          y={datum => (datum.door)}
          color={this.props.color}
          label={label}
          value={lastReading}
          tooltip={this.props.tooltip}
          tooltipBooleanTrueLabel="Open"
          tooltipBooleanFalseLabel="Close"
          linkTag={linkTag}
          to={to}
          {...rest}
        />

      );
    }
    return (
      <FridgeOpenTimeChartComponent
        color={color}
        label={label}
        value={lastReading}
        tooltip={tooltip}
        linkTag={linkTag}
        to={to}
        {...rest}
      />
    );
  }

}

FridgeOpenTimeChartContainer.propTypes = propTypes;
FridgeOpenTimeChartContainer.defaultProps = defaultPropTypes;
