// bluerain-app-device-explorer/src/components/Asset/Charts/FridgeOpenTimeChart/index.js
/**
 * Created by anser on 3/7/16.
 */
import { graphql } from 'react-apollo';
import { fromGlobalId } from 'graphql-relay';
// import FridgeOpenTimeChartStateful from './FridgeOpenTimeChart.stateful';
import FridgeOpenTimeChartContainer from './FridgeOpenTimeChart.container';
import ViewerQuery from './query.graphql';

export default graphql(ViewerQuery, {
  options: (args) => {
    const startDateFormatted = args.startDate;
    const endDateFormatted = args.endDate;

    const queryString = {
      where: {
        type: 'state',
        assetId: fromGlobalId(args.assetId).id,
        time: { between: [startDateFormatted, endDateFormatted] },
        'data.door': { exists: true }
      },
      order: 'time DESC'
    };

    // `[{\"condition\":\"exists\",\"key\":\"data.door\"},{\"condition\":\"descending\",\"key\":\"time\"},{\"condition\":\"lessThanOrEqualTo\",\"key\":\"time\",\"value\":{\"__type\":\"Date\",
    // \"iso\":\"${endDateFormatted}\"}}, {\"condition\":\"greaterThanOrEqualTo\",\"key\":\"time\",\"value\":{\"__type\":\"Date\",\"iso\":\"${startDateFormatted}\"}}]`;

    return { variables: { assetId: args.assetId, queryString } };
  }
})(FridgeOpenTimeChartContainer);
