// bluerain-app-device-explorer/src/components/Device/AddDevice/SecondScreen/AddDeviceSecondScreen.component.js
import React, { PropTypes } from 'react';
import DataList from 'bluerain-ui/lib/DataList/DataList';
import DataListItem from 'bluerain-ui/lib/DataList/DataListItem';
import DataListItemLabel from 'bluerain-ui/lib/DataList/DataListItemLabel';
import DataListItemValue from 'bluerain-ui/lib/DataList/DataListItemValue';
import { Icon } from 'react-fa';

const propTypes = {
  wifiName: PropTypes.string,
  onClick: PropTypes.func,
  secure: PropTypes.string
};

class AddDeviceSecondScreen extends React.Component {

  constructor(props) {
    super(props);
    this.wifiName = 'WifiName';
    this.secure = '';
    this.clickHandler = (event) => {


      event.preventDefault();
      if (this.onClick) {
        this.onClick(event, {
          wifiName: this.wifiName,
          secure: this.secure
        });
      }
    };
  }

  getLockIcon() {
    if (this.secure) {
      return <DataListItemValue><Icon name="lock" /></DataListItemValue>;
    }
    return <span />;
  }

  render() {
    this.wifiName = this.props.wifiName || this.wifiName;
    this.onClick = this.props.onClick;
    this.secure = this.props.secure || this.secure;

    return (
      <DataList>
        <DataListItem style={{ cursor: 'pointer' }} action chevronRight onClick={this.clickHandler}>
          <DataListItemLabel>
            <div>{this.wifiName}</div>
          </DataListItemLabel>
          {this.getLockIcon()}
        </DataListItem>
      </DataList>
    );
  }
}

AddDeviceSecondScreen.propTypes = propTypes;

export default AddDeviceSecondScreen;
