// bluerain-app-device-explorer/lib/components/Asset/Charts/VoltageChart/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _reactApollo = require('react-apollo');

var _graphqlRelay = require('graphql-relay');

var _VoltageChart = require('./VoltageChart.container');

var _VoltageChart2 = _interopRequireDefault(_VoltageChart);

var _query = require('./query.graphql');

var _query2 = _interopRequireDefault(_query);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// import VoltageChartStateful from './VoltageChart.stateful';
/**
 * Created by atif on 12/18/16.
 */
exports.default = (0, _reactApollo.graphql)(_query2.default, {
  options: function options(args) {

    // const queryString = {
    //   type: 'state',
    //   time: { between: [args.startDate, args.endDate] },
    //   'data.voltage': { exists: true }
    // };
    var queryString = {
      where: {
        type: 'state',
        assetId: (0, _graphqlRelay.fromGlobalId)(args.assetId).id,
        time: { between: [args.startDate, args.endDate] },
        'data.voltage': { exists: true }
      },
      order: 'time DESC'
    };
    return { variables: { assetId: args.assetId, queryString: queryString } };
  }
})(_VoltageChart2.default);
