// bluerain-ui/lib/DataList/DataList.component.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactFa = require('react-fa');

var _reactFa2 = _interopRequireDefault(_reactFa);

var _reactRouter = require('react-router');

var _DataList = require('./DataList');

var _DataList2 = _interopRequireDefault(_DataList);

var _DataListItem = require('./DataListItem');

var _DataListItem2 = _interopRequireDefault(_DataListItem);

var _DataListItemLabel = require('./DataListItemLabel');

var _DataListItemLabel2 = _interopRequireDefault(_DataListItemLabel);

var _DataListItemValue = require('./DataListItemValue');

var _DataListItemValue2 = _interopRequireDefault(_DataListItemValue);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Created by anser on 12/23/16.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */

var propTypes = {
  myData: _react.PropTypes.array
};

var defaultProps = {
  myData: []
};

var DataListComponent = function (_React$Component) {
  _inherits(DataListComponent, _React$Component);

  function DataListComponent() {
    _classCallCheck(this, DataListComponent);

    return _possibleConstructorReturn(this, (DataListComponent.__proto__ || Object.getPrototypeOf(DataListComponent)).apply(this, arguments));
  }

  _createClass(DataListComponent, [{
    key: 'render',
    value: function render() {
      var myData = this.props.myData;


      return _react2.default.createElement(
        _DataList2.default,
        null,
        myData.map(function (item, i) {
          return _react2.default.createElement(
            _DataListItem2.default,
            { key: i, tag: _reactRouter.Link, to: item.to, action: item.action, chevronRight: item.chevronRight },
            DataListComponent.labelText(item.label),
            DataListComponent.valueText(item.valueText, item.valueIcon)
          );
        })
      );
    }
  }], [{
    key: 'valueIcon',
    value: function valueIcon(_valueIcon) {
      if (_valueIcon) {
        return _react2.default.createElement(_reactFa2.default, { name: _valueIcon });
      }
    }
  }, {
    key: 'valueText',
    value: function valueText(_valueText, valueIcon) {
      if (_valueText || valueIcon) {
        return _react2.default.createElement(
          _DataListItemValue2.default,
          null,
          _valueText,
          ' ',
          DataListComponent.valueIcon(valueIcon)
        );
      }
    }
  }, {
    key: 'labelText',
    value: function labelText(label) {
      if (label) {
        return _react2.default.createElement(
          _DataListItemLabel2.default,
          null,
          label
        );
      }
    }
  }]);

  return DataListComponent;
}(_react2.default.Component);

exports.default = DataListComponent;


DataListComponent.propTypes = propTypes;
DataListComponent.defaultProps = defaultProps;
