// bluerain-ui/lib/DataList/DataListItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactFa = require('react-fa');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _ListGroupItem = require('reactstrap/lib/ListGroupItem');

var _ListGroupItem2 = _interopRequireDefault(_ListGroupItem);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; } /**
                                                                                                                                                                                                                              * Created by anser on 11/25/16.
                                                                                                                                                                                                                              */

var propTypes = {
  tag: _react.PropTypes.oneOfType([_react.PropTypes.func, _react.PropTypes.string]),
  className: _react.PropTypes.string,
  chevronRight: _react.PropTypes.bool,
  to: _react.PropTypes.string,
  action: _react.PropTypes.bool,
  children: _react.PropTypes.array
};

var defaultProps = {
  tag: 'div',
  className: '',
  chevronRight: true,
  to: '#',
  action: true,
  children: []
};

var DataListItem = function DataListItem(props) {
  var Tag = props.tag,
      chevronRight = props.chevronRight,
      to = props.to,
      action = props.action,
      children = props.children,
      attributes = _objectWithoutProperties(props, ['tag', 'chevronRight', 'to', 'action', 'children']);

  // Classes


  var className = (0, _classnames2.default)('', {
    'list-group-item-action': action
  }, props.className);

  // Chevron
  var chevron = '';
  if (chevronRight === true) {
    chevron = _react2.default.createElement(
      Tag,
      { to: to, className: 'data-chevron' },
      _react2.default.createElement(_reactFa.Icon, { name: 'chevron-right' })
    );
  }
  return _react2.default.createElement(
    _ListGroupItem2.default,
    _extends({}, attributes, { className: className }),
    children,
    chevron
  );
};

DataListItem.propTypes = propTypes;
DataListItem.defaultProps = defaultProps;

exports.default = DataListItem;
