// bluerain-app-device-explorer/lib/components/Asset/AssetSensorInfo/SensorInfo/AssetSensorInfo.component.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactFa = require('react-fa');

var _reactFa2 = _interopRequireDefault(_reactFa);

require('bluerain-bootstrap-theme/dist/css/components/data-list.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Created by anser on 12/26/16.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


var propTypes = {
  data: _react.PropTypes.object
};

var defaultProps = {};

var AssetSensorInfo = function (_React$Component) {
  _inherits(AssetSensorInfo, _React$Component);

  function AssetSensorInfo() {
    _classCallCheck(this, AssetSensorInfo);

    return _possibleConstructorReturn(this, (AssetSensorInfo.__proto__ || Object.getPrototypeOf(AssetSensorInfo)).apply(this, arguments));
  }

  _createClass(AssetSensorInfo, [{
    key: 'render',
    value: function render() {
      // console.log('this.props.data.node: ' , this.props.data.node)

      if (!this.props.data.node || !this.props.data.node.state) {
        return _react2.default.createElement('div', null);
      }

      var _props$data$node$stat = this.props.data.node.state,
          connected = _props$data$node$stat.connected,
          ambientTemperature = _props$data$node$stat.ambientTemperature,
          current = _props$data$node$stat.current,
          power = _props$data$node$stat.power,
          voltage = _props$data$node$stat.voltage;


      return _react2.default.createElement(
        'ul',
        { className: 'list-group data-list' },
        AssetSensorInfo.renderConnectionStatus(connected),
        AssetSensorInfo.renderAmbientTemperature(ambientTemperature),
        AssetSensorInfo.renderCurrent(current),
        AssetSensorInfo.renderPower(power),
        AssetSensorInfo.renderVoltage(voltage)
      );
    }
  }], [{
    key: 'hasAnySensorData',
    value: function hasAnySensorData() {
      var _props$data$node$stat2 = this.props.data.node.state,
          connected = _props$data$node$stat2.connected,
          ambientTemperature = _props$data$node$stat2.ambientTemperature,
          current = _props$data$node$stat2.current,
          power = _props$data$node$stat2.power,
          voltage = _props$data$node$stat2.voltage;


      if (connected !== undefined && connected !== null) {
        return true;
      } else if (ambientTemperature !== undefined && ambientTemperature !== null) {
        return true;
      } else if (current !== undefined && current !== null) {
        return true;
      } else if (power !== undefined && power !== null) {
        return true;
      } else if (voltage !== undefined && voltage !== null) {
        return true;
      }

      return false;
    }
  }, {
    key: 'renderConnectionStatus',
    value: function renderConnectionStatus(connected) {

      var status = null;

      if (connected === undefined || connected === null) {
        return _react2.default.createElement('div', null);
      } else if (connected) {
        status = _react2.default.createElement(
          'span',
          null,
          'Online ',
          _react2.default.createElement(_reactFa2.default, { name: 'circle text-success' })
        );
      } else {
        status = _react2.default.createElement(
          'span',
          null,
          'Offline ',
          _react2.default.createElement(_reactFa2.default, { name: 'circle text-danger' })
        );
      }
      return _react2.default.createElement(
        'li',
        { className: 'list-group-item' },
        _react2.default.createElement(
          'div',
          { className: 'data-label' },
          'Status'
        ),
        _react2.default.createElement(
          'div',
          { className: 'data-value' },
          status
        )
      );
    }
  }, {
    key: 'renderAmbientTemperature',
    value: function renderAmbientTemperature(ambientTemperature) {
      if (ambientTemperature === undefined || ambientTemperature === null) {
        return _react2.default.createElement('div', null);
      }

      return _react2.default.createElement(
        'li',
        { className: 'list-group-item' },
        _react2.default.createElement(
          'div',
          { className: 'data-label' },
          'Room Temperature'
        ),
        _react2.default.createElement(
          'div',
          { className: 'data-value' },
          ambientTemperature,
          ' \u2103'
        )
      );
    }
  }, {
    key: 'renderCurrent',
    value: function renderCurrent(current) {
      if (current === undefined || current === null) {
        return _react2.default.createElement('div', null);
      }

      return _react2.default.createElement(
        'li',
        { className: 'list-group-item' },
        _react2.default.createElement(
          'div',
          { className: 'data-label' },
          'Current'
        ),
        _react2.default.createElement(
          'div',
          { className: 'data-value' },
          current,
          ' A'
        )
      );
    }
  }, {
    key: 'renderPower',
    value: function renderPower(power) {
      if (power === undefined || power === null) {
        return _react2.default.createElement('div', null);
      }

      return _react2.default.createElement(
        'li',
        { className: 'list-group-item' },
        _react2.default.createElement(
          'div',
          { className: 'data-label' },
          'Power'
        ),
        _react2.default.createElement(
          'div',
          { className: 'data-value' },
          power,
          ' W'
        )
      );
    }
  }, {
    key: 'renderVoltage',
    value: function renderVoltage(voltage) {
      if (voltage === undefined || voltage === null) {
        return _react2.default.createElement('div', null);
      }

      return _react2.default.createElement(
        'li',
        { className: 'list-group-item' },
        _react2.default.createElement(
          'div',
          { className: 'data-label' },
          'Voltage'
        ),
        _react2.default.createElement(
          'div',
          { className: 'data-value' },
          voltage,
          ' V'
        )
      );
    }
  }]);

  return AssetSensorInfo;
}(_react2.default.Component);

exports.default = AssetSensorInfo;


AssetSensorInfo.propTypes = propTypes;
AssetSensorInfo.defaultProps = defaultProps;
