// bluerain-app-device-explorer/src/pages/TransferRequests/TransferRequestsPage.js
/**
 * Created by anser on 2/23/17.
 */
import React, { PropTypes } from 'react';
import { Container, Row, Col } from 'reactstrap';
import AppContent from 'bluerain-ui/lib/Layout/AppContent';
import PrimaryAppNav from 'bluerain-ui/lib/Nav/PrimaryAppNav';
import TransferRequestWidget from '../../components/Asset/AssetTransferRequests/AssetTransferRequests.widget';

const propTypes = {
  title: PropTypes.string
};

const defaultProps = {
  title: 'Ownership Transfer Requests'
};

class SiteFormPage extends React.Component {

  static renderPrimaryNav() {
    const title = this.props.title;
    const previous = {
      title: 'Sites',
      url: '/app/device-explorer'
    };

    return (
      <PrimaryAppNav title={title} previous={previous}>
        <i className="fa fa-plus hidden-sm-up" />
      </PrimaryAppNav>
    );
  }

  static renderSecondaryNav() {
  }

  static renderContent() {
    return (
      <AppContent>
        <Container>
          <Row>
            <Col md={{ size: 8, offset: 2 }} className="col-faded">
              <TransferRequestWidget />
            </Col>
          </Row>
        </Container>
      </AppContent>
    );
  }

  render() {
    return (
      <div>
        {SiteFormPage.renderPrimaryNav()}
        {SiteFormPage.renderSecondaryNav()}
        {SiteFormPage.renderContent()}
      </div>
    );
  }
}


SiteFormPage.propTypes = propTypes;
SiteFormPage.defaultProps = defaultProps;

export default SiteFormPage;
