// bluerain-app-device-explorer/lib/components/Asset/Charts/ChartSlider/ChartSlider.widget.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _widget = require('bluerain-ui/lib/Widget/widget');

var _widget2 = _interopRequireDefault(_widget);

var _widgetTitle = require('bluerain-ui/lib/Widget/widgetTitle');

var _widgetTitle2 = _interopRequireDefault(_widgetTitle);

var _widgetHeader = require('bluerain-ui/lib/Widget/widgetHeader');

var _widgetHeader2 = _interopRequireDefault(_widgetHeader);

var _widgetContent = require('bluerain-ui/lib/Widget/widgetContent');

var _widgetContent2 = _interopRequireDefault(_widgetContent);

var _reactstrap = require('reactstrap');

var _reactRouter = require('react-router');

var _reactSlick = require('react-slick');

var _reactSlick2 = _interopRequireDefault(_reactSlick);

var _OverlaySlider = require('bluerain-ui/lib/Slider/OverlaySlider');

var _OverlaySlider2 = _interopRequireDefault(_OverlaySlider);

var _reactDates = require('react-dates');

require('react-dates/lib/css/_datepicker.css');

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

require('../../../../css/custom_datepicker.css');

var _ComponentRegistry = require('../../../../components/ComponentRegistry');

var _ComponentRegistry2 = _interopRequireDefault(_ComponentRegistry);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Created by atif on 12/18/16.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


// const BillChart = registry.getComponent('BillChart', 'Generic').content;
var CurrentChart = _ComponentRegistry2.default.getComponent('CurrentChart', 'Generic').content;
var EnergySavingChart = _ComponentRegistry2.default.getComponent('EnergySavingChart', 'Generic').content;
var EnergyChart = _ComponentRegistry2.default.getComponent('EnergyChart', 'Generic').content;
var PowerChart = _ComponentRegistry2.default.getComponent('PowerChart', 'Generic').content;
// const TemperatureChart = registry.getComponent('TemperatureChart', 'Generic').content;
// const TimeChart = registry.getComponent('TimeChart', 'Generic').content;
var VoltageChart = _ComponentRegistry2.default.getComponent('VoltageChart', 'Generic').content;

var propTypes = {
  assetId: _react.PropTypes.string
};

var defaultProps = {
  assetId: ''
};

var ChartSliderWidget = function (_React$Component) {
  _inherits(ChartSliderWidget, _React$Component);

  function ChartSliderWidget(props) {
    _classCallCheck(this, ChartSliderWidget);

    var _this = _possibleConstructorReturn(this, (ChartSliderWidget.__proto__ || Object.getPrototypeOf(ChartSliderWidget)).call(this, props));

    _this.state = {
      focused: false,
      date: (0, _moment2.default)()
    };

    _this.onDateChange = _this.onDateChange.bind(_this);
    _this.onFocusChange = _this.onFocusChange.bind(_this);
    return _this;
  }

  _createClass(ChartSliderWidget, [{
    key: 'onDateChange',
    value: function onDateChange(date) {
      this.setState({ date: date });
    }
  }, {
    key: 'onFocusChange',
    value: function onFocusChange(_ref) {
      var focused = _ref.focused;

      this.setState({ focused: focused });
    }
  }, {
    key: 'render',
    value: function render() {
      var _state = this.state,
          focused = _state.focused,
          date = _state.date;

      var assetId = this.props.assetId;
      var startDateFormatted = (0, _moment2.default)(date).utc().startOf('day').toISOString();
      var endDateFormatted = (0, _moment2.default)(date).utc().endOf('day').toISOString();

      var settings = {
        dots: true,
        infinite: false,
        adaptiveHeight: false,
        speed: 500,
        slidesToShow: 3,
        arrows: false,
        slidesToScroll: 1,
        responsive: [{
          breakpoint: 1200,
          settings: {
            slidesToShow: 3,
            slidesToScroll: 3,
            infinite: true
          }
        }, {
          breakpoint: 992,
          settings: {
            slidesToShow: 3,
            slidesToScroll: 3
          }
        }, {
          breakpoint: 768,
          settings: {
            slidesToShow: 2,
            slidesToScroll: 2
          }
        }, {
          breakpoint: 576,
          settings: {
            slidesToShow: 1,
            slidesToScroll: 1
          }
        }]
      };

      var reportBaseUrl = '/app/device-explorer/assets/' + assetId;

      return _react2.default.createElement(
        _widget2.default,
        { style: { marginBottom: 0 } },
        _react2.default.createElement(
          _reactstrap.Container,
          null,
          _react2.default.createElement(
            _widgetHeader2.default,
            null,
            _react2.default.createElement(
              _widgetTitle2.default,
              null,
              'Reports'
            ),
            _react2.default.createElement(_reactDates.SingleDatePicker, _extends({}, this.props, {
              id: 'date_input',
              date: date,
              numberOfMonths: 1,
              focused: focused,
              onDateChange: this.onDateChange,
              onFocusChange: this.onFocusChange,
              isOutsideRange: function isOutsideRange() {
                return false;
              }
            }))
          )
        ),
        _react2.default.createElement(
          _widgetContent2.default,
          null,
          _react2.default.createElement(
            _OverlaySlider2.default,
            null,
            _react2.default.createElement(
              _reactSlick2.default,
              settings,
              _react2.default.createElement(
                'div',
                null,
                _react2.default.createElement(EnergyChart, {
                  linkTag: _reactRouter.Link,
                  to: reportBaseUrl + '/reports/energy',
                  assetId: assetId,
                  startDate: startDateFormatted,
                  endDate: endDateFormatted,
                  queryString: ''
                })
              ),
              _react2.default.createElement(
                'div',
                null,
                _react2.default.createElement(CurrentChart, {
                  linkTag: _reactRouter.Link,
                  to: reportBaseUrl + '/reports/current',
                  assetId: assetId,
                  startDate: startDateFormatted,
                  endDate: endDateFormatted,
                  queryString: ''
                })
              ),
              _react2.default.createElement(
                'div',
                null,
                _react2.default.createElement(EnergySavingChart, {
                  linkTag: _reactRouter.Link,
                  to: reportBaseUrl + '/reports/energy-saving',
                  assetId: assetId,
                  startDate: startDateFormatted,
                  endDate: endDateFormatted,
                  queryString: ''
                })
              ),
              _react2.default.createElement(
                'div',
                null,
                _react2.default.createElement(VoltageChart, {
                  linkTag: _reactRouter.Link,
                  to: reportBaseUrl + '/reports/voltage',
                  assetId: assetId,
                  startDate: startDateFormatted,
                  endDate: endDateFormatted,
                  queryString: ''
                })
              ),
              _react2.default.createElement(
                'div',
                null,
                _react2.default.createElement(PowerChart, {
                  linkTag: _reactRouter.Link,
                  to: reportBaseUrl + '/reports/power',
                  assetId: assetId,
                  startDate: startDateFormatted,
                  endDate: endDateFormatted,
                  queryString: ''
                })
              )
            )
          )
        )
      );
    }
  }]);

  return ChartSliderWidget;
}(_react2.default.Component);

exports.default = ChartSliderWidget;


ChartSliderWidget.propTypes = propTypes;
ChartSliderWidget.defaultProps = defaultProps;
