// bluerain-app-device-explorer/src/components/Asset/AssetSensorInfo/RefrigeratorSensorInfo/RefrigeratorSensorInfo.component.js
import React, { PropTypes } from 'react';
import Icon from 'react-fa';
import 'bluerain-bootstrap-theme/dist/css/components/data-list.css';
// import DataListComponent from '../DataList/DataList.component';

const propTypes = {
  data: PropTypes.object,
};

const defaultProps = {};

export default class RefrigeratorSensorInfo extends React.Component {

  static renderConnectionStatus(connected) {

    let status = null;

    if (typeof (connected) !== 'boolean') {
      return (<div />);
    } else if (connected) {
      status = (<span>Online <Icon name="circle text-success" /></span>);
    } else {
      status = (<span>Offline <Icon name="circle text-danger" /></span>);
    }
    return (
      <li className="list-group-item">
        <div className="data-label">Status</div>
        <div className="data-value">{status}</div>
      </li>
    );
  }

  static renderAmbientTemperature(temperature, connected) {
    if (typeof (temperature) !== 'number') {
      return (<div />);
    }
    if (connected) {
      return (
        <li className="list-group-item">
          <div className="data-label">Temperature</div>
          <div className="data-value">{temperature.toFixed(0)} ℃</div>
        </li>
      );
    }
    return (<div />);
  }

  static renderColor(color, connected) {
    if (connected) {
      return (
        <li className="list-group-item">
          <div className="data-label">Color</div>
          <div className="data-value">{color}</div>
        </li>
      );
    }
    return (<div />);
  }

  static renderCurrent(current, connected, standby) {
    if (typeof (current) !== 'number') {
      return (<div />);
    }
    if (standby) {
      return (<div />);
    }
    if (connected) {
      return (
        <li className="list-group-item">
          <div className="data-label">Current</div>
          <div className="data-value">{current.toFixed(2)} A</div>
        </li>
      );
    }
    return (<div />);
  }

  static renderPower(power, connected, standby) {
    if (typeof (power) !== 'number') {
      return (<div />);
    }
    if (standby) {
      return (<div />);
    }
    if (connected) {
      return (
        <li className="list-group-item">
          <div className="data-label">Power</div>
          <div className="data-value">{power.toFixed(0)} W</div>
        </li>
      );
    }
    return (<div />);
  }

  static renderVoltage(voltage, connected, standby) {
    if (typeof (voltage) !== 'number') {
      return (<div />);
    }
    if (standby) {
      return (<div />);
    }
    if (connected) {
      return (
        <li className="list-group-item">
          <div className="data-label">Voltage</div>
          <div className="data-value">{voltage.toFixed(0)} V</div>
        </li>
      );
    }
    return (<div />);
  }

  render() {
    if (!this.props.data.node || !this.props.data.node.state) {
      return (<div />);
    }

    const {
      connected,
      temperature,
      current,
      power,
      voltage,
      color,
      standby
    } = this.props.data.node.state;

    return (
      <ul className="list-group data-list">
        { RefrigeratorSensorInfo.renderConnectionStatus(connected) }
        { RefrigeratorSensorInfo.renderColor(color, connected) }
        { RefrigeratorSensorInfo.renderAmbientTemperature(temperature, connected) }
        { RefrigeratorSensorInfo.renderCurrent(current, connected, standby) }
        { RefrigeratorSensorInfo.renderPower(power, connected, standby) }
        { RefrigeratorSensorInfo.renderVoltage(voltage, connected, standby) }
      </ul>
    );
  }
}

RefrigeratorSensorInfo.propTypes = propTypes;
RefrigeratorSensorInfo.defaultProps = defaultProps;
