// bluerain-app-device-explorer/lib/components/Asset/Charts/TemperatureChart/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _reactApollo = require('react-apollo');

var _query = require('./query.graphql');

var _query2 = _interopRequireDefault(_query);

var _graphqlRelay = require('graphql-relay');

var _TemperatureChart = require('./TemperatureChart.container');

var _TemperatureChart2 = _interopRequireDefault(_TemperatureChart);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// import TemperatureChartStateful from './TemperatureChart.stateful';
exports.default = (0, _reactApollo.graphql)(_query2.default, {
  options: function options(args) {
    var startDateFormatted = args.startDate;
    var endDateFormatted = args.endDate;

    // const queryString = {
    //   type: 'state',
    //   time: { between: [startDateFormatted, endDateFormatted] },
    //   'data.ambientTemperature': { exists: true }
    // };

    var queryString = {
      where: {
        type: 'state',
        assetId: (0, _graphqlRelay.fromGlobalId)(args.assetId).id,
        time: { between: [startDateFormatted, endDateFormatted] },
        'data.ambientTemperature': { exists: true }
      },
      order: 'time DESC'
    };

    return { variables: { assetId: args.assetId, queryString: queryString } };
  }
})(_TemperatureChart2.default);
