// bluerain-app-device-explorer/src/components/Report/CurrentReport/CurrentReport.component.js
/**
 * Created by atif on 12/18/16.
 */
import React, { PropTypes } from 'react';
import AreaChart from 'bluerain-ui/lib/Card/ChartCard/AreaChartAxis.component';
// import moment from 'moment';
// import { Table } from 'reactstrap';

const propTypes = {
  data: PropTypes.array,
  x: PropTypes.string,
  y: PropTypes.func,
  color: PropTypes.string,
  tooltip: PropTypes.bool
};

const defaultProps = {
  data: []
};
let records;
let fillColor;
export default class CurrentReportComponent extends React.Component {

  render() {
    try {
      records = this.props.data;
      fillColor = '#C84DD6';
    }catch (error) {
      console.info('Current Report: no data found');
    }

    return (
      <div>
        <div style={{ width: '100%', margin: '0 auto' }}>
          <AreaChart
            data={records}
            x={this.props.x}
            y={this.props.y}
            tooltip={this.props.tooltip}
            height={300}
            fillColor={fillColor}
            strokeColor={this.props.color}
            dateFormat="DD, MMM YYYY, hh:mm:ss a"
            interpolation="basis"
          />
        </div>
        {/*<div className="panel-data border-top">*/}
          {/*<div className="table-responsive">*/}
            {/*<Table>*/}
              {/*<thead className="thead-default">*/}
                {/*<tr>*/}
                  {/*<th>Date/Time</th>*/}
                  {/*<th>Value (A)</th>*/}
                {/*</tr>*/}
              {/*</thead>*/}
              {/*<tbody>*/}
                {/*{*/}
                  {/*records.map((item, index) => (*/}
                  {/*<tr key={index}>*/}
                    {/*<td>{moment(item.time).format('DD, MMM YYYY, hh:mm:ss a')}</td>*/}
                    {/*<td>{ item.current !== undefined ? item.current.toFixed(2) : 0 }</td>*/}
                  {/*</tr>*/}
                {/*))*/}
              {/*}*/}
              {/*</tbody>*/}
            {/*</Table>*/}
          {/*</div>*/}
        {/*</div>*/}
      </div>
    );
  }
}

CurrentReportComponent.propTypes = propTypes;
CurrentReportComponent.defaultProps = defaultProps;
