// bluerain-app-device-explorer/lib/pages/SitePage/SiteEditForm/SiteEditFormPage.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactstrap = require('reactstrap');

var _AppContent = require('bluerain-ui/lib/Layout/AppContent');

var _AppContent2 = _interopRequireDefault(_AppContent);

var _PrimaryAppNav = require('bluerain-ui/lib/Nav/PrimaryAppNav');

var _PrimaryAppNav2 = _interopRequireDefault(_PrimaryAppNav);

var _SiteEditForm = require('../../../components/Site/SiteEditForm/SiteEditForm.widget');

var _SiteEditForm2 = _interopRequireDefault(_SiteEditForm);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Created by anser on 29/6/17.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


var propTypes = {
  title: _react.PropTypes.string,
  params: _react.PropTypes.object
};

var defaultProps = {
  title: 'Edit Site Form'
};

var SiteEditFormPage = function (_React$Component) {
  _inherits(SiteEditFormPage, _React$Component);

  function SiteEditFormPage() {
    _classCallCheck(this, SiteEditFormPage);

    return _possibleConstructorReturn(this, (SiteEditFormPage.__proto__ || Object.getPrototypeOf(SiteEditFormPage)).apply(this, arguments));
  }

  _createClass(SiteEditFormPage, [{
    key: 'renderPrimaryNav',
    value: function renderPrimaryNav() {
      var title = this.props.title;
      var previous = {
        title: 'Sites',
        url: '/app/device-explorer'
      };
      return _react2.default.createElement(
        _PrimaryAppNav2.default,
        { title: title, previous: previous },
        _react2.default.createElement('i', { className: 'fa fa-plus hidden-sm-up' })
      )
      // <PrimaryAppNav title={title} />
      ;
    }
  }, {
    key: 'render',
    value: function render() {
      var siteId = this.props.params.siteId;

      return _react2.default.createElement(
        'div',
        null,
        this.renderPrimaryNav(),
        SiteEditFormPage.renderSecondaryNav(),
        SiteEditFormPage.renderContent(siteId)
      );
    }
  }], [{
    key: 'renderSecondaryNav',
    value: function renderSecondaryNav() {}
  }, {
    key: 'renderContent',
    value: function renderContent(siteId) {
      return _react2.default.createElement(
        _AppContent2.default,
        null,
        _react2.default.createElement(
          _reactstrap.Container,
          null,
          _react2.default.createElement(
            _reactstrap.Row,
            null,
            _react2.default.createElement(
              _reactstrap.Col,
              { md: { size: 8, offset: 2 }, className: 'col-faded' },
              _react2.default.createElement(_SiteEditForm2.default, { siteId: siteId })
            )
          )
        )
      );
    }
  }]);

  return SiteEditFormPage;
}(_react2.default.Component);

SiteEditFormPage.propTypes = propTypes;
SiteEditFormPage.defaultProps = defaultProps;

exports.default = SiteEditFormPage;
