// bluerain-app-device-explorer/src/components/Report/RefrigeratorTemperatureReport/RefrigeratorTemperatureReport.container.js
/**
 * Created by atif on 12/18/16.
 */
import React, { PropTypes } from 'react';
import TemperatureReportComponent from './RefrigeratorTemperatureReport.component';

const propTypes = {
  data: PropTypes.array,
  tooltip: PropTypes.bool,
  startDate: PropTypes.string,
  endDate: PropTypes.string,
  color: PropTypes.string
};

const defaultPropTypes = {
  startDate: '01/01/1970',
  endDate: '01/01/2099',
  tooltip: true,
  color: 'yellow'
};

export default class TemperatureReportContainer extends React.Component {

  render() {

    const {
      color,
      tooltip,
      data: dbData,
      ...rest
    } = this.props;


    const data = [];

    try {
      // let dataFromQuery = dbData.node.events.edges;
      let dataFromQuery = dbData.Event.EventFind.edges;
      dataFromQuery = dataFromQuery.map(item => item.node);

      dataFromQuery.forEach((item) => {
        if (item.data && item.data.temperature !== undefined) {
          data.push({
            time: new Date(item.time),
            temperature: item.data.temperature
          });
        }
      });

      if (data.length > 0) {
        data.sort((a, b) => (a.time > b.time) ? 1 : ((b.time > a.time) ? -1 : 0));
      }

    } catch (error) {
      console.info('Temperature Report: no data found');
    }

    return (
      <TemperatureReportComponent
        data={data}
        x="time"
        y={datum => (datum.temperature)}
        tooltip={tooltip}
        color={color}
        {...rest}
      />
    );
  }
}

TemperatureReportContainer.propTypes = propTypes;
TemperatureReportContainer.defaultProps = defaultPropTypes;
