// bluerain-app-device-explorer/lib/components/Post/Timeline/Timeline.component.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _PostCard = require('bluerain-ui/lib/Card/PostCard/PostCard.component');

var _PostCard2 = _interopRequireDefault(_PostCard);

var _reactRouter = require('react-router');

var _reactCollapse = require('react-collapse');

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  posts: _react.PropTypes.array
};

var defaultProps = {
  posts: []
};

var user = void 0;
var imageUrl = void 0;
var initiator = void 0;
var source = void 0;

var TimelineComponent = function (_React$Component) {
  _inherits(TimelineComponent, _React$Component);

  function TimelineComponent() {
    _classCallCheck(this, TimelineComponent);

    return _possibleConstructorReturn(this, (TimelineComponent.__proto__ || Object.getPrototypeOf(TimelineComponent)).apply(this, arguments));
  }

  _createClass(TimelineComponent, [{
    key: 'render',
    value: function render() {
      var posts = this.props.posts;
      posts = posts.sort(function (a, b) {
        return a.time < b.time ? 1 : b.time < a.time ? -1 : 0;
      });
      return _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(
          _reactCollapse.Collapse,
          { isOpened: true },
          posts.map(function (post) {
            return TimelineComponent.renderPost(post);
          })
        )
      );
    }
  }], [{
    key: 'renderPost',
    value: function renderPost(post) {
      var description = post.description;
      switch (post.type && post.type.toUpperCase()) {
        case 'USER_COMMAND':
          if (post.accounts && post.accounts.edges.length !== 0 && post.accounts.edges[0] && post.accounts.edges[0].node !== null) {
            user = post.accounts.edges[0].node;
            imageUrl = 'https://s3-us-west-2.amazonaws.com/bluerainimages/default-user.jpg';
            if (user.profilePicture && user.profilePicture.url) {
              imageUrl = user.profilePicture.url;
            } else {
              imageUrl = 'https://s3-us-west-2.amazonaws.com/bluerainimages/default-user.jpg';
            }
            source = 'Web App';

            initiator = user.firstName && user.lastName ? user.firstName + ' ' + user.lastName : user.username;
          }
          break;
        case 'DEVICE_REGISTRATION':
          if (post.accounts && post.accounts.edges.length !== 0 && post.accounts.edges[0] && post.accounts.edges[0].node !== null) {
            user = post.accounts.edges[0].node;
            if (user.profilePicture && user.profilePicture.url) {
              imageUrl = user.profilePicture.url;
            } else {
              imageUrl = 'https://s3-us-west-2.amazonaws.com/bluerainimages/default-user.jpg';
            }
            source = 'Web App';
            initiator = user.firstName + ' ' + user.lastName;
          }
          break;
        case 'REGISTER':
          imageUrl = 'https://s3-us-west-2.amazonaws.com/bluerainimages/register_icon.svg';
          initiator = '';
          source = 'Web App';
          description = description + ' at ' + (0, _moment2.default)(post.time).format('LLL');
          break;
        case 'SCHEDULE':
          imageUrl = 'https://s3-us-west-2.amazonaws.com/bluerainimages/schedule.svg';
          initiator = '';
          source = 'Scheduled App';
          break;
        case 'FIRMWARE_UPGRADE':
          imageUrl = 'https://s3-us-west-2.amazonaws.com/bluerainimages/firmware_update.jpg';
          initiator = '';
          source = 'Network';
          break;
        case 'ALERT':
          imageUrl = 'https://s3-us-west-2.amazonaws.com/bluerainimages/error_icon.png';
          initiator = '';
          source = 'Device';
          break;
        case 'IR_CONTROL':
          imageUrl = 'https://s3-us-west-2.amazonaws.com/bluerainimages/remote.svg';
          initiator = '';
          source = 'Remote';
          break;
        case 'FLOW_RULE_EXECUTED':
          imageUrl = 'https://s3-us-west-2.amazonaws.com/bluerainimages/node-red-icon.svg';
          initiator = '';
          source = 'Automated Rule';
          break;
        case 'CONNECTION':
          imageUrl = 'https://s3-us-west-2.amazonaws.com/bluerainimages/device-offline.svg';
          initiator = '';
          source = 'Network';
          break;
        case 'DOOR_OPEN':
          imageUrl = 'https://s3-us-west-2.amazonaws.com/bluerainimages/icon-fridge-door-open.svg';
          initiator = '';
          source = 'Network';
          break;
        case 'DOOR_CLOSED':
          imageUrl = 'https://s3-us-west-2.amazonaws.com/bluerainimages/icon-fridge-door-close.svg';
          initiator = '';
          source = 'Network';
          break;
        default:
          imageUrl = 'https://s3-us-west-2.amazonaws.com/bluerainimages/default-user.jpg';
          initiator = '';
          source = '';
      }

      var assetId = post.assets.edges.length !== 0 && post.assets.edges[0] ? post.assets.edges[0].node.id : 'noId';
      // const eventId = '123'; // post.events.edges[0].node.id;
      var data = {
        description: description,
        initiator: initiator,
        time: post.time,
        source: source,
        roundImage: true,
        image: imageUrl,
        // image: user.profilePicture.url || 'https://s3-us-west-2.amazonaws.com/bluerainimages/default-user.jpg',
        userProfileLink: '',
        href: '/app/device-explorer/assets/' + assetId + '/posts/' + post.id
      };
      if (data.description && data.description.includes('rgb(')) {

        var index = data.description.indexOf('rgb');
        var strOut = data.description.substr(index);
        var style = {
          width: 24,
          height: 10,
          borderRadius: 7,
          backgroundColor: strOut,
          display: 'inline-block'
        };
        data.description = _react2.default.createElement(
          'span',
          null,
          'changed color to ',
          _react2.default.createElement('div', { style: style })
        );
      }
      return _react2.default.createElement(_PostCard2.default, {
        key: post.id,
        initiator: data.initiator,
        description: data.description,
        time: data.time,
        source: data.source,
        roundImage: data.roundImage,
        image: data.image,
        userProfileLink: data.userProfileLink,
        linkTag: _reactRouter.Link,
        to: data.href
      });
    }
  }]);

  return TimelineComponent;
}(_react2.default.Component);

TimelineComponent.propTypes = propTypes;
TimelineComponent.defaultProps = defaultProps;

exports.default = TimelineComponent;
