// bluerain-app-device-explorer/lib/components/Post/Timeline/Timeline.container.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

require('bluerain-bootstrap-theme/dist/css/components/timeline.css');

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _bluerainClientServices = require('bluerain-client-services');

var _Timeline = require('./Timeline.component');

var _Timeline2 = _interopRequireDefault(_Timeline);

var _reactWaypoint = require('react-waypoint');

var _reactWaypoint2 = _interopRequireDefault(_reactWaypoint);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  data: _react.PropTypes.object,
  showLoading: _react.PropTypes.func,
  assetId: _react.PropTypes.string,
  hideLoading: _react.PropTypes.func
};

var defaultProps = {
  data: {}
};

var TimelineContainer = function (_React$Component) {
  _inherits(TimelineContainer, _React$Component);

  function TimelineContainer() {
    var _ref;

    var _temp, _this, _ret;

    _classCallCheck(this, TimelineContainer);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = TimelineContainer.__proto__ || Object.getPrototypeOf(TimelineContainer)).call.apply(_ref, [this].concat(args))), _this), _this._handleEnter = function (props) {
      _this.handleLoadMore();
    }, _this.handleLoadMore = function () {
      var endCursor = _this.props.data.node.posts.pageInfo.endCursor;

      if (endCursor) {
        _this.props.showLoading();
        _this.props.data.fetchMore({
          variables: { assetId: _this.props.assetId, before: endCursor },
          updateQuery: function updateQuery(previousResult, _ref2) {
            var fetchMoreResult = _ref2.fetchMoreResult;

            var node = Object.assign({}, previousResult.node, {
              posts: {
                __typename: fetchMoreResult.node.posts.__typename,
                pageInfo: fetchMoreResult.node.posts.pageInfo,
                edges: [].concat(_toConsumableArray(fetchMoreResult.node.posts.edges), _toConsumableArray(previousResult.node.posts.edges))
              }
            });
            return Object.assign({}, previousResult, { node: node });
          }
        });
      }
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(TimelineContainer, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(props) {
      if (!props.data.loading) {
        this.props.hideLoading();
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _this2 = this;

      var posts = [];
      var endCursor = this.props.data.node.posts.pageInfo.endCursor;

      try {
        posts = this.props.data.node.posts.edges;
        posts = posts.map(function (post) {
          return post.node;
        });
      } catch (error) {
        console.log(error);
      }

      var btnStyle = {
        width: '100%',
        height: "50px",
        fontSize: "16px"
      };
      // console.log('this.props.data.loading : ', this.props.data.loading);
      var containerStyle = {
        marginBottom: "10px",
        marginTop: "10px",
        textAlign: "center"
      };
      return _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(_Timeline2.default, _extends({ posts: posts }, this.props)),
        _react2.default.createElement(_reactWaypoint2.default, { onEnter: function onEnter(props) {
            return _this2._handleEnter(props);
          } }),
        endCursor ? _react2.default.createElement(
          'button',
          { ref: 'abc', style: btnStyle, className: 'btn btn-outline-secondary',
            onClick: this.handleLoadMore },
          this.props.data.loading ? 'Loading' : 'Load More'
        ) : _react2.default.createElement(
          'div',
          { style: containerStyle },
          _react2.default.createElement(
            'div',
            { className: 'alert alert-info', role: 'alert' },
            _react2.default.createElement(
              'strong',
              null,
              'Heads up!'
            ),
            ' All posts have been loaded.'
          )
        )
      );
    }
  }]);

  return TimelineContainer;
}(_react2.default.Component);

TimelineContainer.propTypes = propTypes;
TimelineContainer.defaultProps = defaultProps;

exports.default = (0, _bluerainClientServices.withLoadingBar)(TimelineContainer);
