// bluerain-ui/lib/Controls/SliderControl.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _MuiThemeProvider = require('material-ui/styles/MuiThemeProvider');

var _MuiThemeProvider2 = _interopRequireDefault(_MuiThemeProvider);

var _Slider = require('material-ui/Slider');

var _Slider2 = _interopRequireDefault(_Slider);

var _Block = require('./Block/Block');

var _Block2 = _interopRequireDefault(_Block);

var _SliderItem = require('./SliderComponent/SliderItem');

var _SliderItem2 = _interopRequireDefault(_SliderItem);

var _SliderItemLabel = require('./SliderComponent/SliderItemLabel');

var _SliderItemLabel2 = _interopRequireDefault(_SliderItemLabel);

var _SliderItemValue = require('./SliderComponent/SliderItemValue');

var _SliderItemValue2 = _interopRequireDefault(_SliderItemValue);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  label: _react.PropTypes.string,
  className: _react.PropTypes.string,
  min: _react.PropTypes.number,
  max: _react.PropTypes.number,
  step: _react.PropTypes.number,
  value: _react.PropTypes.number,
  onChange: _react.PropTypes.func,
  onDragStart: _react.PropTypes.func,
  onDragStop: _react.PropTypes.func
};

var defaultProps = {
  label: '',
  className: '',
  min: 0,
  max: 100,
  step: 1,
  value: 50,
  onChange: function onChange() {},
  onDragStart: function onDragStart() {},
  onDragStop: function onDragStop() {}
};

var style = {
  control: {
    display: 'flex'
  },
  label: {
    flexGrow: 1
  },
  value: {
    paddingLeft: 10,
    fontWeight: 'bold'
  }
};

var SliderControl = function (_React$Component) {
  _inherits(SliderControl, _React$Component);

  function SliderControl(props) {
    _classCallCheck(this, SliderControl);

    var _this = _possibleConstructorReturn(this, (SliderControl.__proto__ || Object.getPrototypeOf(SliderControl)).call(this, props));

    _this.onChange = _this.onChange.bind(_this);
    _this.onDragStart = _this.onDragStart.bind(_this);
    _this.onDragStop = _this.onDragStop.bind(_this);

    _this.state = {
      value: props.value
    };
    return _this;
  }

  _createClass(SliderControl, [{
    key: 'onChange',
    value: function onChange(ev, value) {
      this.setState({
        value: value
      });

      this.props.onChange(value);
    }
  }, {
    key: 'onDragStart',
    value: function onDragStart() {
      this.props.onDragStart(this.state.value);
    }
  }, {
    key: 'onDragStop',
    value: function onDragStop() {
      this.props.onDragStop(this.state.value);
    }
  }, {
    key: 'render',
    value: function render() {
      var value = this.state.value;
      var _props = this.props,
          label = _props.label,
          min = _props.min,
          max = _props.max,
          step = _props.step,
          className = _props.className;


      return _react2.default.createElement(
        _Block2.default,
        { className: 'device-control-item slider-control ' + className },
        _react2.default.createElement(
          _SliderItem2.default,
          { style: style.control },
          _react2.default.createElement(
            _SliderItemLabel2.default,
            { className: 'control-label', style: style.label },
            label
          ),
          _react2.default.createElement(
            _SliderItemValue2.default,
            { className: 'control-value', style: style.value },
            value
          )
        ),
        _react2.default.createElement(
          _MuiThemeProvider2.default,
          null,
          _react2.default.createElement(_Slider2.default, {
            value: value,
            min: min,
            max: max,
            step: step,
            className: 'control-slider',
            onChange: this.onChange,
            onDragStart: this.onDragStart,
            onDragStop: this.onDragStop
          })
        )
      );
    }
  }]);

  return SliderControl;
}(_react2.default.Component);

exports.default = SliderControl;


SliderControl.propTypes = propTypes;
SliderControl.defaultProps = defaultProps;
