// bluerain-ui/lib/Controls/SwitchControl.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _MuiThemeProvider = require('material-ui/styles/MuiThemeProvider');

var _MuiThemeProvider2 = _interopRequireDefault(_MuiThemeProvider);

var _Toggle = require('material-ui/Toggle');

var _Toggle2 = _interopRequireDefault(_Toggle);

var _Block = require('./Block/Block');

var _Block2 = _interopRequireDefault(_Block);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propTypes = {
  label: _react.PropTypes.string,
  onToggle: _react.PropTypes.func,
  defaultValue: _react.PropTypes.bool,
  switchState: _react.PropTypes.bool
};

var defaultProps = {
  label: '',
  onToggle: function onToggle() {},
  defaultValue: false,
  switchState: false
};

var style = {
  label: {
    marginBottom: 0
  }
};

var SwitchControl = function SwitchControl() {
  return {
    render: function render() {
      var _props = this.props,
          label = _props.label,
          _onToggle = _props.onToggle,
          defaultValue = _props.defaultValue,
          className = _props.className,
          switchState = _props.switchState;

      return _react2.default.createElement(
        _Block2.default,
        { className: 'device-control-item switch-control ' + className },
        _react2.default.createElement(
          _MuiThemeProvider2.default,
          null,
          _react2.default.createElement(_Toggle2.default, {
            label: label,
            labelStyle: style.label,
            defaultToggled: defaultValue,
            onToggle: function onToggle(ev, toggle) {
              return _onToggle(toggle);
            },
            toggled: switchState
          })
        )
      );
    }
  };
};

SwitchControl.propTypes = propTypes;
SwitchControl.defaultProps = defaultProps;

exports.default = SwitchControl;
