// bluerain-app-device-explorer/lib/components/Asset/Charts/RefrigeratorCompressorTemperatureChart/CompressorTemperatureChart.container.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _CompressorTemperatureChart = require('./CompressorTemperatureChart.component');

var _CompressorTemperatureChart2 = _interopRequireDefault(_CompressorTemperatureChart);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Created by atif on 12/18/16.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


var propTypes = {
  data: _react.PropTypes.object,
  tooltip: _react.PropTypes.bool,
  startDate: _react.PropTypes.string,
  endDate: _react.PropTypes.string,
  label: _react.PropTypes.string,
  showGraph: _react.PropTypes.bool,
  linkTag: _react.PropTypes.oneOfType([_react2.default.PropTypes.func, _react2.default.PropTypes.string]),
  to: _react.PropTypes.string,
  color: _react.PropTypes.string
};

var defaultPropTypes = {
  startDate: '01/01/1970',
  endDate: '01/01/2099',
  tooltip: true,
  showGraph: true,
  label: 'Compressor shell Temperature',
  linkTag: 'a',
  to: '#'
};

var TemperatureChartContainer = function (_React$Component) {
  _inherits(TemperatureChartContainer, _React$Component);

  function TemperatureChartContainer() {
    _classCallCheck(this, TemperatureChartContainer);

    return _possibleConstructorReturn(this, (TemperatureChartContainer.__proto__ || Object.getPrototypeOf(TemperatureChartContainer)).apply(this, arguments));
  }

  _createClass(TemperatureChartContainer, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          color = _props.color,
          tooltip = _props.tooltip,
          dbData = _props.data,
          label = _props.label,
          linkTag = _props.linkTag,
          to = _props.to,
          showGraph = _props.showGraph,
          rest = _objectWithoutProperties(_props, ['color', 'tooltip', 'data', 'label', 'linkTag', 'to', 'showGraph']);

      var data = [];
      var lastReading = 'No data found';

      if (this.props.data === undefined || this.props.data.loading) {
        lastReading = 'Loading data';
      }

      try {

        var dataFromQuery = dbData.Event.EventFind.edges;
        dataFromQuery = dataFromQuery.map(function (item) {
          return item.node;
        });

        dataFromQuery.forEach(function (item) {
          if (item.data && item.data.temperature.t1 !== undefined) {
            data.push({
              time: new Date(item.time),
              temperature: (item.data.temperature.t1 / 1024 - 0.5) * 100
            });
          }
        });

        if (data.length > 0) {
          data.sort(function (a, b) {
            return a.time > b.time ? 1 : b.time > a.time ? -1 : 0;
          });
          var sum = data.reduce(function (a, b) {
            return a + b.temperature;
          }, 0);
          var average = sum / data.length;
          lastReading = average.toFixed(0) + ' \u2103';
        }
      } catch (error) {
        console.info('Temperature Chart no data found');
      }

      if (showGraph) {
        return _react2.default.createElement(_CompressorTemperatureChart2.default, _extends({
          data: data,
          x: 'time',
          y: function y(datum) {
            return datum.temperature;
          },
          color: this.props.color,
          label: label,
          value: lastReading,
          tooltip: this.props.tooltip,
          linkTag: linkTag,
          to: to
        }, rest));
      }
      return _react2.default.createElement(_CompressorTemperatureChart2.default, _extends({
        color: color,
        label: label,
        value: lastReading,
        tooltip: tooltip,
        linkTag: linkTag,
        to: to
      }, rest));
    }
  }]);

  return TemperatureChartContainer;
}(_react2.default.Component);

exports.default = TemperatureChartContainer;


TemperatureChartContainer.propTypes = propTypes;
TemperatureChartContainer.defaultProps = defaultPropTypes;
