// bluerain-app-device-explorer/src/components/Asset/AssetSensorInfo/SensorInfo/AssetSensorInfo.container.js
import React, { PropTypes } from 'react';
import Icon from 'react-fa';
import 'bluerain-bootstrap-theme/dist/css/components/data-list.css';
// import DataListComponent from '../DataList/DataList.component';

const propTypes = {
  data: PropTypes.object,
};

const defaultProps = {};

export default class AssetSensorInfo extends React.Component {

  static hasAnySensorData() {
    const {
      connected,
      ambientTemperature,
      current,
      power,
      voltage
    } = this.props.data.node.state;

    if (connected !== undefined && connected !== null) {
      return true;
    } else if (ambientTemperature !== undefined && ambientTemperature !== null) {
      return true;
    } else if (current !== undefined && current !== null) {
      return true;
    } else if (power !== undefined && power !== null) {
      return true;
    } else if (voltage !== undefined && voltage !== null) {
      return true;
    }

    return false;
  }

  static renderConnectionStatus(connected) {

    let status = null;

    if (typeof (connected) !== 'boolean') {
      return (<div />);
    } else if (connected) {
      status = (<span>Online <Icon name="circle text-success" /></span>);
    } else {
      status = (<span>Offline <Icon name="circle text-danger" /></span>);
    }
    return (
      <li className="list-group-item">
        <div className="data-label">Status</div>
        <div className="data-value">{status}</div>
      </li>
    );
  }

  static renderAmbientTemperature(ambientTemperature, connected) {

    if (typeof (ambientTemperature) !== 'number') {
      return (<div />);
    }
    if (connected) {
      return (
        <li className="list-group-item">
          <div className="data-label">Room Temperature</div>
          <div className="data-value">{ambientTemperature} ℃</div>
        </li>
      );
    }
    return (<div />);
  }

  static renderCurrent(current, connected, standby) {
    if (typeof (current) !== 'number') {
      return (<div />);
    }
    if (standby) {
      return (<div />);
    }
    if (connected) {
      return (
        <li className="list-group-item">
          <div className="data-label">Current</div>
          <div className="data-value">{current.toFixed(2)} A</div>
        </li>
      );
    }
    return (<div />);
  }

  static renderPower(power, connected, standby) {
    if (typeof (power) !== 'number') {
      return (<div />);
    }
    if (standby) {
      return (<div />);
    }
    if (connected) {
      return (
        <li className="list-group-item">
          <div className="data-label">Power</div>
          <div className="data-value">{power.toFixed(0)} W</div>
        </li>
      );
    }
    return (<div />);
  }

  static renderVoltage(voltage, connected, standby) {
    if (typeof (voltage) !== 'number') {
      return (<div />);
    }
    if (standby) {
      return (<div />);
    }
    if (connected) {
      return (
        <li className="list-group-item">
          <div className="data-label">Voltage</div>
          <div className="data-value">{voltage.toFixed(0)} V</div>
        </li>
      );
    }
    return (<div />);
  }

  render() {
    if (!this.props.data.node || !this.props.data.node.state) {
      return (<div />);
    }

    const {
      connected,
      // ambientTemperature,
      current,
      power,
      voltage,
      standby
    } = this.props.data.node.state;

    return (
      <ul className="list-group data-list">
        { AssetSensorInfo.renderConnectionStatus(connected) }
        { AssetSensorInfo.renderCurrent(current, connected, standby) }
        { AssetSensorInfo.renderPower(power, connected, standby) }
        { AssetSensorInfo.renderVoltage(voltage, connected, standby) }
      </ul>
    );
  }
}

AssetSensorInfo.propTypes = propTypes;
AssetSensorInfo.defaultProps = defaultProps;

// { this.renderAmbientTemperature(ambientTemperature, connected) }
