// bluerain-app-device-explorer/src/pages/SitePage/SiteForm/SiteFormPage.js
/**
 * Created by anser on 1/3/17.
 */
import React, { PropTypes } from 'react';
import { Container, Row, Col } from 'reactstrap';
import AppContent from 'bluerain-ui/lib/Layout/AppContent';
import PrimaryAppNav from 'bluerain-ui/lib/Nav/PrimaryAppNav';
import SiteFormWidget from '../../../components/Site/SiteForm/SiteForm.widget';

const propTypes = {
  title: PropTypes.string
};

const defaultProps = {
  title: 'New Site Form'
};

class SiteFormPage extends React.Component {

  renderPrimaryNav() {
    const title = this.props.title;
    const previous = {
      title: 'Sites',
      url: '/app/device-explorer'
    };
    return (
      <PrimaryAppNav title={title} previous={previous}>
        <i className="fa fa-plus hidden-sm-up" />
      </PrimaryAppNav>
      // <PrimaryAppNav title={title} />
    );
  }

  static renderSecondaryNav() {
  }

  static renderContent() {
    return (
      <AppContent>
        <Container>
          <Row>
            <Col md={{ size: 8, offset: 2 }} className="col-faded">
              <SiteFormWidget />
            </Col>
          </Row>
        </Container>
      </AppContent>
    );
  }

  render() {
    return (
      <div>
        {this.renderPrimaryNav()}
        {SiteFormPage.renderSecondaryNav()}
        {SiteFormPage.renderContent()}
      </div>
    );
  }
}


SiteFormPage.propTypes = propTypes;
SiteFormPage.defaultProps = defaultProps;

export default SiteFormPage;
