// bluerain-app-device-explorer/lib/components/Event/EventDetailPage/PostCardComponent/PostCard.component.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _PostCard = require('bluerain-ui/lib/Card/PostCard/PostCard.component');

var _PostCard2 = _interopRequireDefault(_PostCard);

require('bluerain-bootstrap-theme/dist/css/components/timeline.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Created by anser on 1/2/17.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


var propTypes = {
  data: _react.PropTypes.object
};

var defaultProps = {};

var data = void 0;

var PostCardComponent = function (_React$Component) {
  _inherits(PostCardComponent, _React$Component);

  function PostCardComponent() {
    _classCallCheck(this, PostCardComponent);

    return _possibleConstructorReturn(this, (PostCardComponent.__proto__ || Object.getPrototypeOf(PostCardComponent)).apply(this, arguments));
  }

  _createClass(PostCardComponent, [{
    key: 'render',
    value: function render() {
      try {
        if (this.props.data) {
          var post = this.props.data.node;
          if (post.type === 'USER_COMMAND') {
            var user = this.props.data.node.events.edges[0].node.account;
            data = {
              initiator: user.firstName && user.lastName ? user.firstName + ' ' + user.lastName : user.username,
              source: 'Web App',
              roundImage: true,
              image: user.image || 'https://s3-us-west-2.amazonaws.com/bluerainimages/default-user.jpg'
            };
          } else if (post.type === 'FIRMWARE_UPGRADE') {
            data = {
              initiator: '',
              source: 'Network',
              roundImage: true,
              image: 'https://s3-us-west-2.amazonaws.com/bluerainimages/firmware_update.jpg'
            };
          } else {
            data = {
              initiator: '',
              source: 'Remote',
              roundImage: true,
              image: 'https://s3-us-west-2.amazonaws.com/bluerainimages/remote.svg'
            };
          }
          return _react2.default.createElement(_PostCard2.default, {
            key: post.id,
            initiator: data.initiator,
            description: post.description,
            time: post.time,
            source: data.source,
            roundImage: data.roundImage,
            image: data.image,
            userProfileLink: data.userProfileLink
          });
        }
        return _react2.default.createElement('div', null);
      } catch (error) {
        console.log(error);
      }
    }
  }]);

  return PostCardComponent;
}(_react2.default.Component);

exports.default = PostCardComponent;


PostCardComponent.propTypes = propTypes;
PostCardComponent.defaultProps = defaultProps;
