// bluerain-app-device-explorer/lib/components/Report/BillReport/BillReport.widget.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

require('react-dates/lib/css/_datepicker.css');

var _widget = require('bluerain-ui/lib/Widget/widget');

var _widget2 = _interopRequireDefault(_widget);

var _widgetTitle = require('bluerain-ui/lib/Widget/widgetTitle');

var _widgetTitle2 = _interopRequireDefault(_widgetTitle);

var _widgetHeader = require('bluerain-ui/lib/Widget/widgetHeader');

var _widgetHeader2 = _interopRequireDefault(_widgetHeader);

var _reactDates = require('react-dates');

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

require('../../../css/custom_datepicker.css');

var _index = require('./index');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Created by atif on 12/18/16.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


var propTypes = {
  assetId: _react.PropTypes.string
};

var defaultProps = {
  assetId: ''
};

var BillReportWidget = function (_React$Component) {
  _inherits(BillReportWidget, _React$Component);

  function BillReportWidget(props) {
    _classCallCheck(this, BillReportWidget);

    var _this = _possibleConstructorReturn(this, (BillReportWidget.__proto__ || Object.getPrototypeOf(BillReportWidget)).call(this, props));

    if (localStorage && localStorage.end_date && localStorage.start_date) {
      _this.state = {
        focusedInput: null,
        startDate: (0, _moment2.default)(localStorage.start_date),
        endDate: (0, _moment2.default)(localStorage.end_date)
      };
    } else {
      _this.state = {
        focusedInput: null,
        startDate: (0, _moment2.default)().subtract(1, 'days'),
        endDate: (0, _moment2.default)()
      };
    }

    _this.onDatesChange = _this.onDatesChange.bind(_this);
    _this.onFocusChange = _this.onFocusChange.bind(_this);
    return _this;
  }

  _createClass(BillReportWidget, [{
    key: 'onDatesChange',
    value: function onDatesChange(_ref) {
      var startDate = _ref.startDate,
          endDate = _ref.endDate;

      localStorage.setItem('start_date', startDate);
      localStorage.setItem('end_date', endDate);
      this.setState({ startDate: startDate, endDate: endDate });
    }
  }, {
    key: 'onFocusChange',
    value: function onFocusChange(focusedInput) {
      this.setState({ focusedInput: focusedInput });
    }
  }, {
    key: 'render',
    value: function render() {
      var _state = this.state,
          focusedInput = _state.focusedInput,
          startDate = _state.startDate,
          endDate = _state.endDate;


      var assetId = this.props.assetId;

      var startDateFormatted = (0, _moment2.default)(startDate).toISOString();
      var endDateFormatted = (0, _moment2.default)(endDate).toISOString();

      return _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(
          _widget2.default,
          null,
          _react2.default.createElement(
            _widgetHeader2.default,
            null,
            _react2.default.createElement(
              'div',
              { className: 'widget-title-holder' },
              _react2.default.createElement(
                _widgetTitle2.default,
                null,
                'Bill'
              )
            ),
            _react2.default.createElement(_reactDates.DateRangePicker, _extends({}, this.props, {
              isOutsideRange: function isOutsideRange() {
                return false;
              },
              onDatesChange: this.onDatesChange,
              onFocusChange: this.onFocusChange,
              focusedInput: focusedInput,
              startDate: startDate,
              endDate: endDate
            }))
          )
        ),
        _react2.default.createElement('div', { className: 'clearfix' }),
        _react2.default.createElement(_index2.default, {
          assetId: assetId, startDate: startDateFormatted, endDate: endDateFormatted, queryString: '',
          color: 'blue'
        })
      );
    }
  }]);

  return BillReportWidget;
}(_react2.default.Component);

exports.default = BillReportWidget;


BillReportWidget.propTypes = propTypes;
BillReportWidget.defaultProps = defaultProps;
