// bluerain-app-device-explorer/src/routes/index.js
function errorLoading(error) {
  throw new Error(`Dynamic page loading failed: ${error}`);
}

function loadRoute(cb) {
  return module => cb(null, module.default);
}

function beforeLoadingRoute() {
  // <consol></consol>e.log('Before Loading Route');
  // store.dispatch(showLoading());
}

module.exports = {
  path: 'device-explorer',
  indexRoute: {
    getComponent(location, cb) {
      beforeLoadingRoute();
      System.import('../pages/IndexPage')
      .then(loadRoute(cb))
      .catch(errorLoading);
    }
  },
  childRoutes: [
    {
      path: 'fitbit/callback',
      indexRoute: {
        getComponent(location, cb) {
          beforeLoadingRoute();
          System.import('../pages/CallBack/FitbitAuthHandler')
          .then(loadRoute(cb))
          .catch(errorLoading);
        }
      }
    },
    {
      path: 'sites/:siteId',
      indexRoute: {
        getComponent(location, cb) {
          beforeLoadingRoute();
          System.import('../pages/SitePage')
          .then(loadRoute(cb))
          .catch(errorLoading);
        }
      },
      childRoutes: [{
        path: 'add-device',
        indexRoute: {
          getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/Assets/AddAsset')
            .then(loadRoute(cb))
            .catch(errorLoading);
          }
        }
      },{
        path: 'edit',
        indexRoute: {
          getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/SitePage/SiteEditForm/SiteEditFormPage')
            .then(loadRoute(cb))
            .catch(errorLoading);
          }
        }
      }]
    },
    {
      path: 'site/new',
      indexRoute: {
        getComponent(location, cb) {
          beforeLoadingRoute();
          System.import('../pages/SitePage/SiteForm/SiteFormPage')
          .then(loadRoute(cb))
          .catch(errorLoading);
        }
      },
    },
    {
      path: 'asset_locations',
      indexRoute: {
        getComponent(location, cb) {
          beforeLoadingRoute();
          System.import('../pages/Assets/Location/AssetLocationsPage')
          .then(loadRoute(cb))
          .catch(errorLoading);
        }
      },
    },
    {
      path: 'transfer_requests',
      indexRoute: {
        getComponent(location, cb) {
          beforeLoadingRoute();
          System.import('../pages/TransferRequests/TransferRequestsPage')
          .then(loadRoute(cb))
          .catch(errorLoading);
        }
      },
    },
    {
      path: 'assets/:assetId',
      getComponent(nextState, cb) {
        beforeLoadingRoute();
        System.import('../pages/Assets/Single')
        .then(loadRoute(cb))
        .catch(errorLoading);
      },
      indexRoute: {
        getComponent(nextState, cb) {
          beforeLoadingRoute();
          System.import('../pages/Assets/Single/Overview')
          .then(loadRoute(cb))
          .catch(errorLoading);
        }
      },
      childRoutes: [
        {
          path: 'reports',
          getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/Assets/Single/Reports')
            .then(loadRoute(cb))
            .catch(errorLoading);
          },
          childRoutes: [
            {
              path: 'bill',
              indexRoute: {
                getComponent(location, cb) {
                  beforeLoadingRoute();
                  System.import('../pages/Assets/Single/Reports/Bill')
                  .then(loadRoute(cb))
                  .catch(errorLoading);
                }
              }
            },
            {
              path: 'current',
              indexRoute: {
                getComponent(location, cb) {
                  beforeLoadingRoute();
                  System.import('../pages/Assets/Single/Reports/Current')
                  .then(loadRoute(cb))
                  .catch(errorLoading);
                }
              }
            },
            {
              path: 'energy-saving',
              indexRoute: {
                getComponent(location, cb) {
                  beforeLoadingRoute();
                  System.import('../pages/Assets/Single/Reports/EnergySaving')
                  .then(loadRoute(cb))
                  .catch(errorLoading);
                }
              }
            },
            {
              path: 'energy',
              indexRoute: {
                getComponent(location, cb) {
                  beforeLoadingRoute();
                  System.import('../pages/Assets/Single/Reports/Energy')
                  .then(loadRoute(cb))
                  .catch(errorLoading);
                }
              }
            },
            {
              path: 'power',
              indexRoute: {
                getComponent(location, cb) {
                  beforeLoadingRoute();
                  System.import('../pages/Assets/Single/Reports/Power')
                  .then(loadRoute(cb))
                  .catch(errorLoading);
                }
              }
            },
            {
              path: 'temperature',
              indexRoute: {
                getComponent(location, cb) {
                  beforeLoadingRoute();
                  System.import('../pages/Assets/Single/Reports/Temperature')
                  .then(loadRoute(cb))
                  .catch(errorLoading);
                }
              }
            },
            {
              path: 'ref_temperature',
              indexRoute: {
                getComponent(location, cb) {
                  beforeLoadingRoute();
                  System.import('../pages/Assets/Single/Reports/RefrigeratorTemperature')
                  .then(loadRoute(cb))
                  .catch(errorLoading);
                }
              }
            },
            {
              path: 'time',
              indexRoute: {
                getComponent(location, cb) {
                  beforeLoadingRoute();
                  System.import('../pages/Assets/Single/Reports/Time')
                  .then(loadRoute(cb))
                  .catch(errorLoading);
                }
              }
            },
            {
              path: 'ref_time',
              indexRoute: {
                getComponent(location, cb) {
                  beforeLoadingRoute();
                  System.import('../pages/Assets/Single/Reports/RefrigeratorTime')
                  .then(loadRoute(cb))
                  .catch(errorLoading);
                }
              }
            },
            {
              path: 'voltage',
              indexRoute: {
                getComponent(location, cb) {
                  beforeLoadingRoute();
                  System.import('../pages/Assets/Single/Reports/Voltage')
                  .then(loadRoute(cb))
                  .catch(errorLoading);
                }
              }
            }
          ]
        },
        {
          path: 'posts/:eventId',
          getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/EventDetail/EventDetailPage')
            .then(loadRoute(cb))
            .catch(errorLoading);
          }
        },
        {
          path: 'device-info',
          getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/Device/DeviceInfoPage')
            .then(loadRoute(cb))
            .catch(errorLoading);
          }
        },
        {
          path: 'device-network-settings',
          getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/Device/DeviceNetworkSettingsPage')
            .then(loadRoute(cb))
            .catch(errorLoading);
          }
        },
        {
          path: 'device-general-settings',
          getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/Device/DeviceGeneralSettingsPage')
            .then(loadRoute(cb))
            .catch(errorLoading);
          }
        },
        {
          path: 'ownership-transfer/:userId',
          getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/Device/OwnershipTransferUserPage')
            .then(loadRoute(cb))
            .catch(errorLoading);
          }
        }
      ]
    }]
};
