// bluerain-app-device-explorer/src/routes/urlBuilder.js
export function sitePageUrl(siteId) {
  return `/app/device-explorer/sites/${siteId}`;
}

export function assetPageUrl(assetId) {
  return `/app/device-explorer/assets/${assetId}`;
}

export function assetReportPageUrl(assetId, report) {

  if (!report) {
    // return `/app/device-explorer/assets/${assetId}/reports/temperature`;
    return `/app/device-explorer/assets/${assetId}/reports/current`;
  }

  return `/app/device-explorer/assets/${assetId}/reports/${report}`;
}

export function refAssetReportPageUrl(assetId, report) {

  if (!report) {
    // return `/app/device-explorer/assets/${assetId}/reports/ref_temperature`;
    return `/app/device-explorer/assets/${assetId}/reports/ref_time`;
  }

  return `/app/device-explorer/assets/${assetId}/reports/${report}`;
}

export function deviceSettingsPageUrl(assetId, settingsPath) {
  // if (settingsPath === 'device-general-settings' || settingsPath === 'device-network-settings' || settingsPath === 'device-info') {
  //   return `/app/device-explorer/assets/${assetId}/${settingsPath}`;
  // }
  return `/app/device-explorer/assets/${assetId}/device-general-settings`;
}


export function sitesGridViewUrl() {
  return '/app/device-explorer';
}

export function sitesMapViewUrl() {
  return '/app/device-explorer/asset_locations';
}
