// bluerain-ui/lib/BarChartCard/BarChartCard.component.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

require('bluerain-bootstrap-theme/dist/css/components/chart-card.css');

var _CardBlock = require('reactstrap/lib/CardBlock');

var _CardBlock2 = _interopRequireDefault(_CardBlock);

var _CustomCardLink = require('../Card/CustomCardLink');

var _CustomCardLink2 = _interopRequireDefault(_CustomCardLink);

var _ChartCard = require('../Card/ChartCard/ChartCard');

var _ChartCard2 = _interopRequireDefault(_ChartCard);

var _BarChart = require('../BarChart/BarChart.component');

var _BarChart2 = _interopRequireDefault(_BarChart);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Created by anser on 5/22/17.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */

var propTypes = {
  label: _react.PropTypes.string,
  value: _react.PropTypes.string,
  unit: _react.PropTypes.string,
  data: _react.PropTypes.array,
  color: _react.PropTypes.string,
  linkTag: _react.PropTypes.oneOfType([_react.PropTypes.func, _react.PropTypes.string]),
  to: _react.PropTypes.string,
  href: _react.PropTypes.string,
  onClick: _react.PropTypes.func.isRequired,
  x: _react.PropTypes.string,
  y: _react.PropTypes.func,
  tooltip: _react.PropTypes.bool,
  tooltipIsBoolean: _react.PropTypes.bool,
  tooltipBooleanTrueLabel: _react.PropTypes.string,
  tooltipBooleanFalseLabel: _react.PropTypes.string,
  interpolation: _react.PropTypes.string
};

var defaultProps = {
  label: '',
  value: '',
  unit: '',
  data: [],
  color: 'pink',
  linkTag: 'a',
  to: '',
  href: '#',
  onClick: function onClick() {},
  tooltip: undefined,
  interpolation: 'step',
  tooltipIsBoolean: false,
  tooltipBooleanTrueLabel: 'Boolean On',
  tooltipBooleanFalseLabel: 'Boolean False'
};

var _ = require('lodash');

var BarChartCard = function (_React$Component) {
  _inherits(BarChartCard, _React$Component);

  _createClass(BarChartCard, null, [{
    key: 'drawChart',
    value: function drawChart(data, x, y, tooltip, unit, interpolation, tooltipIsBoolean, tooltipBooleanTrueLabel, tooltipBooleanFalseLabel) {
      if (!_.isEmpty(data)) {
        return _react2.default.createElement(_BarChart2.default, {
          data: data,
          x: x,
          y: y,
          tooltip: tooltip,
          unit: unit,
          interpolation: interpolation,
          tooltipIsBoolean: tooltipIsBoolean,
          tooltipBooleanTrueLabel: tooltipBooleanTrueLabel,
          tooltipBooleanFalseLabel: tooltipBooleanFalseLabel
        });
      }
    }
  }, {
    key: 'drawLabel',
    value: function drawLabel(label) {
      if (label) {
        return _react2.default.createElement(
          'div',
          { className: 'label' },
          label
        );
      }
    }
  }, {
    key: 'drawValue',
    value: function drawValue(value) {
      if (value) {
        return _react2.default.createElement(
          'div',
          { className: 'value' },
          value
        );
      }
    }
  }]);

  function BarChartCard(props) {
    _classCallCheck(this, BarChartCard);

    var _this = _possibleConstructorReturn(this, (BarChartCard.__proto__ || Object.getPrototypeOf(BarChartCard)).call(this, props));

    _this.handleClickAction = _this.handleClickAction.bind(_this);
    return _this;
  }

  _createClass(BarChartCard, [{
    key: 'handleClickAction',
    value: function handleClickAction() {
      this.props.onClick();
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          data = _props.data,
          x = _props.x,
          y = _props.y,
          linkTag = _props.linkTag,
          to = _props.to,
          href = _props.href,
          label = _props.label,
          value = _props.value,
          tooltip = _props.tooltip,
          unit = _props.unit,
          interpolation = _props.interpolation,
          tooltipIsBoolean = _props.tooltipIsBoolean,
          tooltipBooleanTrueLabel = _props.tooltipBooleanTrueLabel,
          tooltipBooleanFalseLabel = _props.tooltipBooleanFalseLabel,
          props = _objectWithoutProperties(_props, ['data', 'x', 'y', 'linkTag', 'to', 'href', 'label', 'value', 'tooltip', 'unit', 'interpolation', 'tooltipIsBoolean', 'tooltipBooleanTrueLabel', 'tooltipBooleanFalseLabel']);

      return _react2.default.createElement(
        _ChartCard2.default,
        props,
        _react2.default.createElement(
          _CustomCardLink2.default,
          {
            tag: linkTag,
            to: to,
            href: href
          },
          _react2.default.createElement(
            _CardBlock2.default,
            null,
            BarChartCard.drawLabel(label),
            BarChartCard.drawValue(value)
          ),
          BarChartCard.drawChart(data, x, y, tooltip, unit, interpolation, tooltipIsBoolean, tooltipBooleanTrueLabel, tooltipBooleanFalseLabel)
        )
      );
    }
  }]);

  return BarChartCard;
}(_react2.default.Component);

exports.default = BarChartCard;


BarChartCard.propTypes = propTypes;
BarChartCard.defaultProps = defaultProps;
