// bluerain-app-device-explorer/lib/components/Report/EnergySavingReport/EnergySavingReport.container.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _EnergySavingReport = require('./EnergySavingReport.component');

var _EnergySavingReport2 = _interopRequireDefault(_EnergySavingReport);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  data: _react.PropTypes.object,
  tooltip: _react.PropTypes.bool,
  startDate: _react.PropTypes.string,
  endDate: _react.PropTypes.string,
  color: _react.PropTypes.string
};

var defaultPropTypes = {
  startDate: '01/01/1970',
  endDate: '01/01/2099',
  tooltip: true,
  color: 'yellow'
};

var EnergySavingReportContainer = function (_React$Component) {
  _inherits(EnergySavingReportContainer, _React$Component);

  function EnergySavingReportContainer() {
    _classCallCheck(this, EnergySavingReportContainer);

    return _possibleConstructorReturn(this, (EnergySavingReportContainer.__proto__ || Object.getPrototypeOf(EnergySavingReportContainer)).apply(this, arguments));
  }

  _createClass(EnergySavingReportContainer, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          color = _props.color,
          tooltip = _props.tooltip,
          dbData = _props.data,
          rest = _objectWithoutProperties(_props, ['color', 'tooltip', 'data']);

      var data = [];

      try {
        var dataFromQuery = dbData.Event.EventFind.edges;
        var maxMetaValue = dbData.node.meta.max;
        dataFromQuery = dataFromQuery.map(function (item) {
          return item.node;
        });
        dataFromQuery.forEach(function (item) {
          if (item.data && item.data.current !== undefined) {
            var currentLatest = item.data.current;
            var effPercent = (maxMetaValue - currentLatest) / maxMetaValue * 100;
            if (effPercent < 0) {
              effPercent = 0;
            } else if (effPercent > 100) {
              effPercent = 100;
            }
            data.push({
              time: new Date(item.time),
              current: effPercent
            });
          }
        });

        if (data.length > 0) {
          data.sort(function (a, b) {
            return a.time > b.time ? 1 : b.time > a.time ? -1 : 0;
          });
        }
      } catch (error) {
        console.info('Energy Saving Report: no data found');
      }
      return _react2.default.createElement(_EnergySavingReport2.default, _extends({
        data: data,
        x: 'time',
        y: function y(datum) {
          return datum.current;
        },
        tooltip: tooltip,
        color: color
      }, rest));
    }
  }]);

  return EnergySavingReportContainer;
}(_react2.default.Component);

exports.default = EnergySavingReportContainer;


EnergySavingReportContainer.propTypes = propTypes;
EnergySavingReportContainer.defaultProps = defaultPropTypes;
