// bluerain-app-device-explorer/src/components/Device/AddDevice/ThirdScreen/AddDeviceThirdScreen.component.js
import React, { PropTypes } from 'react';
import ListViewItem from 'bluerain-ui/lib/ListView/ListViewItem';

const propTypes = {
  ssId: PropTypes.string,
  onSubmit: PropTypes.func
};

class AddDeviceThirdScreen extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      password: ''
    };

    this.handleChangeD = this.handleChangeD.bind(this);
    this.submitEventHandler = this.submitEventHandler.bind(this);
  }

  // Get value of password
  handleChangeD(event) {
    this.setState({
      password: event.target.value
    });
  }

  // Form on submit
  submitEventHandler(event) {
    event.preventDefault();
    if (this.onSubmit) {
      this.onSubmit(event, {
        ssid: this.props.ssId,
        password: this.state.password
      });
    }
  }


  render() {

    const Prompt = {

      fontSize: 14,
      textAlign: 'left',
      color: 'red'

    };
    this.onSubmit = this.props.onSubmit;
    return (
      <form onSubmit={this.submitEventHandler}>

        <ListViewItem>
          <label htmlFor="ssid">SSID</label>
          <input
            id="ssid" disabled type="text" value={this.props.ssId} className="form-control" aria-describedby="id"
            placeholder="SSID"
          />
        </ListViewItem>

        <ListViewItem>
          <label htmlFor="password">Password</label>
          <input
            id="password" type="password" value={this.state.password} onChange={this.handleChangeD}
            className="form-control" aria-describedby="name" placeholder="Password"
          />
        </ListViewItem>

        <ListViewItem className="text-xs-right">
          <p style={Prompt}>*Please enter your WIFI password and press submit</p>
          <button type="submit" className="btn btn-primary">Submit</button>
        </ListViewItem>
      </form>
    );
  }
}

AddDeviceThirdScreen.propTypes = propTypes;

export default AddDeviceThirdScreen;
