// bluerain-app-device-explorer/lib/components/Asset/Charts/FridgeOpenTimeChart/FridgeOpenTimeChart.container.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _FridgeOpenTimeChart = require('./FridgeOpenTimeChart.component');

var _FridgeOpenTimeChart2 = _interopRequireDefault(_FridgeOpenTimeChart);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Created by anser on 3/7/16.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


var propTypes = {
  data: _react.PropTypes.object,
  tooltip: _react.PropTypes.bool,
  startDate: _react.PropTypes.string,
  endDate: _react.PropTypes.string,
  label: _react.PropTypes.string,
  showGraph: _react.PropTypes.bool,
  linkTag: _react.PropTypes.oneOfType([_react2.default.PropTypes.func, _react2.default.PropTypes.string]),
  to: _react.PropTypes.string,
  color: _react.PropTypes.string
};

var defaultPropTypes = {
  startDate: '01/01/1970',
  endDate: '01/01/2099',
  tooltip: true,
  showGraph: true,
  linkTag: 'a',
  to: '#'
};

var data = [];
var i = void 0;
var diffArr = [];
var prevStatus = 1;
var roundedValue = void 0;
var sum = void 0;
var sumInMints = void 0;
var sumInHours = void 0;
var diff = void 0;
var doorOpenCount = 0;
var label = void 0;

var FridgeOpenTimeChartContainer = function (_React$Component) {
  _inherits(FridgeOpenTimeChartContainer, _React$Component);

  function FridgeOpenTimeChartContainer() {
    _classCallCheck(this, FridgeOpenTimeChartContainer);

    return _possibleConstructorReturn(this, (FridgeOpenTimeChartContainer.__proto__ || Object.getPrototypeOf(FridgeOpenTimeChartContainer)).apply(this, arguments));
  }

  _createClass(FridgeOpenTimeChartContainer, [{
    key: 'renderTimeDiff',
    value: function renderTimeDiff(data, i) {
      diffArr = [];
      data.reverse(); // to get data in descending order
      for (i = 1; i < data.length; i++) {
        prevStatus = data[i - 1].data.door; // get previous value of door status
        if (prevStatus === 1) {
          // only get values for door false as previous value
          var time1 = (0, _moment2.default)(new Date(data[i].time));
          var time2 = (0, _moment2.default)(new Date(data[i - 1].time));
          diff = time1.diff(time2, 'seconds'); // difference of two consecutive values in seconds
          if (diffArr.indexOf(diff) < 0) diffArr.push(diff); // push in array while avoiding duplicate entries
        }
      }
      doorOpenCount = 0;
      for (i = 0; i < data.length; i++) {
        prevStatus = data[i].data.door; // get previous value of door status
        if (prevStatus === 1) {
          // only get values for door true as previous value
          doorOpenCount++;
        }
      }

      sum = diffArr.reduce(function (a, b) {
        return a + b;
      }); // get sum of values in array
      sumInHours = sum / 3600; // convert seconds to hours
      sum %= 3600;
      sumInMints = sum / 60; // convert seconds to minutes
      if (sumInMints > 0 && sumInMints < 1 && sumInHours < 1) {
        sumInMints = 1;
        // roundedValue = 'Less Than 1 m';
        roundedValue = ' ' + Math.round(sum) + ' s';
      } else {
        roundedValue = Math.floor(sumInHours) + ' h' + (' ' + Math.round(sumInMints) + ' m');
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          color = _props.color,
          tooltip = _props.tooltip,
          dbData = _props.data,
          linkTag = _props.linkTag,
          to = _props.to,
          showGraph = _props.showGraph,
          rest = _objectWithoutProperties(_props, ['color', 'tooltip', 'data', 'linkTag', 'to', 'showGraph']);

      var lastReading = 'No data found';

      if (this.props.data === undefined || this.props.data.loading) {
        lastReading = 'Loading data';
      }

      try {

        var dataFromQuery = dbData.Event.EventFind.edges;
        if (dataFromQuery.length === 0) {
          data = [];
          label = 'Door Open Time';
        } else {
          dataFromQuery = dataFromQuery.map(function (item) {
            return item.node;
          });
          data = [];
          label = 'Door Open Time';
          {
            this.renderTimeDiff(dataFromQuery, i);
          }
          dataFromQuery.forEach(function (item, i) {
            if (item.data && item.data.door !== undefined) {
              data.push({
                time: new Date(item.time),
                door: item.data.door ? 0 : 1
              });
            }
          });
          if (data.length > 0) {
            data.sort(function (a, b) {
              return a.time > b.time ? 1 : b.time > a.time ? -1 : 0;
            });
            // lastReading = `${data[data.length - 1].standby ? 'On' : 'Off'}`;
            lastReading = roundedValue;
            label = 'Door Opened ' + doorOpenCount + ' Times';
          }
        }
      } catch (error) {}
      // console.info('Time Chart no data found');

      // lastReading = roundedValue;
      if (showGraph) {
        return _react2.default.createElement(_FridgeOpenTimeChart2.default, _extends({
          data: data,
          x: 'time',
          y: function y(datum) {
            return datum.door;
          },
          color: this.props.color,
          label: label,
          value: lastReading,
          tooltip: this.props.tooltip,
          tooltipBooleanTrueLabel: 'Open',
          tooltipBooleanFalseLabel: 'Close',
          linkTag: linkTag,
          to: to
        }, rest));
      }
      return _react2.default.createElement(_FridgeOpenTimeChart2.default, _extends({
        color: color,
        label: label,
        value: lastReading,
        tooltip: tooltip,
        linkTag: linkTag,
        to: to
      }, rest));
    }
  }]);

  return FridgeOpenTimeChartContainer;
}(_react2.default.Component);

exports.default = FridgeOpenTimeChartContainer;


FridgeOpenTimeChartContainer.propTypes = propTypes;
FridgeOpenTimeChartContainer.defaultProps = defaultPropTypes;
