// bluerain-app-device-explorer/lib/components/Asset/Charts/FridgeOpenTimeChart/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _reactApollo = require('react-apollo');

var _graphqlRelay = require('graphql-relay');

var _FridgeOpenTimeChart = require('./FridgeOpenTimeChart.container');

var _FridgeOpenTimeChart2 = _interopRequireDefault(_FridgeOpenTimeChart);

var _query = require('./query.graphql');

var _query2 = _interopRequireDefault(_query);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// import FridgeOpenTimeChartStateful from './FridgeOpenTimeChart.stateful';
/**
 * Created by anser on 3/7/16.
 */
exports.default = (0, _reactApollo.graphql)(_query2.default, {
  options: function options(args) {
    var startDateFormatted = args.startDate;
    var endDateFormatted = args.endDate;

    var queryString = {
      where: {
        type: 'state',
        assetId: (0, _graphqlRelay.fromGlobalId)(args.assetId).id,
        time: { between: [startDateFormatted, endDateFormatted] },
        'data.door': { exists: true }
      },
      order: 'time DESC'
    };

    // `[{\"condition\":\"exists\",\"key\":\"data.door\"},{\"condition\":\"descending\",\"key\":\"time\"},{\"condition\":\"lessThanOrEqualTo\",\"key\":\"time\",\"value\":{\"__type\":\"Date\",
    // \"iso\":\"${endDateFormatted}\"}}, {\"condition\":\"greaterThanOrEqualTo\",\"key\":\"time\",\"value\":{\"__type\":\"Date\",\"iso\":\"${startDateFormatted}\"}}]`;

    return { variables: { assetId: args.assetId, queryString: queryString } };
  }
})(_FridgeOpenTimeChart2.default);
