// bluerain-app-device-explorer/lib/pages/Assets/AddAsset/Recheck.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

require('bluerain-bootstrap-theme/dist/css/components/data-list.css');

var _reactProgressLabel = require('react-progress-label');

var _reactProgressLabel2 = _interopRequireDefault(_reactProgressLabel);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  jumpToStep: _react.PropTypes.func

};

var Recheck = function (_React$Component) {
  _inherits(Recheck, _React$Component);

  function Recheck(props) {
    _classCallCheck(this, Recheck);

    var _this = _possibleConstructorReturn(this, (Recheck.__proto__ || Object.getPrototypeOf(Recheck)).call(this));

    _this.state = {
      text: 'Establishing Connection',
      progress: 1
    };
    return _this;
  }

  _createClass(Recheck, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      var _this2 = this;

      this.interval = setInterval(function () {
        var progress = _this2.state.progress;

        if (progress === 100) {
          clearInterval(_this2.interval);
          _this2.props.jumpToStep(4);
          _this2.setState({ dymmyvalue: true });
        }

        progress++;
        _this2.setState({ progress: progress });
      }, 200);
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {

      if (this.interval) {
        clearInterval(this.interval);
      }
    }
  }, {
    key: 'isValidated',
    value: function isValidated() {
      if (this.state.dymmyvalue === true) {

        return true;
      }
      return false;
    }
  }, {
    key: 'renderToolbar',
    value: function renderToolbar() {
      return _react2.default.createElement(
        Ons.Toolbar,
        null,
        _react2.default.createElement(
          'div',
          { className: 'center' },
          'this.state.Progress'
        )
      );
    }
  }, {
    key: 'render',
    value: function render() {
      var style1 = {
        padding: 10,
        margin: 10,
        backgroundColor: '#ececec',
        color: '#333',
        fontfamily: 'monospace',
        fontSize: 32,
        textAlign: 'center',
        textColor: '#23a3ba'
      };
      return _react2.default.createElement(
        'div',
        { style: style1 },
        _react2.default.createElement(
          'p',
          null,
          this.state.text
        ),
        _react2.default.createElement(
          _reactProgressLabel2.default,
          {
            progress: this.state.progress,
            startDegree: 0,
            progressWidth: 50,
            trackWidth: 50,
            cornersWidth: 12,
            size: 200,
            fillColor: '#f3f3f3',
            trackColor: '#23a3ba',
            progressColor: '#bff3fc'
          },
          _react2.default.createElement(
            'text',
            null,
            this.state.progress
          )
        )
      );
    }
  }]);

  return Recheck;
}(_react2.default.Component);

exports.default = Recheck;


Recheck.propTypes = propTypes;
