// bluerain-app-device-explorer/src/components/Event/EventDetailPage/EventDetailPage.widget.js
/**
 * Created by anser on 12/30/16.
 */
import React, { PropTypes } from 'react';
import { Icon } from 'react-fa';
import Widget from 'bluerain-ui/lib/Widget/widget';
import WidgetHeader from 'bluerain-ui/lib/Widget/widgetHeader';
import WidgetTitle from 'bluerain-ui/lib/Widget/widgetTitle';
import WidgetContent from 'bluerain-ui/lib/Widget/widgetContent';
import EventDetail from './EventDetailContainer/EventDetail.container';
import EventParameter from './EventParameterContainer/EventParameter.container';
import Device from './DeviceContainer/Device.container';
import PostCard from './PostCardComponent/PostCard.component';

const propTypes = {
  data: PropTypes.object,
};

const defaultProps = {};
export default class EventDetailWidget extends React.Component {
  static renderDeviceWidget(data) {
    if (data.node.sourceDevice === null) {
      return (<div />);
    }
    return (
      <Widget>
        <WidgetHeader>
          <WidgetTitle>Device</WidgetTitle>
        </WidgetHeader>
        <WidgetContent>
          <Device data={data} />
        </WidgetContent>
      </Widget>
    );
  }

  static renderParameterWidget(data) {
    return (
      <Widget>
        <WidgetHeader>
          <WidgetTitle>Event Parameters</WidgetTitle>
        </WidgetHeader>
        <WidgetContent>
          <EventParameter data={data} />
        </WidgetContent>
      </Widget>
    );
  }

  static renderDetailWidget(data) {
    return (
      <Widget>
        <WidgetHeader>
          <WidgetTitle>Event Details</WidgetTitle>
        </WidgetHeader>
        <WidgetContent>
          <EventDetail data={data} />
        </WidgetContent>
      </Widget>
    );
  }

  static renderPostCardWidget(data) {
    if (data.node.events.edges[0].node.asset === null) {
      return (<div />);
    }
    return (
      <Widget>
        <WidgetHeader>
          <WidgetTitle>{data.node.events.edges[0].node.asset.name}</WidgetTitle>
        </WidgetHeader>
        <WidgetContent>
          <PostCard data={data} />
        </WidgetContent>
      </Widget>
    );
  }

  render() {
    if (this.props.data.node) {
      const event = this.props.data.node.events.edges[0];
      if (event === null || event === undefined) { // exception handling
        return (
          <div className="app-content">
            <div className="container">
              <div className="v-align">
                <div className="v-align-middle empty-block">
                  <Icon name="home fa-5x" />
                  <h4>No events found for this post</h4>
                </div>
              </div>
            </div>
          </div>
        );
      }
      return (
        <div>
          { EventDetailWidget.renderPostCardWidget(this.props.data) }
          { EventDetailWidget.renderDetailWidget(event) }
          { EventDetailWidget.renderParameterWidget(event) }
          { EventDetailWidget.renderDeviceWidget(event) }
        </div>
      );
    }
    return (<div />);
  }
}

EventDetailWidget.propTypes = propTypes;
EventDetailWidget.defaultProps = defaultProps;
