// bluerain-app-device-explorer/lib/components/Asset/Charts/RefrigeratorInsideTemperatureChart/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _reactApollo = require('react-apollo');

var _graphqlRelay = require('graphql-relay');

var _query = require('./query.graphql');

var _query2 = _interopRequireDefault(_query);

var _InsideRefTemperatureChart = require('./InsideRefTemperatureChart.container');

var _InsideRefTemperatureChart2 = _interopRequireDefault(_InsideRefTemperatureChart);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = (0, _reactApollo.graphql)(_query2.default, {
  options: function options(args) {
    var startDateFormatted = args.startDate;
    var endDateFormatted = args.endDate;

    var queryString = {
      where: {
        type: 'state',
        assetId: (0, _graphqlRelay.fromGlobalId)(args.assetId).id,
        time: { between: [startDateFormatted, endDateFormatted] },
        'data.temperature': { exists: true }
      },
      order: 'time DESC'
    };

    return { variables: { assetId: args.assetId, queryString: queryString } };
  }
})(_InsideRefTemperatureChart2.default);
// import TemperatureChartStateful from './TemperatureChart.stateful';
