// bluerain-app-device-explorer/lib/components/Asset/SonosSensorInfo/SonosSensorInfo.container.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactFa = require('react-fa');

var _reactFa2 = _interopRequireDefault(_reactFa);

require('bluerain-bootstrap-theme/dist/css/components/data-list.css');

require('bluerain-bootstrap-theme/dist/css/components/card-view.css');

var _CardBlock = require('reactstrap/lib/CardBlock');

var _CardBlock2 = _interopRequireDefault(_CardBlock);

var _CardViewImg = require('./CardViewImg');

var _CardViewImg2 = _interopRequireDefault(_CardViewImg);

var _CardView = require('./CardView');

var _CardView2 = _interopRequireDefault(_CardView);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  data: _react.PropTypes.object
};

var defaultProps = {};

var SonosSensorInfo = function (_React$Component) {
  _inherits(SonosSensorInfo, _React$Component);

  function SonosSensorInfo() {
    _classCallCheck(this, SonosSensorInfo);

    return _possibleConstructorReturn(this, (SonosSensorInfo.__proto__ || Object.getPrototypeOf(SonosSensorInfo)).apply(this, arguments));
  }

  _createClass(SonosSensorInfo, [{
    key: 'render',
    value: function render() {
      if (!this.props.data.node || !this.props.data.node.state) {
        return _react2.default.createElement('div', null);
      }

      var _props$data$node$stat = this.props.data.node.state,
          connected = _props$data$node$stat.connected,
          song = _props$data$node$stat.song,
          album = _props$data$node$stat.album,
          artist = _props$data$node$stat.artist,
          image = _props$data$node$stat.image;


      return _react2.default.createElement(
        _CardView2.default,
        null,
        SonosSensorInfo.renderImage(image, connected),
        _react2.default.createElement(
          _CardBlock2.default,
          null,
          _react2.default.createElement(
            'ul',
            { className: 'list-group data-list' },
            SonosSensorInfo.renderConnectionStatus(connected),
            SonosSensorInfo.renderSongName(song, connected),
            SonosSensorInfo.renderAlbumName(album, connected),
            SonosSensorInfo.renderArtistName(artist, connected)
          )
        )
      );
    }
  }], [{
    key: 'renderConnectionStatus',
    value: function renderConnectionStatus(connected) {

      var status = null;

      if (typeof connected !== 'boolean') {
        return _react2.default.createElement('div', null);
      } else if (connected) {
        status = _react2.default.createElement(
          'span',
          null,
          'Online ',
          _react2.default.createElement(_reactFa2.default, { name: 'circle text-success' })
        );
      } else {
        status = _react2.default.createElement(
          'span',
          null,
          'Offline ',
          _react2.default.createElement(_reactFa2.default, { name: 'circle text-danger' })
        );
      }
      return _react2.default.createElement(
        'li',
        { className: 'list-group-item' },
        _react2.default.createElement(
          'div',
          { className: 'data-label' },
          'Status'
        ),
        _react2.default.createElement(
          'div',
          { className: 'data-value' },
          status
        )
      );
    }
  }, {
    key: 'renderSongName',
    value: function renderSongName(song, connected) {
      if (!song) {
        return _react2.default.createElement(
          'li',
          { className: 'list-group-item' },
          _react2.default.createElement(
            'div',
            { className: 'data-label' },
            'Song Name'
          ),
          _react2.default.createElement(
            'div',
            { className: 'data-value' },
            'Unknown'
          )
        );
      }
      if (connected) {
        return _react2.default.createElement(
          'li',
          { className: 'list-group-item' },
          _react2.default.createElement(
            'div',
            { className: 'data-label' },
            'Song Name'
          ),
          _react2.default.createElement(
            'div',
            { className: 'data-value' },
            song
          )
        );
      }
      return _react2.default.createElement('div', null);
    }
  }, {
    key: 'renderAlbumName',
    value: function renderAlbumName(album, connected) {
      if (!album) {
        return _react2.default.createElement(
          'li',
          { className: 'list-group-item' },
          _react2.default.createElement(
            'div',
            { className: 'data-label' },
            'Album Name'
          ),
          _react2.default.createElement(
            'div',
            { className: 'data-value' },
            'Unknown'
          )
        );
      }
      if (connected) {
        return _react2.default.createElement(
          'li',
          { className: 'list-group-item' },
          _react2.default.createElement(
            'div',
            { className: 'data-label' },
            'Album Name'
          ),
          _react2.default.createElement(
            'div',
            { className: 'data-value' },
            album
          )
        );
      }
      return _react2.default.createElement('div', null);
    }
  }, {
    key: 'renderArtistName',
    value: function renderArtistName(artist, connected) {
      if (!artist) {
        return _react2.default.createElement(
          'li',
          { className: 'list-group-item' },
          _react2.default.createElement(
            'div',
            { className: 'data-label' },
            'Artist Name'
          ),
          _react2.default.createElement(
            'div',
            { className: 'data-value' },
            'Unknown'
          )
        );
      }
      if (connected) {
        return _react2.default.createElement(
          'li',
          { className: 'list-group-item' },
          _react2.default.createElement(
            'div',
            { className: 'data-label' },
            'Artist Name'
          ),
          _react2.default.createElement(
            'div',
            { className: 'data-value' },
            artist
          )
        );
      }
      return _react2.default.createElement('div', null);
    }
  }, {
    key: 'renderImage',
    value: function renderImage(image, connected) {
      if (!image) {
        return _react2.default.createElement(_CardViewImg2.default, { image: 'https://s3-us-west-2.amazonaws.com/bluerainimages/music.png' });
      }
      if (connected) {
        return _react2.default.createElement(_CardViewImg2.default, { image: image });
      }
      return _react2.default.createElement('div', null);
    }
  }]);

  return SonosSensorInfo;
}(_react2.default.Component);

exports.default = SonosSensorInfo;


SonosSensorInfo.propTypes = propTypes;
SonosSensorInfo.defaultProps = defaultProps;
