// bluerain-app-device-explorer/src/components/Device/AddDevice/FourthScreen/AddDeviceFourthScreen.component.js
import React, { PropTypes } from 'react';
import ComponentState from 'bluerain-ui/lib/ComponentState/ComponentState.component';
import ListViewItem from 'bluerain-ui/lib/ListView/ListViewItem';

const propTypes = {
  onClick: PropTypes.func
};

class AddDeviceFirstScreen extends React.Component {

  constructor(props) {
    super(props);
    this.onSubmit = this.onSubmit.bind(this);
  }

  // Form on submit
  onSubmit(event) {
    event.preventDefault();
    if (this.submitHandler) {
      this.submitHandler(event, this.props);
    }
  }


  render() {
    this.submitHandler = this.props.onClick;
    const Prompt = {

      fontSize: 14,
      textAlign: 'left',
      color: 'red'

    };
    return (
      <div>
        <ComponentState
          imageSize="lg"
          card
          title="AC is now connected to internet."
          description=""
        />

        <form onSubmit={this.onSubmit}>
          <ListViewItem className="text-xs-left">
            <div><p style={Prompt}>*Please connect your System/mobile to Internet before moving To Next Step</p></div>
          </ListViewItem>
        </form>
      </div>
    );
  }
}

AddDeviceFirstScreen.propTypes = propTypes;

export default AddDeviceFirstScreen;
