// bluerain-app-device-explorer/src/components/Site/SitesGrid/SitesGrid.container.js
import React, { PropTypes } from 'react';
import { Link } from 'react-router';
import { Icon } from 'react-fa';
import { withLoadingBar } from 'bluerain-client-services';
import GridView from 'bluerain-ui/lib/GridView/GridView.component';
import GridViewItem from 'bluerain-ui/lib/GridView/GridViewItem.component';
import CardView from 'bluerain-ui/lib/Card/CardView/CardView.component';

const propTypes = {
  data: PropTypes.object,
  hideLoading: PropTypes.func
};

const defaultProps = {
  onComponentUpdate: () => {
  }
};

export class SitesGrid extends React.Component {

  constructor(props) {
    super(props);

    this.renderSiteCard = this.renderSiteCard.bind(this);
  }

  componentDidMount() {
    if (this.props.hideLoading) {
      this.props.hideLoading();
    }
    return null;
  }

  getImageUrl(site) {
    if (site.image) {
      return site.image;
    }

    if (site.siteType && site.siteType.icon) {
      return site.siteType.icon;
    }

    return null;
  }

  renderSiteCard(site) {
    const url = `/app/device-explorer/sites/${site.id}`;
    // url = encodeURI(url);

    const props = {
      title: site.name,
      linkTag: Link,
      to: url
    };

    const imageUrl = this.getImageUrl(site);

    if (imageUrl) {
      props.image = imageUrl;
    }

    return (
      <GridViewItem key={site.id}>
        <CardView {...props} />
      </GridViewItem>
    );
  }

  render() {
    try {
      let sites = this.props.data.viewer.sites.edges;
      if (sites.length > 0) {
        sites = sites.map(item => item.node);
        return (
          <GridView>
            {sites.map(this.renderSiteCard)}
          </GridView>
        );
      }else {
        this.props.hideLoading();
      }

      return (
        <div className="app-content">
          <div className="container">
            <div className="v-align">
              <div className="v-align-middle empty-block">
                <Icon name="home fa-5x" />
                <h4>No site added</h4>
                <p>To monitor this app you will need to add your sites first.</p>
                <Link className="btn btn-outline-primary btn-lg" to={'/app/device-explorer/site/new'}>Add Site</Link>
              </div>
            </div>
          </div>
        </div>
      );

    } catch (e) {
      this.props.hideLoading();
      console.info('There was an error fetching sites.');
    }

    return (<div />);

  }
}

SitesGrid.propTypes = propTypes;
SitesGrid.defaultProps = defaultProps;

export default withLoadingBar(SitesGrid);
