// bluerain-app-device-explorer/src/pages/SitePage/SiteEditForm/SiteEditFormPage.js
/**
 * Created by anser on 29/6/17.
 */
import React, { PropTypes } from 'react';
import { Container, Row, Col } from 'reactstrap';
import AppContent from 'bluerain-ui/lib/Layout/AppContent';
import PrimaryAppNav from 'bluerain-ui/lib/Nav/PrimaryAppNav';
import SiteEditFormWidget from '../../../components/Site/SiteEditForm/SiteEditForm.widget';

const propTypes = {
  title: PropTypes.string,
  params: PropTypes.object
};

const defaultProps = {
  title: 'Edit Site Form'
};

class SiteEditFormPage extends React.Component {

  renderPrimaryNav() {
    const title = this.props.title;
    const previous = {
      title: 'Sites',
      url: '/app/device-explorer'
    };
    return (
      <PrimaryAppNav title={title} previous={previous}>
        <i className="fa fa-plus hidden-sm-up" />
      </PrimaryAppNav>
      // <PrimaryAppNav title={title} />
    );
  }

  static renderSecondaryNav() {
  }

  static renderContent(siteId) {
    return (
      <AppContent>
        <Container>
          <Row>
            <Col md={{ size: 8, offset: 2 }} className="col-faded">
              <SiteEditFormWidget siteId={siteId} />
            </Col>
          </Row>
        </Container>
      </AppContent>
    );
  }

  render() {
    const { siteId } = this.props.params;
    return (
      <div>
        {this.renderPrimaryNav()}
        {SiteEditFormPage.renderSecondaryNav()}
        {SiteEditFormPage.renderContent(siteId)}
      </div>
    );
  }
}


SiteEditFormPage.propTypes = propTypes;
SiteEditFormPage.defaultProps = defaultProps;

export default SiteEditFormPage;
