// bluerain-app-device-explorer/src/components/Asset/Charts/FitbitCaloriesChart/CaloriesChart.component.js
import React, { PropTypes } from 'react';
import AreaChartCard from 'bluerain-ui/lib/Card/ChartCard/AreaChartCard.component';

const propTypes = {
  label: PropTypes.string,
  value: PropTypes.string,
  data: PropTypes.array,
  color: PropTypes.string,
  tooltip: PropTypes.string,
  to: PropTypes.string,
  linkTag: PropTypes.oneOfType([
    React.PropTypes.func,
    React.PropTypes.string
  ]),
  onClick: PropTypes.func.isRequired,
  x: PropTypes.string,
  y: PropTypes.func
};

const defaultProps = {
  label: 'Average Calories Burnt',
  value: '',
  data: [],
  color: 'blue',
  linkTag: 'a',
  to: '#',
  onClick: () => {
  }
};

export default class CaloriesChartComponent extends React.Component {

  render() {
    const {
      data,
      x,
      y,
      color,
      label,
      value,
      tooltip,
      to,
      linkTag,
      ...props
    } = this.props;

    return (
      <AreaChartCard
        data={data}
        x={x}
        y={y}
        color={color}
        label={label}
        value={value}
        tooltip={tooltip}
        linkTag={linkTag}
        to={to}
        interpolation="basis"
        {...props}
      />
    );
  }
}

CaloriesChartComponent.propTypes = propTypes;
CaloriesChartComponent.defaultProps = defaultProps;
