// bluerain-app-device-explorer/src/components/Device/DeviceSettings/DeviceGeneralSettings/OwnershipTransfer/OwnershipTransfer.component.js
/**
 * Created by anser on 1/17/17.
 */
import React, { PropTypes } from 'react';
import { Collapse } from 'reactstrap';
import OwnershipTransferForm from './OwnershipTransferForm';
const propTypes = {
  data: PropTypes.object,
};

const defaultProps = {};

export default class OwnershipTransfer extends React.Component {

  static renderDeleteDevice() {
    return (
      <li className="list-group-item list-group-item-danger">
        <div className="data-value">Delete Device</div>
      </li>
    );
  }

  static renderOwnershipTransfer() {
    return (
      <div>
        <li className="list-group-item list-group-item-danger">
          <div className="data-value">Transfer Ownership</div>
        </li>
      </div>
    );
  }

  constructor(props) {
    super(props);
    this.toggle = this.toggle.bind(this);
    this.state = { collapse: false };
  }

  toggle() {
    this.setState({ collapse: !this.state.collapse });
  }


  render() {
    return (
      <div>
        {/* <ul className="list-group data-list" style={{ cursor: 'pointer' }} onClick={this.toggle}>*/}
        {/* { OwnershipTransfer.renderOwnershipTransfer()}*/}
        {/* </ul>*/}
        <Collapse isOpen={this.state.collapse}>
          <OwnershipTransferForm assetId={this.props.data.variables.assetId} />
        </Collapse>
        <ul className="list-group data-list">
          { OwnershipTransfer.renderDeleteDevice() }
        </ul>
      </div>
    );
  }
}

OwnershipTransfer.propTypes = propTypes;
OwnershipTransfer.defaultProps = defaultProps;
