// bluerain-app-device-explorer/src/components/Device/DeviceSettings/DeviceGeneralSettings/OwnershipTransfer/OwnershipTransferForm.js
/**
 * Created by anser on 2/21/17.
 */
import React, { PropTypes } from 'react';
import ListViewItem from 'bluerain-ui/lib/ListView/ListViewItem';
import Form2 from './OwnershipTransferForm2';

const propTypes = {
  onSubmit: PropTypes.func,
  assetId: PropTypes.string
};

const defaultProps = {};
// let emailId;
class OwnershipTransferForm extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      value: '',
      onSubmitCallback: props.onSubmit
    };

    this.handleChange = this.handleChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  // Form on submit
  onSubmit(event) {
    // emailId = this.state.value;
    event.preventDefault();
  }

  // Get value of asset name
  handleChange(event) {
    // emailId = event.target.value;
    this.setState({
      value: event.target.value,
      // value: limitLength(extraValidations(event.target.value), 25)
    });
  }

  render() {
    return (
      <div>
        <form>
          <ListViewItem>
            <input
              type="email" value={this.state.value} onChange={this.handleChange} className="form-control"
              aria-describedby="email" placeholder="Email Address"
            />
            <small className="form-text text-muted">Enter email address of the user you want to transfer ownership.
            </small>
          </ListViewItem>
          <ListViewItem style={{ display: 'none' }}>
            <button type="submit" className="btn btn-primary">Transfer</button>
          </ListViewItem>
        </form>
        <Form2 emailId={this.state.value} assetId={this.props.assetId} />
      </div>
    );
  }
}


export default OwnershipTransferForm;

OwnershipTransferForm.propTypes = propTypes;
OwnershipTransferForm.defaultProps = defaultProps;
