// bluerain-app-device-explorer/lib/components/Report/RefrigeratorTimeReport/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _reactApollo = require('react-apollo');

var _RefrigeratorTimeReport = require('./RefrigeratorTimeReport.container');

var _RefrigeratorTimeReport2 = _interopRequireDefault(_RefrigeratorTimeReport);

var _query = require('./query.graphql');

var _query2 = _interopRequireDefault(_query);

var _graphqlRelay = require('graphql-relay');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Created by atif on 12/18/16.
 */
exports.default = (0, _reactApollo.graphql)(_query2.default, {
  options: function options(args) {
    var startDateFormatted = args.startDate;
    var endDateFormatted = args.endDate;

    //   const queryString = `[{\"condition\":\"exists\",\"key\":\"data.door\"},{\"condition\":\"descending\",\"key\":\"time\"},{\"condition\":\"lessThanOrEqualTo\",\"key\":\"time\",\"value\":{\"__type\":\"Date\",
    // \"iso\":\"${endDateFormatted}\"}}, {\"condition\":\"greaterThanOrEqualTo\",\"key\":\"time\",\"value\":{\"__type\":\"Date\",\"iso\":\"${startDateFormatted}\"}}]`;

    var queryString = {
      where: {
        type: 'state',
        assetId: (0, _graphqlRelay.fromGlobalId)(args.assetId).id,
        time: { between: [startDateFormatted, endDateFormatted] },
        'data.door': { exists: true }
      },
      order: 'time DESC'
    };
    return { variables: { assetId: args.assetId, queryString: queryString } };
  }
})(_RefrigeratorTimeReport2.default);
