// bluerain-app-device-explorer/lib/components/Asset/Charts/CurrentChart/CurrentChart.component.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _AreaChartCard = require('bluerain-ui/lib/Card/ChartCard/AreaChartCard.component');

var _AreaChartCard2 = _interopRequireDefault(_AreaChartCard);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Created by atif on 12/18/16.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


var propTypes = {
  label: _react.PropTypes.string,
  value: _react.PropTypes.string,
  data: _react.PropTypes.array,
  color: _react.PropTypes.string,
  linkTag: _react.PropTypes.oneOfType([_react2.default.PropTypes.func, _react2.default.PropTypes.string]),
  onClick: _react.PropTypes.func.isRequired,
  x: _react.PropTypes.string,
  y: _react.PropTypes.func,
  to: _react.PropTypes.string,
  tooltip: _react.PropTypes.bool
};

var defaultProps = {
  label: 'Current',
  value: '',
  data: [],
  color: 'purple',
  linkTag: 'a',
  to: '#',
  onClick: function onClick() {}
};

var CurrentChartComponent = function (_React$Component) {
  _inherits(CurrentChartComponent, _React$Component);

  function CurrentChartComponent() {
    _classCallCheck(this, CurrentChartComponent);

    return _possibleConstructorReturn(this, (CurrentChartComponent.__proto__ || Object.getPrototypeOf(CurrentChartComponent)).apply(this, arguments));
  }

  _createClass(CurrentChartComponent, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          data = _props.data,
          x = _props.x,
          y = _props.y,
          color = _props.color,
          label = _props.label,
          value = _props.value,
          tooltip = _props.tooltip,
          to = _props.to,
          linkTag = _props.linkTag,
          props = _objectWithoutProperties(_props, ['data', 'x', 'y', 'color', 'label', 'value', 'tooltip', 'to', 'linkTag']);

      return _react2.default.createElement(_AreaChartCard2.default, _extends({
        data: data,
        x: x,
        y: y,
        color: color,
        label: label,
        value: value,
        tooltip: tooltip,
        linkTag: linkTag,
        to: to,
        interpolation: 'basis'
      }, props));
    }
  }]);

  return CurrentChartComponent;
}(_react2.default.Component);

exports.default = CurrentChartComponent;


CurrentChartComponent.propTypes = propTypes;
CurrentChartComponent.defaultProps = defaultProps;
