// bluerain-app-device-explorer/lib/components/Asset/Charts/CurrentChart/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _reactApollo = require('react-apollo');

var _graphqlRelay = require('graphql-relay');

var _CurrentChart = require('./CurrentChart.container');

var _CurrentChart2 = _interopRequireDefault(_CurrentChart);

var _query = require('./query.graphql');

var _query2 = _interopRequireDefault(_query);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// import CurrentChartStateful from './CurrentChart.stateful';
/**
 * Created by atif on 12/18/16.
 */
exports.default = (0, _reactApollo.graphql)(_query2.default, {
  options: function options(args) {
    var startDateFormatted = args.startDate;
    var endDateFormatted = args.endDate;

    // const queryString = {
    //   type: 'state',
    //   time: { between: [startDateFormatted, endDateFormatted] },
    //   'data.current': { exists: true }
    // };

    var queryString = {
      where: {
        type: 'state',
        assetId: (0, _graphqlRelay.fromGlobalId)(args.assetId).id,
        time: { between: [startDateFormatted, endDateFormatted] },
        "or": [{ 'data.standby': { exists: true } }, { 'data.current': { exists: true } }]
      },
      order: 'time DESC'
    };

    return { variables: { assetId: args.assetId, queryString: queryString } };
  }
})(_CurrentChart2.default);
