// bluerain-app-device-explorer/src/components/Asset/AssetControl/ControlItems/DropdownControl.js
import React, { PropTypes } from 'react';
import Select from 'react-select';
import 'react-select/dist/react-select.css';

// import './dropdown.scss';

const propTypes = {
  label: PropTypes.string,
  value: PropTypes.string,
  disabled: PropTypes.bool,
  className: PropTypes.string,
  onChange: PropTypes.func,
  items: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  }))
};

const defaultProps = {
  label: '',
  disabled: false,
  className: '',
  value: 1,
  items: [],
  onChange: () => {
  }
};

const style = {
  control: {
    display: 'flex'
  },
  label: {
    flexGrow: 1,
    lineHeight: '29px'
  },
  dropdown: {
    height: '29px'
  },
  dropdownLabel: {
    lineHeight: '29px',
    paddingRight: '32px'
  },
  dropdownIcon: {
    top: '3px',
    right: 0
  },
  dropdownUnderline: {
    position: 'relative'
  },
};


export default class DropdownControl extends React.Component {

  constructor(props) {
    super(props);

    this.onChange = this.onChange.bind(this);

    this.state = {
      value: props.value,
      disabled: props.disabled,
    };
  }

  componentWillReceiveProps(props) {
    this.setState(props);
  }

  onChange(value) {
    this.setState({
      value
    });

    this.props.onChange(value);
  }

  render() {
    const { value } = this.state;

    const {
      label,
      items,
      className,
    } = this.props;

    return (
      <div
        className={`device-control-item dropdown-control clearfix ${className}`}
        style={style.control}
      >
        <div
          className="control-label"
          style={style.label}
        >
          {label}
        </div>
        <Select
          name="form-field-name"
          value={value}
          options={items}
          onChange={this.onChange}
          clearable={false}
          searchable={false}
          style={{ width: '65px' }}
        />
      </div>
    );
  }
}

DropdownControl.propTypes = propTypes;
DropdownControl.defaultProps = defaultProps;
