// bluerain-app-device-explorer/src/components/Asset/AssetControl/ControlItems/SwitchControl.js
import React, { PropTypes } from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Toggle from 'material-ui/Toggle';

const propTypes = {
  label: PropTypes.string,
  onToggle: PropTypes.func,
  defaultValue: PropTypes.bool,
  toggled: PropTypes.bool,
  disabled: PropTypes.bool
};

const defaultProps = {
  label: '',
  onToggle: () => {
  },
  defaultValue: false,
  toggled: false,
  disabled: false
};

const style = {
  label: {
    marginBottom: 0,
  }
};

const SwitchControl = () => ({

  render() {
    const {
      label,
      onToggle,
      defaultValue,
      className,
      toggled,
      disabled
    } = this.props;

    return (
      <div className={`device-control-item switch-control ${className}`}>
        <MuiThemeProvider>
          <Toggle
            label={label}
            labelStyle={style.label}
            defaultToggled={defaultValue}
            onToggle={(ev, toggle) => onToggle(toggle)}
            toggled={toggled}
            disabled={disabled}
          />
        </MuiThemeProvider>
      </div>
    );
  },
});

SwitchControl.propTypes = propTypes;
SwitchControl.defaultProps = defaultProps;

export default SwitchControl;
