// bluerain-app-device-explorer/src/components/Report/RefrigeratorTimeReport/index.js
/**
 * Created by atif on 12/18/16.
 */
import { graphql } from 'react-apollo';
import TimeReportContainer from './RefrigeratorTimeReport.container';
import ViewerQuery from './query.graphql';
import { fromGlobalId } from 'graphql-relay';

export default graphql(ViewerQuery, {
  options: (args) => {
    const startDateFormatted = args.startDate;
    const endDateFormatted = args.endDate;

  //   const queryString = `[{\"condition\":\"exists\",\"key\":\"data.door\"},{\"condition\":\"descending\",\"key\":\"time\"},{\"condition\":\"lessThanOrEqualTo\",\"key\":\"time\",\"value\":{\"__type\":\"Date\",
  // \"iso\":\"${endDateFormatted}\"}}, {\"condition\":\"greaterThanOrEqualTo\",\"key\":\"time\",\"value\":{\"__type\":\"Date\",\"iso\":\"${startDateFormatted}\"}}]`;

    const queryString = {
      where: {
        type: 'state',
        assetId: fromGlobalId(args.assetId).id,
        time: { between: [startDateFormatted, endDateFormatted] },
        'data.door': { exists: true }
      },
      order: 'time DESC'
    };
    return { variables: { assetId: args.assetId, queryString } };
  }
})(TimeReportContainer);
