// bluerain-app-device-explorer/src/components/Report/RefrigeratorTimeReport/RefrigeratorTimeReport.component.js
/**
 * Created by atif on 12/18/16.
 */
import React, { PropTypes } from 'react';
import AreaChartAxis from 'bluerain-ui/lib/Card/ChartCard/AreaChartAxis.component';
// import moment from 'moment';
// import { Table } from 'reactstrap';

const propTypes = {
  data: PropTypes.array,
  x: PropTypes.string,
  y: PropTypes.func,
  color: PropTypes.string,
  tooltip: PropTypes.bool,
  tooltipIsBoolean: PropTypes.bool,
  tooltipBooleanTrueLabel: PropTypes.string,
  tooltipBooleanFalseLabel: PropTypes.string
};

const defaultProps = {
  data: [],
  tooltipIsBoolean: undefined,
  tooltipBooleanTrueLabel: '',
  tooltipBooleanFalseLabel: '',
};

export default class TimeReportComponent extends React.Component {

  render() {

    const fillColor = 'rgba(0,255,0,0.3)';
    return (
      <div>
        <div style={{ width: '100%', margin: '0 auto' }}>
          <AreaChartAxis
            data={this.props.data}
            x={this.props.x}
            y={this.props.y}
            tooltip={this.props.tooltip}
            height={300}
            fillColor={fillColor}
            strokeColor={this.props.color}
            dateFormat="DD, MMM YYYY, hh:mm:ss a"
            xTickCount={5}
            yTickCount={1}
            tickFormat={['on']}
            tooltipIsBoolean={this.props.tooltipIsBoolean}
            tooltipBooleanTrueLabel={this.props.tooltipBooleanTrueLabel}
            tooltipBooleanFalseLabel={this.props.tooltipBooleanFalseLabel}
          />
        </div>
        {/*<div className="panel-data border-top">*/}
          {/*<div className="table-responsive">*/}
            {/*<Table>*/}
              {/*<thead className="thead-default">*/}
                {/*<tr>*/}
                  {/*<th>Date/Time</th>*/}
                  {/*<th>Value</th>*/}
                {/*</tr>*/}
              {/*</thead>*/}
              {/*<tbody>*/}
                {/*{*/}
                {/*this.props.data.map((item, index) => (*/}
                  {/*<tr key={index}>*/}
                    {/*<td>{moment(item.time).format('DD, MMM YYYY, hh:mm:ss a')}</td>*/}
                    {/*<td>{item.door ? 'Open' : 'Close'}</td>*/}
                  {/*</tr>*/}
                {/*))*/}
              {/*}*/}
              {/*</tbody>*/}
            {/*</Table>*/}
          {/*</div>*/}
        {/*</div>*/}
      </div>
    );
  }
}

TimeReportComponent.propTypes = propTypes;
TimeReportComponent.defaultProps = defaultProps;
