// bluerain-app-device-explorer/src/components/Report/RefrigeratorTimeReport/RefrigeratorTimeReport.container.js
/**
 * Created by atif on 12/18/16.
 */
import React, { PropTypes } from 'react';
import TimeReportComponent from './RefrigeratorTimeReport.component';

const propTypes = {
  data: PropTypes.array,
  tooltip: PropTypes.bool,
  startDate: PropTypes.string,
  endDate: PropTypes.string,
  color: PropTypes.string
};

const defaultPropTypes = {
  startDate: '01/01/1970',
  endDate: '01/01/2099',
  tooltip: true,
  color: 'yellow'
};

export default class TimeReportContainer extends React.Component {

  render() {

    const {
      color,
      tooltip,
      data: dbData,
      ...rest
    } = this.props;


    const data = [];

    try {
      // let dataFromQuery = dbData.node.events.edges;
      let dataFromQuery = dbData.Event.EventFind.edges;
      dataFromQuery = dataFromQuery.map(item => item.node);

      dataFromQuery.forEach((item) => {
        if (item.data && item.data.door !== undefined) {
          data.push({
            time: new Date(item.time),
            door: item.data.door ? false : true
          });
        }
      });

      if (data.length > 0) {
        data.sort((a, b) => (a.time > b.time) ? 1 : ((b.time > a.time) ? -1 : 0));
      }

    } catch (error) {
      console.info('Time Report: no data found');
    }

    return (
      <TimeReportComponent
        data={data}
        x="time"
        y={datum => (datum.door)}
        tooltip={tooltip}
        color={color}
        tooltipIsBoolean
        tooltipBooleanTrueLabel="On"
        tooltipBooleanFalseLabel="Off"
        {...rest}
      />
    );
  }
}

TimeReportContainer.propTypes = propTypes;
TimeReportContainer.defaultProps = defaultPropTypes;
