// bluerain-app-device-explorer/lib/components/Asset/AssetSensorInfo/RefrigeratorSensorInfo/RefrigeratorSensorInfo.container.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactFa = require('react-fa');

var _reactFa2 = _interopRequireDefault(_reactFa);

require('bluerain-bootstrap-theme/dist/css/components/data-list.css');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

// import DataListComponent from '../DataList/DataList.component';

var propTypes = {
  data: _react.PropTypes.object
};

var defaultProps = {};

var RefrigeratorSensorInfo = function (_React$Component) {
  _inherits(RefrigeratorSensorInfo, _React$Component);

  function RefrigeratorSensorInfo() {
    _classCallCheck(this, RefrigeratorSensorInfo);

    return _possibleConstructorReturn(this, (RefrigeratorSensorInfo.__proto__ || Object.getPrototypeOf(RefrigeratorSensorInfo)).apply(this, arguments));
  }

  _createClass(RefrigeratorSensorInfo, [{
    key: 'render',
    value: function render() {
      if (!this.props.data.node || !this.props.data.node.state) {
        return _react2.default.createElement('div', null);
      }

      var _props$data$node$stat = this.props.data.node.state,
          temperature = _props$data$node$stat.temperature,
          door = _props$data$node$stat.door;
      // todo: where true is passed connected will be passs and where false is passed standby will be pass in below functions

      return _react2.default.createElement(
        'ul',
        { className: 'list-group data-list' },
        RefrigeratorSensorInfo.renderConnectionStatus(true),
        RefrigeratorSensorInfo.renderdoor(door, true),
        RefrigeratorSensorInfo.renderCompressor((temperature.t1 / 1024 - 0.5) * 100, true, false),
        RefrigeratorSensorInfo.renderCondenserOut((temperature.t2 / 1024 - 0.5) * 100, true, false),
        RefrigeratorSensorInfo.renderCondenserIn((temperature.t3 / 1024 - 0.5) * 100, true, false),
        RefrigeratorSensorInfo.renderinsideFreezer((temperature.t5 / 1024 - 0.5) * 100, true, false),
        RefrigeratorSensorInfo.renderinsideRefrigerator((temperature.t6 / 1024 - 0.5) * 100, true, false)
      );
    }
  }], [{
    key: 'renderConnectionStatus',
    value: function renderConnectionStatus(connected) {

      var status = null;

      if (typeof connected !== 'boolean') {
        return _react2.default.createElement('div', null);
      } else if (connected) {
        status = _react2.default.createElement(
          'span',
          null,
          'Online ',
          _react2.default.createElement(_reactFa2.default, { name: 'circle text-success' })
        );
      } else {
        status = _react2.default.createElement(
          'span',
          null,
          'Offline ',
          _react2.default.createElement(_reactFa2.default, { name: 'circle text-danger' })
        );
      }
      return _react2.default.createElement(
        'li',
        { className: 'list-group-item' },
        _react2.default.createElement(
          'div',
          { className: 'data-label' },
          'Status'
        ),
        _react2.default.createElement(
          'div',
          { className: 'data-value' },
          status
        )
      );
    }
  }, {
    key: 'renderAmbientTemperature',
    value: function renderAmbientTemperature(temperature, connected) {
      if (typeof temperature !== 'number') {
        return _react2.default.createElement('div', null);
      }
      if (connected) {
        return _react2.default.createElement(
          'li',
          { className: 'list-group-item' },
          _react2.default.createElement(
            'div',
            { className: 'data-label' },
            'Ambient Temperature'
          ),
          _react2.default.createElement(
            'div',
            { className: 'data-value' },
            temperature.toFixed(0),
            ' \u2103'
          )
        );
      }
      return _react2.default.createElement('div', null);
    }
  }, {
    key: 'renderdoor',
    value: function renderdoor(door, connected) {

      if (typeof door !== 'number') {
        return _react2.default.createElement('div', null);
      } else if (door) {
        door = 'Open';
      } else {
        door = 'Close';
      }
      if (connected) {
        return _react2.default.createElement(
          'li',
          { className: 'list-group-item' },
          _react2.default.createElement(
            'div',
            { className: 'data-label' },
            'Door'
          ),
          _react2.default.createElement(
            'div',
            { className: 'data-value' },
            door
          )
        );
      }
      return _react2.default.createElement('div', null);
    }
  }, {
    key: 'renderCompressor',
    value: function renderCompressor(compressor, connected, standby) {
      if (typeof compressor !== 'number') {
        return _react2.default.createElement('div', null);
      }
      if (standby) {
        return _react2.default.createElement('div', null);
      }
      if (connected) {
        return _react2.default.createElement(
          'li',
          { className: 'list-group-item' },
          _react2.default.createElement(
            'div',
            { className: 'data-label' },
            'Compressor Shell'
          ),
          _react2.default.createElement(
            'div',
            { className: 'data-value' },
            compressor.toFixed(0),
            ' \u2103'
          )
        );
      }
      return _react2.default.createElement('div', null);
    }
  }, {
    key: 'renderCondenserIn',
    value: function renderCondenserIn(condenserIn, connected, standby) {
      if (typeof condenserIn !== 'number') {
        return _react2.default.createElement('div', null);
      }
      if (standby) {
        return _react2.default.createElement('div', null);
      }
      if (connected) {
        return _react2.default.createElement(
          'li',
          { className: 'list-group-item' },
          _react2.default.createElement(
            'div',
            { className: 'data-label' },
            'Condenser Input'
          ),
          _react2.default.createElement(
            'div',
            { className: 'data-value' },
            condenserIn.toFixed(0),
            ' \u2103'
          )
        );
      }
      return _react2.default.createElement('div', null);
    }
  }, {
    key: 'renderCondenserOut',
    value: function renderCondenserOut(condenserOut, connected, standby) {
      if (typeof condenserOut !== 'number') {
        return _react2.default.createElement('div', null);
      }
      if (standby) {
        return _react2.default.createElement('div', null);
      }
      if (connected) {
        return _react2.default.createElement(
          'li',
          { className: 'list-group-item' },
          _react2.default.createElement(
            'div',
            { className: 'data-label' },
            'Condenser Output'
          ),
          _react2.default.createElement(
            'div',
            { className: 'data-value' },
            condenserOut.toFixed(0),
            ' \u2103'
          )
        );
      }
      return _react2.default.createElement('div', null);
    }
  }, {
    key: 'renderinsideFreezer',
    value: function renderinsideFreezer(insideFreezer, connected, standby) {
      if (typeof insideFreezer !== 'number') {
        return _react2.default.createElement('div', null);
      }
      if (standby) {
        return _react2.default.createElement('div', null);
      }
      if (connected) {
        return _react2.default.createElement(
          'li',
          { className: 'list-group-item' },
          _react2.default.createElement(
            'div',
            { className: 'data-label' },
            'Inside Freezer'
          ),
          _react2.default.createElement(
            'div',
            { className: 'data-value' },
            insideFreezer.toFixed(0),
            ' \u2103'
          )
        );
      }
      return _react2.default.createElement('div', null);
    }
  }, {
    key: 'renderinsideRefrigerator',
    value: function renderinsideRefrigerator(insideRefrigerator, connected, standby) {
      if (typeof insideRefrigerator !== 'number') {
        return _react2.default.createElement('div', null);
      }
      if (standby) {
        return _react2.default.createElement('div', null);
      }
      if (connected) {
        return _react2.default.createElement(
          'li',
          { className: 'list-group-item' },
          _react2.default.createElement(
            'div',
            { className: 'data-label' },
            'Inside Refrigerator'
          ),
          _react2.default.createElement(
            'div',
            { className: 'data-value' },
            insideRefrigerator.toFixed(0),
            ' \u2103'
          )
        );
      }
      return _react2.default.createElement('div', null);
    }
  }]);

  return RefrigeratorSensorInfo;
}(_react2.default.Component);

exports.default = RefrigeratorSensorInfo;


RefrigeratorSensorInfo.propTypes = propTypes;
RefrigeratorSensorInfo.defaultProps = defaultProps;

// { this.renderAmbientTemperature(((temperature.t4 / 1024) - 0.5) * 100, true) }
