// bluerain-ui/lib/Card/CardView/CardView.component.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _CardBlock = require('reactstrap/lib/CardBlock');

var _CardBlock2 = _interopRequireDefault(_CardBlock);

var _CardText = require('reactstrap/lib/CardText');

var _CardText2 = _interopRequireDefault(_CardText);

require('bluerain-bootstrap-theme/dist/css/components/card-view.css');

var _CustomCardLink = require('../CustomCardLink');

var _CustomCardLink2 = _interopRequireDefault(_CustomCardLink);

var _CardViewImg = require('../CardViewImg');

var _CardViewImg2 = _interopRequireDefault(_CardViewImg);

var _CardView = require('../CardView');

var _CardView2 = _interopRequireDefault(_CardView);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Created by amna on 12/7/16.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */

var propTypes = {
  title: _react.PropTypes.string,
  subtitle: _react.PropTypes.string,
  image: _react.PropTypes.string,
  linkTag: _react.PropTypes.oneOfType([_react.PropTypes.func, _react.PropTypes.string]),
  to: _react.PropTypes.string,
  href: _react.PropTypes.string,
  onClick: _react.PropTypes.func
};

var defaultProps = {
  title: '',
  subtitle: '',
  image: '',
  linkTag: 'a',
  to: '',
  href: '',
  onClick: function onClick() {}
};

var CardView = function (_React$Component) {
  _inherits(CardView, _React$Component);

  _createClass(CardView, null, [{
    key: 'renderTitle',
    value: function renderTitle(title) {
      if (title) {
        return _react2.default.createElement(
          _CardText2.default,
          { tag: 'span', className: 'text-truncate' },
          title
        );
      }
    }
  }, {
    key: 'renderSubtitle',
    value: function renderSubtitle(subtitle) {
      if (subtitle) {
        return _react2.default.createElement(
          _CardText2.default,
          { tag: 'span' },
          subtitle
        );
      }
    }
  }]);

  function CardView(props) {
    _classCallCheck(this, CardView);

    var _this = _possibleConstructorReturn(this, (CardView.__proto__ || Object.getPrototypeOf(CardView)).call(this, props));

    _this.handleClickAction = _this.handleClickAction.bind(_this);
    return _this;
  }

  _createClass(CardView, [{
    key: 'handleClickAction',
    value: function handleClickAction() {
      this.props.onClick();
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          title = _props.title,
          subtitle = _props.subtitle,
          image = _props.image,
          linkTag = _props.linkTag,
          href = _props.href,
          to = _props.to,
          props = _objectWithoutProperties(_props, ['title', 'subtitle', 'image', 'linkTag', 'href', 'to']);

      return _react2.default.createElement(
        _CardView2.default,
        _extends({ onClick: this.handleClickAction }, props),
        _react2.default.createElement(
          _CustomCardLink2.default,
          {
            tag: linkTag,
            to: to,
            href: href
          },
          _react2.default.createElement(_CardViewImg2.default, { image: image }),
          _react2.default.createElement(
            _CardBlock2.default,
            null,
            _react2.default.createElement(
              'div',
              { className: 'primary-text' },
              CardView.renderTitle(title)
            ),
            _react2.default.createElement(
              'div',
              { className: 'text-muted secondary-text' },
              CardView.renderSubtitle(subtitle)
            )
          )
        )
      );
    }
  }]);

  return CardView;
}(_react2.default.Component);

exports.default = CardView;


CardView.propTypes = propTypes;
CardView.defaultProps = defaultProps;
