// bluerain-ui/lib/Card/PostCard/PostCardItemContent.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactFa = require('react-fa');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; } /**
                                                                                                                                                                                                                              * Created by anser on 12/7/16.
                                                                                                                                                                                                                              */

var propTypes = {
  tag: _react.PropTypes.oneOfType([_react.PropTypes.func, _react.PropTypes.string]),
  children: _react.PropTypes.oneOfType([_react.PropTypes.func, _react.PropTypes.string, _react.PropTypes.array]),
  holder: _react.PropTypes.oneOfType([_react.PropTypes.func, _react.PropTypes.string]),
  chevronRight: _react.PropTypes.bool,
  to: _react.PropTypes.string,
  href: _react.PropTypes.string
};

var defaultProps = {
  tag: undefined,
  children: {},
  holder: 'div',
  chevronRight: true,
  to: '',
  href: ''
};

var PostCardItemContent = function PostCardItemContent(props) {
  var Tag = props.tag,
      Holder = props.holder,
      to = props.to,
      href = props.href,
      chevronRight = props.chevronRight,
      children = props.children,
      attributes = _objectWithoutProperties(props, ['tag', 'holder', 'to', 'href', 'chevronRight', 'children']);
  // Chevron


  var chevron = '';
  if (chevronRight === true && Tag) {
    chevron = _react2.default.createElement(
      Tag,
      { to: to, href: href, className: 'card-chevron' },
      _react2.default.createElement(_reactFa.Icon, { name: 'chevron-right', style: { alignSelf: 'center' } })
    );
  }

  return _react2.default.createElement(
    Holder,
    _extends({}, attributes, { className: 'card-content' }),
    children,
    chevron
  );
};

PostCardItemContent.propTypes = propTypes;
PostCardItemContent.defaultProps = defaultProps;

exports.default = PostCardItemContent;
