// bluerain-ui/lib/Card/PostCard/PostCard.component.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRouter = require('react-router');

var _moment = require('moment');

var _moment2 = _interopRequireDefault(_moment);

var _reactFa = require('react-fa');

var _PostCardItem = require('./PostCardItem');

var _PostCardItem2 = _interopRequireDefault(_PostCardItem);

var _PostCardItemContent = require('./PostCardItemContent');

var _PostCardItemContent2 = _interopRequireDefault(_PostCardItemContent);

var _PostCardItemImage = require('./PostCardItemImage');

var _PostCardItemImage2 = _interopRequireDefault(_PostCardItemImage);

var _PostCardItemBody = require('./PostCardItemBody');

var _PostCardItemBody2 = _interopRequireDefault(_PostCardItemBody);

var _PostCardItemSource = require('./PostCardItemSource');

var _PostCardItemSource2 = _interopRequireDefault(_PostCardItemSource);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  initiator: _react.PropTypes.string,
  description: _react.PropTypes.string,
  time: _react.PropTypes.string,
  source: _react.PropTypes.string,
  image: _react.PropTypes.string,
  roundImage: _react.PropTypes.bool,
  userProfileLink: _react.PropTypes.string,
  linkTag: _react.PropTypes.oneOfType([_react.PropTypes.func, _react.PropTypes.string]),
  to: _react.PropTypes.string,
  href: _react.PropTypes.string,
  onClick: _react.PropTypes.func
};
var defaultProps = {
  initiator: '',
  description: '',
  time: '',
  source: '',
  image: '',
  roundImage: false,
  userProfileLink: undefined,
  linkTag: undefined,
  to: '',
  href: '#',
  onClick: function onClick() {}
};

var PostCard = function (_React$Component) {
  _inherits(PostCard, _React$Component);

  function PostCard() {
    _classCallCheck(this, PostCard);

    return _possibleConstructorReturn(this, (PostCard.__proto__ || Object.getPrototypeOf(PostCard)).apply(this, arguments));
  }

  _createClass(PostCard, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          initiator = _props.initiator,
          description = _props.description,
          time = _props.time,
          image = _props.image,
          roundImage = _props.roundImage,
          userProfileLink = _props.userProfileLink,
          source = _props.source,
          linkTag = _props.linkTag,
          to = _props.to,
          href = _props.href,
          props = _objectWithoutProperties(_props, ['initiator', 'description', 'time', 'image', 'roundImage', 'userProfileLink', 'source', 'linkTag', 'to', 'href']);

      return _react2.default.createElement(
        _PostCardItem2.default,
        _extends({ onClick: this.props.onClick }, props),
        _react2.default.createElement(
          _PostCardItemContent2.default,
          { tag: linkTag, to: to, href: href, chevronRight: true },
          PostCard.renderImage(image, roundImage),
          _react2.default.createElement(
            _PostCardItemBody2.default,
            null,
            _react2.default.createElement(
              'div',
              null,
              PostCard.renderInitiator(initiator, userProfileLink),
              ' ',
              description
            ),
            _react2.default.createElement(
              _PostCardItemSource2.default,
              null,
              PostCard.renderTimeAndSource(time, source, linkTag, to, href)
            )
          )
        )
      );
    }
  }], [{
    key: 'renderInitiator',
    value: function renderInitiator(initiator, userProfileLink) {
      if (initiator) {
        return userProfileLink ? _react2.default.createElement(
          'strong',
          null,
          _react2.default.createElement(
            _reactRouter.Link,
            { href: userProfileLink },
            initiator
          )
        ) : _react2.default.createElement(
          'strong',
          null,
          initiator
        );
      }
    }
  }, {
    key: 'handleNavigation',
    value: function handleNavigation(html, linkTag, to, href) {
      var Tag = linkTag;
      return linkTag ? _react2.default.createElement(
        Tag,
        { href: href, to: to },
        html
      ) : { html: html };
    }
  }, {
    key: 'renderTimeAndSource',
    value: function renderTimeAndSource(time, source, linkTag, to, href) {
      var html = void 0;
      var finalhtml = void 0;
      if (time) {
        html = _react2.default.createElement(
          'span',
          null,
          ' ',
          (0, _moment2.default)(time).fromNow()
        );
        if (linkTag) {
          html = PostCard.handleNavigation(html, linkTag, to, href);
        }
        if (!source) {
          finalhtml = _react2.default.createElement(
            'small',
            null,
            html
          );
        }
      }
      if (time && source) {
        finalhtml = _react2.default.createElement(
          'small',
          null,
          _react2.default.createElement(_reactFa.Icon, { name: 'clock-o' }),
          ' ',
          html,
          ' . ',
          source
        );
      } else {
        finalhtml = _react2.default.createElement(
          'small',
          null,
          source
        );
      }
      return finalhtml;
    }
  }, {
    key: 'renderImage',
    value: function renderImage(image, roundImage) {
      if (image) {
        return _react2.default.createElement(_PostCardItemImage2.default, { image: image, roundImage: roundImage });
      }
    }
  }]);

  return PostCard;
}(_react2.default.Component);

exports.default = PostCard;


PostCard.propTypes = propTypes;
PostCard.defaultProps = defaultProps;
