// bluerain-app-device-explorer/lib/components/Report/TemperatureReport/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _reactApollo = require('react-apollo');

var _graphqlRelay = require('graphql-relay');

var _TemperatureReport = require('./TemperatureReport.container');

var _TemperatureReport2 = _interopRequireDefault(_TemperatureReport);

var _query = require('./query.graphql');

var _query2 = _interopRequireDefault(_query);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Created by atif on 12/18/16.
 */
exports.default = (0, _reactApollo.graphql)(_query2.default, {
  options: function options(args) {
    var startDateFormatted = args.startDate;
    var endDateFormatted = args.endDate;

    var queryString = {
      where: {
        type: 'state',
        assetId: (0, _graphqlRelay.fromGlobalId)(args.assetId).id,
        time: { between: [startDateFormatted, endDateFormatted] },
        'data.ambientTemperature': { exists: true }
      },
      order: 'time DESC'
    };

    return { variables: { assetId: args.assetId, queryString: queryString } };
  }
})(_TemperatureReport2.default);
