// bluerain-app-device-explorer/src/components/Event/EventDetailPage/PostCardComponent/PostCard.component.js
/**
 * Created by anser on 1/2/17.
 */
import React, { PropTypes } from 'react';
import PostCard from 'bluerain-ui/lib/Card/PostCard/PostCard.component';
import 'bluerain-bootstrap-theme/dist/css/components/timeline.css';

const propTypes = {
  data: PropTypes.object,
};

const defaultProps = {};

let data;
export default class PostCardComponent extends React.Component {

  render() {
    try {
      if (this.props.data) {
        const post = this.props.data.node;
        if (post.type === 'USER_COMMAND') {
          const user = this.props.data.node.events.edges[0].node.account;
          data = {
            initiator: user.firstName && user.lastName ? `${user.firstName} ${user.lastName}` : user.username,
            source: 'Web App',
            roundImage: true,
            image: user.image || 'https://s3-us-west-2.amazonaws.com/bluerainimages/default-user.jpg',
          };
        } else if (post.type === 'FIRMWARE_UPGRADE') {
          data = {
            initiator: '',
            source: 'Network',
            roundImage: true,
            image: 'https://s3-us-west-2.amazonaws.com/bluerainimages/firmware_update.jpg',
          };
        } else {
          data = {
            initiator: '',
            source: 'Remote',
            roundImage: true,
            image: 'https://s3-us-west-2.amazonaws.com/bluerainimages/remote.svg'
          };
        }
        return (
          <PostCard
            key={post.id}
            initiator={data.initiator}
            description={post.description}
            time={post.time}
            source={data.source}
            roundImage={data.roundImage}
            image={data.image}
            userProfileLink={data.userProfileLink}
          />
        );
      }
      return (<div />);
    } catch (error) {
      console.log(error);
    }

  }
}

PostCardComponent.propTypes = propTypes;
PostCardComponent.defaultProps = defaultProps;
