// bluerain-app-device-explorer/lib/components/Device/DeviceSettings/DeviceGeneralSettings/OwnershipTransfer/UserInfo.widget.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _widget = require('bluerain-ui/lib/Widget/widget');

var _widget2 = _interopRequireDefault(_widget);

var _widgetHeader = require('bluerain-ui/lib/Widget/widgetHeader');

var _widgetHeader2 = _interopRequireDefault(_widgetHeader);

var _widgetTitle = require('bluerain-ui/lib/Widget/widgetTitle');

var _widgetTitle2 = _interopRequireDefault(_widgetTitle);

var _widgetContent = require('bluerain-ui/lib/Widget/widgetContent');

var _widgetContent2 = _interopRequireDefault(_widgetContent);

var _reactRouter = require('react-router');

var _UserInfoTransfer = require('./UserInfoTransfer.component');

var _UserInfoTransfer2 = _interopRequireDefault(_UserInfoTransfer);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Created by anser on 2/23/17.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


var propTypes = {
  params: _react.PropTypes.object,
  location: _react.PropTypes.object
};

var defaultProps = {};

var UserInfo = function (_React$Component) {
  _inherits(UserInfo, _React$Component);

  function UserInfo() {
    _classCallCheck(this, UserInfo);

    return _possibleConstructorReturn(this, (UserInfo.__proto__ || Object.getPrototypeOf(UserInfo)).apply(this, arguments));
  }

  _createClass(UserInfo, [{
    key: 'render',
    value: function render() {
      if (!this.props.params || !this.props.params.userId || !this.props.location || !this.props.location.query || !this.props.location.query.current_user) {
        return _react2.default.createElement('div', null);
      }
      var currentUser = this.props.location.query.current_user;
      return _react2.default.createElement(
        _widget2.default,
        null,
        _react2.default.createElement(
          _widgetHeader2.default,
          null,
          _react2.default.createElement(
            _widgetTitle2.default,
            null,
            'User Information'
          )
        ),
        _react2.default.createElement(
          _widgetContent2.default,
          null,
          _react2.default.createElement(_UserInfoTransfer2.default, {
            assetId: this.props.params.assetId, ownerId: currentUser,
            userId: this.props.params.userId
          })
        )
      );
    }
  }]);

  return UserInfo;
}(_react2.default.Component);

exports.default = (0, _reactRouter.withRouter)(UserInfo);


UserInfo.propTypes = propTypes;
UserInfo.defaultProps = defaultProps;
