// bluerain-app-device-explorer/src/components/Asset/Charts/EnergySavingChart/index.js
import { graphql } from 'react-apollo';
import { fromGlobalId } from 'graphql-relay';
// import EnergySavingChartStateful from './EnergySavingChart.stateful';
import EnergySavingChartContainer from './EnergySavingChart.container';
import ViewerQuery from './query.graphql';

export default graphql(ViewerQuery, {
  options: (args) => {
    const startDateFormatted = args.startDate;
    const endDateFormatted = args.endDate;

    // const queryString = {
    //   type: 'state',
    //   time: { between: [startDateFormatted, endDateFormatted] },
    //   'data.current': { exists: true }
    // };

    const queryString = {
      where: {
        type: 'state',
        assetId: fromGlobalId(args.assetId).id,
        time: { between: [startDateFormatted, endDateFormatted] },
        "or":[{'data.standby': { exists: true }}, {'data.current': { exists: true }}]
      },
      order: 'time DESC'
    };

    return { variables: { assetId: args.assetId, queryString } };
  }
})(EnergySavingChartContainer);
