// bluerain-app-device-explorer/lib/components/Asset/RefrigeratorLocationMapView/RefrigeratorLocationMapView.widget.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reverseGeocoding = require('reverse-geocoding');

var _reverseGeocoding2 = _interopRequireDefault(_reverseGeocoding);

var _widget = require('bluerain-ui/lib/Widget/widget');

var _widget2 = _interopRequireDefault(_widget);

var _widgetHeader = require('bluerain-ui/lib/Widget/widgetHeader');

var _widgetHeader2 = _interopRequireDefault(_widgetHeader);

var _widgetTitle = require('bluerain-ui/lib/Widget/widgetTitle');

var _widgetTitle2 = _interopRequireDefault(_widgetTitle);

var _widgetContent = require('bluerain-ui/lib/Widget/widgetContent');

var _widgetContent2 = _interopRequireDefault(_widgetContent);

var _RefrigeratorLocationMapView = require('./RefrigeratorLocationMapView.container');

var _RefrigeratorLocationMapView2 = _interopRequireDefault(_RefrigeratorLocationMapView);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; } /**
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                * Created by anser on 3/7/16.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                */


var propTypes = {
  data: _react.PropTypes.object
};

var defaultProps = {};

var RefrigeratorLocationMapViewWidget = function (_React$Component) {
  _inherits(RefrigeratorLocationMapViewWidget, _React$Component);

  function RefrigeratorLocationMapViewWidget(props) {
    _classCallCheck(this, RefrigeratorLocationMapViewWidget);

    var _this = _possibleConstructorReturn(this, (RefrigeratorLocationMapViewWidget.__proto__ || Object.getPrototypeOf(RefrigeratorLocationMapViewWidget)).call(this, props));

    _this.state = {
      myLocation: 'Loading'
    };
    return _this;
  }

  _createClass(RefrigeratorLocationMapViewWidget, [{
    key: 'render',
    value: function render() {
      if (!this.props.data.node || !this.props.data.node.state) {
        return _react2.default.createElement('div', null);
      }
      var _props$data$node$stat = this.props.data.node.state.location,
          latitude = _props$data$node$stat.latitude,
          longitude = _props$data$node$stat.longitude;

      if (!latitude || !longitude) {
        latitude = 31387640;
        longitude = 74148855;
      }

      var config = {
        latitude: latitude / 1000000,
        longitude: longitude / 1000000
      };

      var ref = this;

      if (this.state.myLocation === 'Loading') {
        _reverseGeocoding2.default.location(config, function (err, data) {
          ref.setState({ myLocation: data.results[0].formatted_address });
        });
      }

      return _react2.default.createElement(
        _widget2.default,
        null,
        _react2.default.createElement(
          _widgetHeader2.default,
          null,
          _react2.default.createElement(
            _widgetTitle2.default,
            null,
            ref.state.myLocation
          )
        ),
        _react2.default.createElement(
          _widgetContent2.default,
          null,
          _react2.default.createElement(_RefrigeratorLocationMapView2.default, { data: this.props.data })
        )
      );
    }
  }]);

  return RefrigeratorLocationMapViewWidget;
}(_react2.default.Component);

exports.default = RefrigeratorLocationMapViewWidget;


RefrigeratorLocationMapViewWidget.propTypes = propTypes;
RefrigeratorLocationMapViewWidget.defaultProps = defaultProps;
