// bluerain-app-device-explorer/lib/routes/index.js
'use strict';

function errorLoading(error) {
  throw new Error('Dynamic page loading failed: ' + error);
}

function loadRoute(cb) {
  return function (module) {
    return cb(null, module.default);
  };
}

function beforeLoadingRoute() {
  // <consol></consol>e.log('Before Loading Route');
  // store.dispatch(showLoading());
}

module.exports = {
  path: 'device-explorer',
  indexRoute: {
    getComponent: function getComponent(location, cb) {
      beforeLoadingRoute();
      System.import('../pages/IndexPage').then(loadRoute(cb)).catch(errorLoading);
    }
  },
  childRoutes: [{
    path: 'fitbit/callback',
    indexRoute: {
      getComponent: function getComponent(location, cb) {
        beforeLoadingRoute();
        System.import('../pages/CallBack/FitbitAuthHandler').then(loadRoute(cb)).catch(errorLoading);
      }
    }
  }, {
    path: 'sites/:siteId',
    indexRoute: {
      getComponent: function getComponent(location, cb) {
        beforeLoadingRoute();
        System.import('../pages/SitePage').then(loadRoute(cb)).catch(errorLoading);
      }
    },
    childRoutes: [{
      path: 'add-device',
      indexRoute: {
        getComponent: function getComponent(location, cb) {
          beforeLoadingRoute();
          System.import('../pages/Assets/AddAsset').then(loadRoute(cb)).catch(errorLoading);
        }
      }
    }, {
      path: 'edit',
      indexRoute: {
        getComponent: function getComponent(location, cb) {
          beforeLoadingRoute();
          System.import('../pages/SitePage/SiteEditForm/SiteEditFormPage').then(loadRoute(cb)).catch(errorLoading);
        }
      }
    }]
  }, {
    path: 'site/new',
    indexRoute: {
      getComponent: function getComponent(location, cb) {
        beforeLoadingRoute();
        System.import('../pages/SitePage/SiteForm/SiteFormPage').then(loadRoute(cb)).catch(errorLoading);
      }
    }
  }, {
    path: 'asset_locations',
    indexRoute: {
      getComponent: function getComponent(location, cb) {
        beforeLoadingRoute();
        System.import('../pages/Assets/Location/AssetLocationsPage').then(loadRoute(cb)).catch(errorLoading);
      }
    }
  }, {
    path: 'transfer_requests',
    indexRoute: {
      getComponent: function getComponent(location, cb) {
        beforeLoadingRoute();
        System.import('../pages/TransferRequests/TransferRequestsPage').then(loadRoute(cb)).catch(errorLoading);
      }
    }
  }, {
    path: 'assets/:assetId',
    getComponent: function getComponent(nextState, cb) {
      beforeLoadingRoute();
      System.import('../pages/Assets/Single').then(loadRoute(cb)).catch(errorLoading);
    },

    indexRoute: {
      getComponent: function getComponent(nextState, cb) {
        beforeLoadingRoute();
        System.import('../pages/Assets/Single/Overview').then(loadRoute(cb)).catch(errorLoading);
      }
    },
    childRoutes: [{
      path: 'reports',
      getComponent: function getComponent(location, cb) {
        beforeLoadingRoute();
        System.import('../pages/Assets/Single/Reports').then(loadRoute(cb)).catch(errorLoading);
      },

      childRoutes: [{
        path: 'bill',
        indexRoute: {
          getComponent: function getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/Assets/Single/Reports/Bill').then(loadRoute(cb)).catch(errorLoading);
          }
        }
      }, {
        path: 'current',
        indexRoute: {
          getComponent: function getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/Assets/Single/Reports/Current').then(loadRoute(cb)).catch(errorLoading);
          }
        }
      }, {
        path: 'energy-saving',
        indexRoute: {
          getComponent: function getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/Assets/Single/Reports/EnergySaving').then(loadRoute(cb)).catch(errorLoading);
          }
        }
      }, {
        path: 'energy',
        indexRoute: {
          getComponent: function getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/Assets/Single/Reports/Energy').then(loadRoute(cb)).catch(errorLoading);
          }
        }
      }, {
        path: 'power',
        indexRoute: {
          getComponent: function getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/Assets/Single/Reports/Power').then(loadRoute(cb)).catch(errorLoading);
          }
        }
      }, {
        path: 'temperature',
        indexRoute: {
          getComponent: function getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/Assets/Single/Reports/Temperature').then(loadRoute(cb)).catch(errorLoading);
          }
        }
      }, {
        path: 'ref_temperature',
        indexRoute: {
          getComponent: function getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/Assets/Single/Reports/RefrigeratorTemperature').then(loadRoute(cb)).catch(errorLoading);
          }
        }
      }, {
        path: 'time',
        indexRoute: {
          getComponent: function getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/Assets/Single/Reports/Time').then(loadRoute(cb)).catch(errorLoading);
          }
        }
      }, {
        path: 'ref_time',
        indexRoute: {
          getComponent: function getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/Assets/Single/Reports/RefrigeratorTime').then(loadRoute(cb)).catch(errorLoading);
          }
        }
      }, {
        path: 'voltage',
        indexRoute: {
          getComponent: function getComponent(location, cb) {
            beforeLoadingRoute();
            System.import('../pages/Assets/Single/Reports/Voltage').then(loadRoute(cb)).catch(errorLoading);
          }
        }
      }]
    }, {
      path: 'posts/:eventId',
      getComponent: function getComponent(location, cb) {
        beforeLoadingRoute();
        System.import('../pages/EventDetail/EventDetailPage').then(loadRoute(cb)).catch(errorLoading);
      }
    }, {
      path: 'device-info',
      getComponent: function getComponent(location, cb) {
        beforeLoadingRoute();
        System.import('../pages/Device/DeviceInfoPage').then(loadRoute(cb)).catch(errorLoading);
      }
    }, {
      path: 'device-network-settings',
      getComponent: function getComponent(location, cb) {
        beforeLoadingRoute();
        System.import('../pages/Device/DeviceNetworkSettingsPage').then(loadRoute(cb)).catch(errorLoading);
      }
    }, {
      path: 'device-general-settings',
      getComponent: function getComponent(location, cb) {
        beforeLoadingRoute();
        System.import('../pages/Device/DeviceGeneralSettingsPage').then(loadRoute(cb)).catch(errorLoading);
      }
    }, {
      path: 'ownership-transfer/:userId',
      getComponent: function getComponent(location, cb) {
        beforeLoadingRoute();
        System.import('../pages/Device/OwnershipTransferUserPage').then(loadRoute(cb)).catch(errorLoading);
      }
    }]
  }]
};
