// bluerain-app-device-explorer/lib/routes/urlBuilder.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.sitePageUrl = sitePageUrl;
exports.assetPageUrl = assetPageUrl;
exports.assetReportPageUrl = assetReportPageUrl;
exports.refAssetReportPageUrl = refAssetReportPageUrl;
exports.deviceSettingsPageUrl = deviceSettingsPageUrl;
exports.sitesGridViewUrl = sitesGridViewUrl;
exports.sitesMapViewUrl = sitesMapViewUrl;
function sitePageUrl(siteId) {
  return '/app/device-explorer/sites/' + siteId;
}

function assetPageUrl(assetId) {
  return '/app/device-explorer/assets/' + assetId;
}

function assetReportPageUrl(assetId, report) {

  if (!report) {
    // return `/app/device-explorer/assets/${assetId}/reports/temperature`;
    return '/app/device-explorer/assets/' + assetId + '/reports/current';
  }

  return '/app/device-explorer/assets/' + assetId + '/reports/' + report;
}

function refAssetReportPageUrl(assetId, report) {

  if (!report) {
    // return `/app/device-explorer/assets/${assetId}/reports/ref_temperature`;
    return '/app/device-explorer/assets/' + assetId + '/reports/ref_time';
  }

  return '/app/device-explorer/assets/' + assetId + '/reports/' + report;
}

function deviceSettingsPageUrl(assetId, settingsPath) {
  // if (settingsPath === 'device-general-settings' || settingsPath === 'device-network-settings' || settingsPath === 'device-info') {
  //   return `/app/device-explorer/assets/${assetId}/${settingsPath}`;
  // }
  return '/app/device-explorer/assets/' + assetId + '/device-general-settings';
}

function sitesGridViewUrl() {
  return '/app/device-explorer';
}

function sitesMapViewUrl() {
  return '/app/device-explorer/asset_locations';
}
