// bluerain-app-device-explorer/src/components/Asset/Charts/BillChart/BillChart.container.js
/**
 * Created by atif on 12/18/16.
 */
import React, { PropTypes } from 'react';
import BillChartComponent from './BillChart.component';

const propTypes = {
  data: PropTypes.object,
  tooltip: PropTypes.bool,
  color: PropTypes.string,
  startDate: PropTypes.string,
  endDate: PropTypes.string,
  label: PropTypes.string,
  value: PropTypes.string,
  showGraph: PropTypes.bool,
  linkTag: PropTypes.oneOfType([
    React.PropTypes.func,
    React.PropTypes.string
  ]),
  to: PropTypes.string
};

const defaultPropTypes = {
  startDate: '01/01/1970',
  endDate: '01/01/2099',
  tooltip: true,
  showGraph: true,
  label: 'Bill',
  linkTag: 'a',
  to: ''
};

export default class BillChartContainer extends React.Component {

  render() {
    const {
      color,
      tooltip,
      data: dbData,
      label,
      linkTag,
      to,
      showGraph,
      ...rest
    } = this.props;


    const data = [];

    let lastReading = 'No data found';
    if (this.props.data === undefined || this.props.data.loading) {
      lastReading = 'Loading data';
    }

    try {
      // let dataFromQuery = dbData.node.events.edges;
      let dataFromQuery = dbData.Event.EventFind.edges;
      dataFromQuery = dataFromQuery.map(item => item.node);

      if (dataFromQuery && dataFromQuery.length > 0) {
        const baseEnergy = dataFromQuery[dataFromQuery.length - 1].data.energy;

        dataFromQuery.forEach((item) => {
          if (item.data && item.data.energy !== undefined) {

            data.push({
              time: new Date(item.time),
              amount: ((item.data.energy - baseEnergy) * 12)
            });
          }
        });
      }
      // Sort Data
      if (data.length > 0) {
        data.sort((a, b) => (a.time > b.time) ? 1 : ((b.time > a.time) ? -1 : 0));
        lastReading = (data[data.length - 1].amount - data[0].amount).toFixed(0);
        lastReading = `Rs. ${lastReading}`;
      }


    } catch (error) {
      // console.info('Bill Chart No Data found');
    }

    if (showGraph) {
      return (
        <BillChartComponent
          data={data}
          x="time"
          y={datum => (datum.amount)}
          label={label}
          value={lastReading}
          tooltip={tooltip}
          linkTag={linkTag}
          to={to}
          {...rest}
        />
      );
    }
    return (
      <BillChartComponent
        color={color}
        label={label}
        value={lastReading}
        tooltip={tooltip}
        linkTag={linkTag}
        to={to}
        {...rest}
      />
    );
  }
}

BillChartContainer.propTypes = propTypes;
BillChartContainer.defaultProps = defaultPropTypes;
