// bluerain-app-device-explorer/lib/components/Asset/AssetControl/SocketControl/SocketControl.container.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _graphqlRelay = require('graphql-relay');

var _SocketControl = require('./SocketControl.component');

var _SocketControl2 = _interopRequireDefault(_SocketControl);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var propTypes = {
  assetId: _react.PropTypes.string,
  assetState: _react.PropTypes.object,
  onError: _react.PropTypes.func,
  mutate: _react.PropTypes.func,
  data: _react.PropTypes.object
};

var defaultProps = {
  assetId: null,
  asset: null,
  viewerQuery: null,
  assetState: {
    standby: true
  },

  mutate: function mutate() {},
  onError: function onError() {}
};

var SocketControl = function (_React$Component) {
  _inherits(SocketControl, _React$Component);

  function SocketControl(props) {
    _classCallCheck(this, SocketControl);

    var _this = _possibleConstructorReturn(this, (SocketControl.__proto__ || Object.getPrototypeOf(SocketControl)).call(this, props));

    _this.persistState = _this.persistState.bind(_this);
    _this.state = props.assetState;
    return _this;
  }

  _createClass(SocketControl, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      try {
        var state = nextProps.data.node.state;

        if (!state) {
          return;
        }

        var newState = Object.assign({}, this.state, state);
        this.setState(newState);
      } catch (e) {
        this.props.onError(e);
      }
    }
  }, {
    key: 'persistState',
    value: function persistState(stateName, stateValue) {
      var _this2 = this;

      // Setup data
      var assetGlobalId = this.props.assetId;
      var assetId = (0, _graphqlRelay.fromGlobalId)(assetGlobalId).id;
      var deviceID = null;
      try {
        deviceID = this.props.data.node.devices.edges[0].node.id;
      } catch (e) {
        return this.props.onError(e);
      }
      deviceID = (0, _graphqlRelay.fromGlobalId)(deviceID).id;
      // Set local state
      var newAssetState = Object.assign({}, this.state, stateValue);
      // newAssetState[stateName] = stateValue;

      var newState = Object.assign({}, this.state, newAssetState);
      this.setState(newState);
      // if (stateName === 'setThermostatMode') {
      //   stateValue.thermostatSetpoint = this.props.data.node.state.thermostatSetpoint;
      //   stateValue.thermostatFanMode = this.props.data.node.state.thermostatFanMode;
      // }
      // Create Object

      var eventObj = {
        type: 'command',
        data: { name: stateName, params: stateValue },
        assetId: assetId,
        deviceId: deviceID
      };

      // Find Viewer
      try {
        var viewerId = this.props.data.viewer.me.id;
        eventObj.sourceAccountId = (0, _graphqlRelay.fromGlobalId)(viewerId).id;
      } catch (e) {
        return this.props.onError(e);
      }
      // Mutate
      this.props.mutate({
        variables: { data: eventObj }
      }).catch(function (error) {
        _this2.props.onError(error);
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var _this3 = this;

      var standby = this.state.standby;


      return _react2.default.createElement(_SocketControl2.default

      // Power
      , { standby: standby,
        onStandbyToggle: function onStandbyToggle(standby, type) {
          return _this3.persistState('setStandby', { standby: standby });
        }
      });
    }
  }]);

  return SocketControl;
}(_react2.default.Component);

SocketControl.propTypes = propTypes;
SocketControl.defaultProps = defaultProps;

exports.default = SocketControl;
