// bluerain-app-device-explorer/src/pages/IndexPage.js
import React, { PropTypes } from 'react';
import { Container, Row, Col } from 'reactstrap';
import { withSystemNav, withLoadingBar } from 'bluerain-client-services';
import AppContent from 'bluerain-ui/lib/Layout/AppContent';
import PrimaryAppNav from 'bluerain-ui/lib/Nav/PrimaryAppNav';
import SecondaryAppNav from 'bluerain-ui/lib/Nav/SecondaryAppNav';
import Button from 'reactstrap/lib/Button';
import { Link, withRouter } from 'react-router';
import SitesGrid from '../components/Site/SitesGrid';
import { sitesGridViewUrl, sitesMapViewUrl } from '../routes/urlBuilder';

const propTypes = {
  title: PropTypes.string,
  showLoading: PropTypes.func,
  systemNav: PropTypes.func,
  location: PropTypes.func
};

const defaultProps = {
  title: 'Sites'
};

class Page extends React.Component {

  componentWillMount() {
    if (this.props.showLoading) {
      this.props.showLoading();
    }
    return null;
  }

  renderPrimaryNav() {
    const title = this.props.title;
    const { toggle } = this.props.systemNav.actions;
    const btnStyle = {
      margin: '0 7px '
    };
    return (
      <PrimaryAppNav title={title} toggleMenu={toggle}>
        <Link to="/app/device-explorer/site/new"><Button color="primary"><i
          className="fa fa-plus hidden-xm-down"
        /><span
          className="hidden-xs-down" style={btnStyle}
        >Add Site</span></Button></Link>&nbsp;
      </PrimaryAppNav>
    );
  }

  renderSecondaryNav() {
    const { pathname } = this.props.location;
    return (<SecondaryAppNav
      items={[
        {
          label: 'Grid View',
          url: '/app/device-explorer',
          class: (pathname === sitesGridViewUrl()) ? 'active' : ''
        }, {
          label: 'Map View',
          url: '/app/device-explorer/asset_locations',
          class: (pathname === sitesMapViewUrl()) ? 'active' : ''
        }
      ]}
    />);
  }

  render() {
    return (
      <div>
        { this.renderPrimaryNav() }
        {this.renderSecondaryNav()}
        <AppContent>
          <Container>
            <Row>
              <Col lg="12" className="pt-2">
                <SitesGrid />
              </Col>
            </Row>
          </Container>
        </AppContent>
      </div>
    );
  }
}

Page.propTypes = propTypes;
Page.defaultProps = defaultProps;

export default withSystemNav(withRouter(withLoadingBar(Page)));


// <Link to="/app/device-explorer/transfer_requests"><Button color="primary"><i className="fa fa-plus hidden-sm-up"  /><span className="hidden-xs-down">Transfer Requests</span></Button></Link>&nbsp;
